// @getopenpay/client/client.js
"use strict";
/**
 * Wrapper class for OpenPay API npm package
 * Don't forget to update these other wrapper classes as well:
 *   - clients/typescript/client.ts
 *   - frontend/next-tailwind/clients/typescript/client.ts
 */
Object.defineProperty(exports, "__esModule", { value: true });
const _1 = require(".");
const utils_1 = require("./utils");
class OpenPayClient {
    constructor(config) {
        this.config = config;
        this.billingMeterEventAdjustmentsApi = new _1.BillingMeterEventAdjustmentsApi(this.config);
        this.billingMeterEventsApi = new _1.BillingMeterEventsApi(this.config);
        this.billingMetersApi = new _1.BillingMetersApi(this.config);
        this.billingPortalApi = new _1.BillingPortalApi(this.config);
        this.chargesApi = new _1.ChargesApi(this.config);
        this.checkoutApi = new _1.CheckoutApi(this.config);
        this.couponsApi = new _1.CouponsApi(this.config);
        this.creditNotesApi = new _1.CreditNotesApi(this.config);
        this.customersApi = new _1.CustomersApi(this.config);
        this.disputesApi = new _1.DisputesApi(this.config);
        this.eventsApi = new _1.EventsApi(this.config);
        this.invoiceItemsApi = new _1.InvoiceItemsApi(this.config);
        this.invoicesApi = new _1.InvoicesApi(this.config);
        this.paymentIntentsApi = new _1.PaymentIntentsApi(this.config);
        this.paymentLinksApi = new _1.PaymentLinksApi(this.config);
        this.paymentMethodsApi = new _1.PaymentMethodsApi(this.config);
        this.pricesApi = new _1.PricesApi(this.config);
        this.productFamilyApi = new _1.ProductFamilyApi(this.config);
        this.productsApi = new _1.ProductsApi(this.config);
        this.promotionCodesApi = new _1.PromotionCodesApi(this.config);
        this.refundsApi = new _1.RefundsApi(this.config);
        this.signupAnswersApi = new _1.SignupAnswersApi(this.config);
        this.subscriptionItemsApi = new _1.SubscriptionItemsApi(this.config);
        this.subscriptionsApi = new _1.SubscriptionsApi(this.config);
        this.taxIntegrationsApi = new _1.TaxIntegrationsApi(this.config);
        this.webhookUtils = new utils_1.WebhookUtils();
    }
}
exports.default = OpenPayClient;
