// @getopenpay/client/apis/CheckoutApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutApi = void 0;
const runtime = __importStar(require("../runtime"));
const index_1 = require("../models/index");
/**
 *
 */
class CheckoutApi extends runtime.BaseAPI {
    /**
     * Create Checkout Session
     */
    createCheckoutSessionRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['createCheckoutSessionRequest'] == null) {
                throw new runtime.RequiredError('createCheckoutSessionRequest', 'Required parameter "createCheckoutSessionRequest" was null or undefined when calling createCheckoutSession().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/checkout/sessions`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.CreateCheckoutSessionRequestToJSON)(requestParameters['createCheckoutSessionRequest']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.CheckoutSessionExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Create Checkout Session
     */
    createCheckoutSession(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.createCheckoutSessionRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Get Checkout Session
     */
    getCheckoutSessionRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['sessionId'] == null) {
                throw new runtime.RequiredError('sessionId', 'Required parameter "sessionId" was null or undefined when calling getCheckoutSession().');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/checkout/sessions/{session_id}`.replace(`{${"session_id"}}`, encodeURIComponent(String(requestParameters['sessionId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.CheckoutSessionExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Get Checkout Session
     */
    getCheckoutSession(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getCheckoutSessionRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * List Checkout Sessions
     */
    listCheckoutSessionsRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['checkoutSessionQueryParams'] == null) {
                throw new runtime.RequiredError('checkoutSessionQueryParams', 'Required parameter "checkoutSessionQueryParams" was null or undefined when calling listCheckoutSessions().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/checkout/list`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.CheckoutSessionQueryParamsToJSON)(requestParameters['checkoutSessionQueryParams']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.ListResponseCheckoutSessionExternalFromJSON)(jsonValue));
        });
    }
    /**
     * List Checkout Sessions
     */
    listCheckoutSessions(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.listCheckoutSessionsRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Validate Checkout Attempt
     */
    validateCheckoutAttemptRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['checkoutAttemptId'] == null) {
                throw new runtime.RequiredError('checkoutAttemptId', 'Required parameter "checkoutAttemptId" was null or undefined when calling validateCheckoutAttempt().');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/checkout/checkout_attempts/validate/{checkout_attempt_id}`.replace(`{${"checkout_attempt_id"}}`, encodeURIComponent(String(requestParameters['checkoutAttemptId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.CheckoutSuccessResponseFromJSON)(jsonValue));
        });
    }
    /**
     * Validate Checkout Attempt
     */
    validateCheckoutAttempt(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.validateCheckoutAttemptRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Verify that a checkout session and payment method belong to the same customer.
     * Verify Checkout Session Payment Method
     */
    verifyCheckoutSessionPaymentMethodRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['checkoutSecureToken'] == null) {
                throw new runtime.RequiredError('checkoutSecureToken', 'Required parameter "checkoutSecureToken" was null or undefined when calling verifyCheckoutSessionPaymentMethod().');
            }
            if (requestParameters['paymentMethodId'] == null) {
                throw new runtime.RequiredError('paymentMethodId', 'Required parameter "paymentMethodId" was null or undefined when calling verifyCheckoutSessionPaymentMethod().');
            }
            const queryParameters = {};
            const headerParameters = {};
            const response = yield this.request({
                path: `/checkout/sessions/{checkout_secure_token}/verify_payment_method/{payment_method_id}`.replace(`{${"checkout_secure_token"}}`, encodeURIComponent(String(requestParameters['checkoutSecureToken']))).replace(`{${"payment_method_id"}}`, encodeURIComponent(String(requestParameters['paymentMethodId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.CheckoutSessionExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Verify that a checkout session and payment method belong to the same customer.
     * Verify Checkout Session Payment Method
     */
    verifyCheckoutSessionPaymentMethod(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.verifyCheckoutSessionPaymentMethodRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
}
exports.CheckoutApi = CheckoutApi;
