// @getopenpay/client/apis/CreditNotesApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreditNotesApi = void 0;
const runtime = __importStar(require("../runtime"));
const index_1 = require("../models/index");
/**
 *
 */
class CreditNotesApi extends runtime.BaseAPI {
    /**
     * Issue a credit note to adjust the amount of a finalized invoice. You may issue multiple credit notes for an invoice. The sum of all credit notes issued for an invoice can’t exceed the  total amount of the invoice. For a paid invoice, the sum of the refund and credit must equal the invoice total. Each credit note will increment the invoice’s pre_payment_credit_notes_amount_atom  or post_payment_credit_notes_amount_atom depending on its status at the time  of credit note creation.
     * Create Credit Note
     */
    createCreditNoteRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['createCreditNoteRequest'] == null) {
                throw new runtime.RequiredError('createCreditNoteRequest', 'Required parameter "createCreditNoteRequest" was null or undefined when calling createCreditNote().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/credit-notes/`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.CreateCreditNoteRequestToJSON)(requestParameters['createCreditNoteRequest']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.CreditNoteExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Issue a credit note to adjust the amount of a finalized invoice. You may issue multiple credit notes for an invoice. The sum of all credit notes issued for an invoice can’t exceed the  total amount of the invoice. For a paid invoice, the sum of the refund and credit must equal the invoice total. Each credit note will increment the invoice’s pre_payment_credit_notes_amount_atom  or post_payment_credit_notes_amount_atom depending on its status at the time  of credit note creation.
     * Create Credit Note
     */
    createCreditNote(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.createCreditNoteRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Get Credit Note
     */
    getCreditNoteRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['creditNoteId'] == null) {
                throw new runtime.RequiredError('creditNoteId', 'Required parameter "creditNoteId" was null or undefined when calling getCreditNote().');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/credit-notes/{credit_note_id}`.replace(`{${"credit_note_id"}}`, encodeURIComponent(String(requestParameters['creditNoteId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.CreditNoteExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Get Credit Note
     */
    getCreditNote(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getCreditNoteRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * List Credit Notes
     */
    listCreditNotesRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['creditNoteQueryParams'] == null) {
                throw new runtime.RequiredError('creditNoteQueryParams', 'Required parameter "creditNoteQueryParams" was null or undefined when calling listCreditNotes().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/credit-notes/list`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.CreditNoteQueryParamsToJSON)(requestParameters['creditNoteQueryParams']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.ListResponseCreditNoteExternalFromJSON)(jsonValue));
        });
    }
    /**
     * List Credit Notes
     */
    listCreditNotes(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.listCreditNotesRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
}
exports.CreditNotesApi = CreditNotesApi;
