// @getopenpay/client/apis/PaymentLinksApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentLinksApi = void 0;
const runtime = __importStar(require("../runtime"));
const index_1 = require("../models/index");
/**
 *
 */
class PaymentLinksApi extends runtime.BaseAPI {
    /**
     * Create Payment Link
     */
    createPaymentLinkRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['createPaymentLinkRequest'] == null) {
                throw new runtime.RequiredError('createPaymentLinkRequest', 'Required parameter "createPaymentLinkRequest" was null or undefined when calling createPaymentLink().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/payment-link/`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.CreatePaymentLinkRequestToJSON)(requestParameters['createPaymentLinkRequest']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PaymentLinkExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Create Payment Link
     */
    createPaymentLink(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.createPaymentLinkRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Get Payment Link
     */
    getPaymentLinkRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['plinkId'] == null) {
                throw new runtime.RequiredError('plinkId', 'Required parameter "plinkId" was null or undefined when calling getPaymentLink().');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/payment-link/{plink_id}`.replace(`{${"plink_id"}}`, encodeURIComponent(String(requestParameters['plinkId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PaymentLinkExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Get Payment Link
     */
    getPaymentLink(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getPaymentLinkRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * List Payment Links
     */
    listPaymentLinksRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['paymentLinkQueryParams'] == null) {
                throw new runtime.RequiredError('paymentLinkQueryParams', 'Required parameter "paymentLinkQueryParams" was null or undefined when calling listPaymentLinks().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/payment-link/list`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.PaymentLinkQueryParamsToJSON)(requestParameters['paymentLinkQueryParams']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.ListResponsePaymentLinkExternalFromJSON)(jsonValue));
        });
    }
    /**
     * List Payment Links
     */
    listPaymentLinks(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.listPaymentLinksRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Open Payment Link Page Public
     */
    openPaymentLinkPagePublicRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['secureToken'] == null) {
                throw new runtime.RequiredError('secureToken', 'Required parameter "secureToken" was null or undefined when calling openPaymentLinkPagePublic().');
            }
            const queryParameters = {};
            if (requestParameters['refId'] != null) {
                queryParameters['ref_id'] = requestParameters['refId'];
            }
            const headerParameters = {};
            const response = yield this.request({
                path: `/payment-link/public/{secure_token}`.replace(`{${"secure_token"}}`, encodeURIComponent(String(requestParameters['secureToken']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            if (this.isJsonMime(response.headers.get('content-type'))) {
                return new runtime.JSONApiResponse(response);
            }
            else {
                return new runtime.TextApiResponse(response);
            }
        });
    }
    /**
     * Open Payment Link Page Public
     */
    openPaymentLinkPagePublic(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.openPaymentLinkPagePublicRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
}
exports.PaymentLinksApi = PaymentLinksApi;
