// @getopenpay/client/models/ActiveSubResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfActiveSubResponse = instanceOfActiveSubResponse;
exports.ActiveSubResponseFromJSON = ActiveSubResponseFromJSON;
exports.ActiveSubResponseFromJSONTyped = ActiveSubResponseFromJSONTyped;
exports.ActiveSubResponseToJSON = ActiveSubResponseToJSON;
exports.ActiveSubResponseToJSONTyped = ActiveSubResponseToJSONTyped;
const SubscriptionExternal_1 = require("./SubscriptionExternal");
/**
 * Check if a given object implements the ActiveSubResponse interface.
 */
function instanceOfActiveSubResponse(value) {
    if (!('hasActiveSubscriptions' in value) || value['hasActiveSubscriptions'] === undefined)
        return false;
    if (!('subscriptions' in value) || value['subscriptions'] === undefined)
        return false;
    return true;
}
function ActiveSubResponseFromJSON(json) {
    return ActiveSubResponseFromJSONTyped(json, false);
}
function ActiveSubResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'hasActiveSubscriptions': json['has_active_subscriptions'],
        'subscriptions': (json['subscriptions'].map(SubscriptionExternal_1.SubscriptionExternalFromJSON)),
    };
}
function ActiveSubResponseToJSON(json) {
    return ActiveSubResponseToJSONTyped(json, false);
}
function ActiveSubResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'has_active_subscriptions': value['hasActiveSubscriptions'],
        'subscriptions': (value['subscriptions'].map(SubscriptionExternal_1.SubscriptionExternalToJSON)),
    };
}
