// @getopenpay/client/models/AddCommentRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAddCommentRequest = instanceOfAddCommentRequest;
exports.AddCommentRequestFromJSON = AddCommentRequestFromJSON;
exports.AddCommentRequestFromJSONTyped = AddCommentRequestFromJSONTyped;
exports.AddCommentRequestToJSON = AddCommentRequestToJSON;
exports.AddCommentRequestToJSONTyped = AddCommentRequestToJSONTyped;
/**
 * Check if a given object implements the AddCommentRequest interface.
 */
function instanceOfAddCommentRequest(value) {
    if (!('comment' in value) || value['comment'] === undefined)
        return false;
    return true;
}
function AddCommentRequestFromJSON(json) {
    return AddCommentRequestFromJSONTyped(json, false);
}
function AddCommentRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'comment': json['comment'],
    };
}
function AddCommentRequestToJSON(json) {
    return AddCommentRequestToJSONTyped(json, false);
}
function AddCommentRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'comment': value['comment'],
    };
}
