// @getopenpay/client/models/AttachPaymentMethodRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAttachPaymentMethodRequest = instanceOfAttachPaymentMethodRequest;
exports.AttachPaymentMethodRequestFromJSON = AttachPaymentMethodRequestFromJSON;
exports.AttachPaymentMethodRequestFromJSONTyped = AttachPaymentMethodRequestFromJSONTyped;
exports.AttachPaymentMethodRequestToJSON = AttachPaymentMethodRequestToJSON;
exports.AttachPaymentMethodRequestToJSONTyped = AttachPaymentMethodRequestToJSONTyped;
/**
 * Check if a given object implements the AttachPaymentMethodRequest interface.
 */
function instanceOfAttachPaymentMethodRequest(value) {
    if (!('customerId' in value) || value['customerId'] === undefined)
        return false;
    return true;
}
function AttachPaymentMethodRequestFromJSON(json) {
    return AttachPaymentMethodRequestFromJSONTyped(json, false);
}
function AttachPaymentMethodRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'customerId': json['customer_id'],
    };
}
function AttachPaymentMethodRequestToJSON(json) {
    return AttachPaymentMethodRequestToJSONTyped(json, false);
}
function AttachPaymentMethodRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'customer_id': value['customerId'],
    };
}
