// @getopenpay/client/models/BillingMeterEventAdjustmentStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingMeterEventAdjustmentStatus = void 0;
exports.instanceOfBillingMeterEventAdjustmentStatus = instanceOfBillingMeterEventAdjustmentStatus;
exports.BillingMeterEventAdjustmentStatusFromJSON = BillingMeterEventAdjustmentStatusFromJSON;
exports.BillingMeterEventAdjustmentStatusFromJSONTyped = BillingMeterEventAdjustmentStatusFromJSONTyped;
exports.BillingMeterEventAdjustmentStatusToJSON = BillingMeterEventAdjustmentStatusToJSON;
exports.BillingMeterEventAdjustmentStatusToJSONTyped = BillingMeterEventAdjustmentStatusToJSONTyped;
/**
 *
 * @export
 */
exports.BillingMeterEventAdjustmentStatus = {
    Pending: 'pending',
    Complete: 'complete'
};
function instanceOfBillingMeterEventAdjustmentStatus(value) {
    for (const key in exports.BillingMeterEventAdjustmentStatus) {
        if (Object.prototype.hasOwnProperty.call(exports.BillingMeterEventAdjustmentStatus, key)) {
            if (exports.BillingMeterEventAdjustmentStatus[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function BillingMeterEventAdjustmentStatusFromJSON(json) {
    return BillingMeterEventAdjustmentStatusFromJSONTyped(json, false);
}
function BillingMeterEventAdjustmentStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function BillingMeterEventAdjustmentStatusToJSON(value) {
    return value;
}
function BillingMeterEventAdjustmentStatusToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
