// @getopenpay/client/models/BillingMeterEventStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingMeterEventStatus = void 0;
exports.instanceOfBillingMeterEventStatus = instanceOfBillingMeterEventStatus;
exports.BillingMeterEventStatusFromJSON = BillingMeterEventStatusFromJSON;
exports.BillingMeterEventStatusFromJSONTyped = BillingMeterEventStatusFromJSONTyped;
exports.BillingMeterEventStatusToJSON = BillingMeterEventStatusToJSON;
exports.BillingMeterEventStatusToJSONTyped = BillingMeterEventStatusToJSONTyped;
/**
 *
 * @export
 */
exports.BillingMeterEventStatus = {
    Active: 'active',
    Canceled: 'canceled',
    Billed: 'billed'
};
function instanceOfBillingMeterEventStatus(value) {
    for (const key in exports.BillingMeterEventStatus) {
        if (Object.prototype.hasOwnProperty.call(exports.BillingMeterEventStatus, key)) {
            if (exports.BillingMeterEventStatus[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function BillingMeterEventStatusFromJSON(json) {
    return BillingMeterEventStatusFromJSONTyped(json, false);
}
function BillingMeterEventStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function BillingMeterEventStatusToJSON(value) {
    return value;
}
function BillingMeterEventStatusToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
