// @getopenpay/client/models/BillingReasonEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingReasonEnum = void 0;
exports.instanceOfBillingReasonEnum = instanceOfBillingReasonEnum;
exports.BillingReasonEnumFromJSON = BillingReasonEnumFromJSON;
exports.BillingReasonEnumFromJSONTyped = BillingReasonEnumFromJSONTyped;
exports.BillingReasonEnumToJSON = BillingReasonEnumToJSON;
exports.BillingReasonEnumToJSONTyped = BillingReasonEnumToJSONTyped;
/**
 *
 * @export
 */
exports.BillingReasonEnum = {
    Manual: 'manual',
    SubscriptionCreate: 'subscription_create',
    SubscriptionCycle: 'subscription_cycle',
    SubscriptionThreshold: 'subscription_threshold',
    SubscriptionUpdate: 'subscription_update',
    Upcoming: 'upcoming'
};
function instanceOfBillingReasonEnum(value) {
    for (const key in exports.BillingReasonEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.BillingReasonEnum, key)) {
            if (exports.BillingReasonEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function BillingReasonEnumFromJSON(json) {
    return BillingReasonEnumFromJSONTyped(json, false);
}
function BillingReasonEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function BillingReasonEnumToJSON(value) {
    return value;
}
function BillingReasonEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
