// @getopenpay/client/models/BillingMeterAdjustmentType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingMeterAdjustmentType = void 0;
exports.instanceOfBillingMeterAdjustmentType = instanceOfBillingMeterAdjustmentType;
exports.BillingMeterAdjustmentTypeFromJSON = BillingMeterAdjustmentTypeFromJSON;
exports.BillingMeterAdjustmentTypeFromJSONTyped = BillingMeterAdjustmentTypeFromJSONTyped;
exports.BillingMeterAdjustmentTypeToJSON = BillingMeterAdjustmentTypeToJSON;
exports.BillingMeterAdjustmentTypeToJSONTyped = BillingMeterAdjustmentTypeToJSONTyped;
/**
 *
 * @export
 */
exports.BillingMeterAdjustmentType = {
    Cancel: 'cancel'
};
function instanceOfBillingMeterAdjustmentType(value) {
    for (const key in exports.BillingMeterAdjustmentType) {
        if (Object.prototype.hasOwnProperty.call(exports.BillingMeterAdjustmentType, key)) {
            if (exports.BillingMeterAdjustmentType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function BillingMeterAdjustmentTypeFromJSON(json) {
    return BillingMeterAdjustmentTypeFromJSONTyped(json, false);
}
function BillingMeterAdjustmentTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function BillingMeterAdjustmentTypeToJSON(value) {
    return value;
}
function BillingMeterAdjustmentTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
