// @getopenpay/client/models/BillingMeterAggregationFormula.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingMeterAggregationFormula = void 0;
exports.instanceOfBillingMeterAggregationFormula = instanceOfBillingMeterAggregationFormula;
exports.BillingMeterAggregationFormulaFromJSON = BillingMeterAggregationFormulaFromJSON;
exports.BillingMeterAggregationFormulaFromJSONTyped = BillingMeterAggregationFormulaFromJSONTyped;
exports.BillingMeterAggregationFormulaToJSON = BillingMeterAggregationFormulaToJSON;
exports.BillingMeterAggregationFormulaToJSONTyped = BillingMeterAggregationFormulaToJSONTyped;
/**
 *
 * @export
 */
exports.BillingMeterAggregationFormula = {
    Sum: 'sum',
    Count: 'count'
};
function instanceOfBillingMeterAggregationFormula(value) {
    for (const key in exports.BillingMeterAggregationFormula) {
        if (Object.prototype.hasOwnProperty.call(exports.BillingMeterAggregationFormula, key)) {
            if (exports.BillingMeterAggregationFormula[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function BillingMeterAggregationFormulaFromJSON(json) {
    return BillingMeterAggregationFormulaFromJSONTyped(json, false);
}
function BillingMeterAggregationFormulaFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function BillingMeterAggregationFormulaToJSON(value) {
    return value;
}
function BillingMeterAggregationFormulaToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
