// @getopenpay/client/models/CalendarIntervalEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalendarIntervalEnum = void 0;
exports.instanceOfCalendarIntervalEnum = instanceOfCalendarIntervalEnum;
exports.CalendarIntervalEnumFromJSON = CalendarIntervalEnumFromJSON;
exports.CalendarIntervalEnumFromJSONTyped = CalendarIntervalEnumFromJSONTyped;
exports.CalendarIntervalEnumToJSON = CalendarIntervalEnumToJSON;
exports.CalendarIntervalEnumToJSONTyped = CalendarIntervalEnumToJSONTyped;
/**
 *
 * @export
 */
exports.CalendarIntervalEnum = {
    Day: 'day',
    Week: 'week',
    Month: 'month',
    Year: 'year'
};
function instanceOfCalendarIntervalEnum(value) {
    for (const key in exports.CalendarIntervalEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.CalendarIntervalEnum, key)) {
            if (exports.CalendarIntervalEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CalendarIntervalEnumFromJSON(json) {
    return CalendarIntervalEnumFromJSONTyped(json, false);
}
function CalendarIntervalEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CalendarIntervalEnumToJSON(value) {
    return value;
}
function CalendarIntervalEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
