// @getopenpay/client/models/ChargeQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfChargeQueryParams = instanceOfChargeQueryParams;
exports.ChargeQueryParamsFromJSON = ChargeQueryParamsFromJSON;
exports.ChargeQueryParamsFromJSONTyped = ChargeQueryParamsFromJSONTyped;
exports.ChargeQueryParamsToJSON = ChargeQueryParamsToJSON;
exports.ChargeQueryParamsToJSONTyped = ChargeQueryParamsToJSONTyped;
const DateTimeFilter_1 = require("./DateTimeFilter");
const ChargeStatusEnum_1 = require("./ChargeStatusEnum");
/**
 * Check if a given object implements the ChargeQueryParams interface.
 */
function instanceOfChargeQueryParams(value) {
    return true;
}
function ChargeQueryParamsFromJSON(json) {
    return ChargeQueryParamsFromJSONTyped(json, false);
}
function ChargeQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'customerId': json['customer_id'] == null ? undefined : json['customer_id'],
        'expand': json['expand'] == null ? undefined : json['expand'],
        'invoiceId': json['invoice_id'] == null ? undefined : json['invoice_id'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'paymentIntentId': json['payment_intent_id'] == null ? undefined : json['payment_intent_id'],
        'paymentMethodId': json['payment_method_id'] == null ? undefined : json['payment_method_id'],
        'refunded': json['refunded'] == null ? undefined : json['refunded'],
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'status': json['status'] == null ? undefined : (0, ChargeStatusEnum_1.ChargeStatusEnumFromJSON)(json['status']),
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function ChargeQueryParamsToJSON(json) {
    return ChargeQueryParamsToJSONTyped(json, false);
}
function ChargeQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'customer_id': value['customerId'],
        'expand': value['expand'],
        'invoice_id': value['invoiceId'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'payment_intent_id': value['paymentIntentId'],
        'payment_method_id': value['paymentMethodId'],
        'refunded': value['refunded'],
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'status': (0, ChargeStatusEnum_1.ChargeStatusEnumToJSON)(value['status']),
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
