// @getopenpay/client/models/CheckoutProcessorsPreferences.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCheckoutProcessorsPreferences = instanceOfCheckoutProcessorsPreferences;
exports.CheckoutProcessorsPreferencesFromJSON = CheckoutProcessorsPreferencesFromJSON;
exports.CheckoutProcessorsPreferencesFromJSONTyped = CheckoutProcessorsPreferencesFromJSONTyped;
exports.CheckoutProcessorsPreferencesToJSON = CheckoutProcessorsPreferencesToJSON;
exports.CheckoutProcessorsPreferencesToJSONTyped = CheckoutProcessorsPreferencesToJSONTyped;
const PaymentProviderType_1 = require("./PaymentProviderType");
/**
 * Check if a given object implements the CheckoutProcessorsPreferences interface.
 */
function instanceOfCheckoutProcessorsPreferences(value) {
    return true;
}
function CheckoutProcessorsPreferencesFromJSON(json) {
    return CheckoutProcessorsPreferencesFromJSONTyped(json, false);
}
function CheckoutProcessorsPreferencesFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'filterProviders': json['filter_providers'] == null ? undefined : (json['filter_providers'].map(PaymentProviderType_1.PaymentProviderTypeFromJSON)),
        'idsWhitelist': json['ids_whitelist'] == null ? undefined : json['ids_whitelist'],
        'namesWhitelist': json['names_whitelist'] == null ? undefined : json['names_whitelist'],
    };
}
function CheckoutProcessorsPreferencesToJSON(json) {
    return CheckoutProcessorsPreferencesToJSONTyped(json, false);
}
function CheckoutProcessorsPreferencesToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'filter_providers': value['filterProviders'] == null ? undefined : (value['filterProviders'].map(PaymentProviderType_1.PaymentProviderTypeToJSON)),
        'ids_whitelist': value['idsWhitelist'],
        'names_whitelist': value['namesWhitelist'],
    };
}
