// @getopenpay/client/models/CheckoutPreferences.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCheckoutPreferences = instanceOfCheckoutPreferences;
exports.CheckoutPreferencesFromJSON = CheckoutPreferencesFromJSON;
exports.CheckoutPreferencesFromJSONTyped = CheckoutPreferencesFromJSONTyped;
exports.CheckoutPreferencesToJSON = CheckoutPreferencesToJSON;
exports.CheckoutPreferencesToJSONTyped = CheckoutPreferencesToJSONTyped;
const CheckoutProcessorsPreferences_1 = require("./CheckoutProcessorsPreferences");
const FallbackConfigurationInput_1 = require("./FallbackConfigurationInput");
/**
 * Check if a given object implements the CheckoutPreferences interface.
 */
function instanceOfCheckoutPreferences(value) {
    return true;
}
function CheckoutPreferencesFromJSON(json) {
    return CheckoutPreferencesFromJSONTyped(json, false);
}
function CheckoutPreferencesFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fallbackCascadePreferences': json['fallback_cascade_preferences'] == null ? undefined : (json['fallback_cascade_preferences'].map(FallbackConfigurationInput_1.FallbackConfigurationInputFromJSON)),
        'fallbackCascadeSelectedProductPriceQuantity': json['fallback_cascade_selected_product_price_quantity'] == null ? undefined : json['fallback_cascade_selected_product_price_quantity'],
        'processorPreferences': json['processor_preferences'] == null ? undefined : (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesFromJSON)(json['processor_preferences']),
    };
}
function CheckoutPreferencesToJSON(json) {
    return CheckoutPreferencesToJSONTyped(json, false);
}
function CheckoutPreferencesToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'fallback_cascade_preferences': value['fallbackCascadePreferences'] == null ? undefined : (value['fallbackCascadePreferences'].map(FallbackConfigurationInput_1.FallbackConfigurationInputToJSON)),
        'fallback_cascade_selected_product_price_quantity': value['fallbackCascadeSelectedProductPriceQuantity'],
        'processor_preferences': (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesToJSON)(value['processorPreferences']),
    };
}
