// @getopenpay/client/models/CheckoutPreferencesOutput.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCheckoutPreferencesOutput = instanceOfCheckoutPreferencesOutput;
exports.CheckoutPreferencesOutputFromJSON = CheckoutPreferencesOutputFromJSON;
exports.CheckoutPreferencesOutputFromJSONTyped = CheckoutPreferencesOutputFromJSONTyped;
exports.CheckoutPreferencesOutputToJSON = CheckoutPreferencesOutputToJSON;
exports.CheckoutPreferencesOutputToJSONTyped = CheckoutPreferencesOutputToJSONTyped;
const CheckoutProcessorsPreferences_1 = require("./CheckoutProcessorsPreferences");
const FallbackConfigurationOutput_1 = require("./FallbackConfigurationOutput");
/**
 * Check if a given object implements the CheckoutPreferencesOutput interface.
 */
function instanceOfCheckoutPreferencesOutput(value) {
    return true;
}
function CheckoutPreferencesOutputFromJSON(json) {
    return CheckoutPreferencesOutputFromJSONTyped(json, false);
}
function CheckoutPreferencesOutputFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fallbackCascadePreferences': json['fallback_cascade_preferences'] == null ? undefined : (json['fallback_cascade_preferences'].map(FallbackConfigurationOutput_1.FallbackConfigurationOutputFromJSON)),
        'fallbackCascadeSelectedProductPriceQuantity': json['fallback_cascade_selected_product_price_quantity'] == null ? undefined : json['fallback_cascade_selected_product_price_quantity'],
        'processorPreferences': json['processor_preferences'] == null ? undefined : (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesFromJSON)(json['processor_preferences']),
        'unused': json['unused'] == null ? undefined : json['unused'],
    };
}
function CheckoutPreferencesOutputToJSON(json) {
    return CheckoutPreferencesOutputToJSONTyped(json, false);
}
function CheckoutPreferencesOutputToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'fallback_cascade_preferences': value['fallbackCascadePreferences'] == null ? undefined : (value['fallbackCascadePreferences'].map(FallbackConfigurationOutput_1.FallbackConfigurationOutputToJSON)),
        'fallback_cascade_selected_product_price_quantity': value['fallbackCascadeSelectedProductPriceQuantity'],
        'processor_preferences': (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesToJSON)(value['processorPreferences']),
        'unused': value['unused'],
    };
}
