// @getopenpay/client/models/CompleteAddress.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCompleteAddress = instanceOfCompleteAddress;
exports.CompleteAddressFromJSON = CompleteAddressFromJSON;
exports.CompleteAddressFromJSONTyped = CompleteAddressFromJSONTyped;
exports.CompleteAddressToJSON = CompleteAddressToJSON;
exports.CompleteAddressToJSONTyped = CompleteAddressToJSONTyped;
/**
 * Check if a given object implements the CompleteAddress interface.
 */
function instanceOfCompleteAddress(value) {
    return true;
}
function CompleteAddressFromJSON(json) {
    return CompleteAddressFromJSONTyped(json, false);
}
function CompleteAddressFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'city': json['city'] == null ? undefined : json['city'],
        'country': json['country'] == null ? undefined : json['country'],
        'line1': json['line1'] == null ? undefined : json['line1'],
        'line2': json['line2'] == null ? undefined : json['line2'],
        'line3': json['line3'] == null ? undefined : json['line3'],
        'state': json['state'] == null ? undefined : json['state'],
        'zipCode': json['zip_code'] == null ? undefined : json['zip_code'],
    };
}
function CompleteAddressToJSON(json) {
    return CompleteAddressToJSONTyped(json, false);
}
function CompleteAddressToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'city': value['city'],
        'country': value['country'],
        'line1': value['line1'],
        'line2': value['line2'],
        'line3': value['line3'],
        'state': value['state'],
        'zip_code': value['zipCode'],
    };
}
