// @getopenpay/client/models/CouponDuration.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CouponDuration = void 0;
exports.instanceOfCouponDuration = instanceOfCouponDuration;
exports.CouponDurationFromJSON = CouponDurationFromJSON;
exports.CouponDurationFromJSONTyped = CouponDurationFromJSONTyped;
exports.CouponDurationToJSON = CouponDurationToJSON;
exports.CouponDurationToJSONTyped = CouponDurationToJSONTyped;
/**
 *
 * @export
 */
exports.CouponDuration = {
    Once: 'once',
    Repeating: 'repeating',
    Forever: 'forever'
};
function instanceOfCouponDuration(value) {
    for (const key in exports.CouponDuration) {
        if (Object.prototype.hasOwnProperty.call(exports.CouponDuration, key)) {
            if (exports.CouponDuration[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CouponDurationFromJSON(json) {
    return CouponDurationFromJSONTyped(json, false);
}
function CouponDurationFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CouponDurationToJSON(value) {
    return value;
}
function CouponDurationToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
