// @getopenpay/client/models/CreateCustomerRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateCustomerRequest = instanceOfCreateCustomerRequest;
exports.CreateCustomerRequestFromJSON = CreateCustomerRequestFromJSON;
exports.CreateCustomerRequestFromJSONTyped = CreateCustomerRequestFromJSONTyped;
exports.CreateCustomerRequestToJSON = CreateCustomerRequestToJSON;
exports.CreateCustomerRequestToJSONTyped = CreateCustomerRequestToJSONTyped;
const CustomerInvoiceSettings_1 = require("./CustomerInvoiceSettings");
const CompleteAddress_1 = require("./CompleteAddress");
const CustomerLanguage_1 = require("./CustomerLanguage");
/**
 * Check if a given object implements the CreateCustomerRequest interface.
 */
function instanceOfCreateCustomerRequest(value) {
    if (!('email' in value) || value['email'] === undefined)
        return false;
    return true;
}
function CreateCustomerRequestFromJSON(json) {
    return CreateCustomerRequestFromJSONTyped(json, false);
}
function CreateCustomerRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'billingEmail': json['billing_email'] == null ? undefined : json['billing_email'],
        'businessName': json['business_name'] == null ? undefined : json['business_name'],
        'city': json['city'] == null ? undefined : json['city'],
        'country': json['country'] == null ? undefined : json['country'],
        'couponId': json['coupon_id'] == null ? undefined : json['coupon_id'],
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'customerBillingAddress': json['customer_billing_address'] == null ? undefined : (0, CompleteAddress_1.CompleteAddressFromJSON)(json['customer_billing_address']),
        'email': json['email'],
        'firstName': json['first_name'] == null ? undefined : json['first_name'],
        'invoiceSettings': json['invoice_settings'] == null ? undefined : (0, CustomerInvoiceSettings_1.CustomerInvoiceSettingsFromJSON)(json['invoice_settings']),
        'language': json['language'] == null ? undefined : (0, CustomerLanguage_1.CustomerLanguageFromJSON)(json['language']),
        'lastName': json['last_name'] == null ? undefined : json['last_name'],
        'line1': json['line1'] == null ? undefined : json['line1'],
        'line2': json['line2'] == null ? undefined : json['line2'],
        'line3': json['line3'] == null ? undefined : json['line3'],
        'notes': json['notes'] == null ? undefined : json['notes'],
        'phoneNumber': json['phone_number'] == null ? undefined : json['phone_number'],
        'promotionCodeId': json['promotion_code_id'] == null ? undefined : json['promotion_code_id'],
        'shippingAddresses': json['shipping_addresses'] == null ? undefined : (json['shipping_addresses'].map(CompleteAddress_1.CompleteAddressFromJSON)),
        'state': json['state'] == null ? undefined : json['state'],
        'zipCode': json['zip_code'] == null ? undefined : json['zip_code'],
    };
}
function CreateCustomerRequestToJSON(json) {
    return CreateCustomerRequestToJSONTyped(json, false);
}
function CreateCustomerRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'billing_email': value['billingEmail'],
        'business_name': value['businessName'],
        'city': value['city'],
        'country': value['country'],
        'coupon_id': value['couponId'],
        'custom_fields': value['customFields'],
        'customer_billing_address': (0, CompleteAddress_1.CompleteAddressToJSON)(value['customerBillingAddress']),
        'email': value['email'],
        'first_name': value['firstName'],
        'invoice_settings': (0, CustomerInvoiceSettings_1.CustomerInvoiceSettingsToJSON)(value['invoiceSettings']),
        'language': (0, CustomerLanguage_1.CustomerLanguageToJSON)(value['language']),
        'last_name': value['lastName'],
        'line1': value['line1'],
        'line2': value['line2'],
        'line3': value['line3'],
        'notes': value['notes'],
        'phone_number': value['phoneNumber'],
        'promotion_code_id': value['promotionCodeId'],
        'shipping_addresses': value['shippingAddresses'] == null ? undefined : (value['shippingAddresses'].map(CompleteAddress_1.CompleteAddressToJSON)),
        'state': value['state'],
        'zip_code': value['zipCode'],
    };
}
