// @getopenpay/client/models/CreateCreditNoteRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateCreditNoteRequest = instanceOfCreateCreditNoteRequest;
exports.CreateCreditNoteRequestFromJSON = CreateCreditNoteRequestFromJSON;
exports.CreateCreditNoteRequestFromJSONTyped = CreateCreditNoteRequestFromJSONTyped;
exports.CreateCreditNoteRequestToJSON = CreateCreditNoteRequestToJSON;
exports.CreateCreditNoteRequestToJSONTyped = CreateCreditNoteRequestToJSONTyped;
const CreditNoteReason_1 = require("./CreditNoteReason");
const CreateCreditNoteLine_1 = require("./CreateCreditNoteLine");
const CurrencyEnum_1 = require("./CurrencyEnum");
/**
 * Check if a given object implements the CreateCreditNoteRequest interface.
 */
function instanceOfCreateCreditNoteRequest(value) {
    if (!('invoiceId' in value) || value['invoiceId'] === undefined)
        return false;
    if (!('lines' in value) || value['lines'] === undefined)
        return false;
    if (!('totalAmountAtom' in value) || value['totalAmountAtom'] === undefined)
        return false;
    return true;
}
function CreateCreditNoteRequestFromJSON(json) {
    return CreateCreditNoteRequestFromJSONTyped(json, false);
}
function CreateCreditNoteRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'creditAmountAtom': json['credit_amount_atom'] == null ? undefined : json['credit_amount_atom'],
        'currency': json['currency'] == null ? undefined : (0, CurrencyEnum_1.CurrencyEnumFromJSON)(json['currency']),
        'invoiceId': json['invoice_id'],
        'lines': (json['lines'].map(CreateCreditNoteLine_1.CreateCreditNoteLineFromJSON)),
        'reason': json['reason'] == null ? undefined : (0, CreditNoteReason_1.CreditNoteReasonFromJSON)(json['reason']),
        'refundAmountAtom': json['refund_amount_atom'] == null ? undefined : json['refund_amount_atom'],
        'totalAmountAtom': json['total_amount_atom'],
    };
}
function CreateCreditNoteRequestToJSON(json) {
    return CreateCreditNoteRequestToJSONTyped(json, false);
}
function CreateCreditNoteRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'credit_amount_atom': value['creditAmountAtom'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumToJSON)(value['currency']),
        'invoice_id': value['invoiceId'],
        'lines': (value['lines'].map(CreateCreditNoteLine_1.CreateCreditNoteLineToJSON)),
        'reason': (0, CreditNoteReason_1.CreditNoteReasonToJSON)(value['reason']),
        'refund_amount_atom': value['refundAmountAtom'],
        'total_amount_atom': value['totalAmountAtom'],
    };
}
