// @getopenpay/client/models/CreateCreditNoteLine.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateCreditNoteLine = instanceOfCreateCreditNoteLine;
exports.CreateCreditNoteLineFromJSON = CreateCreditNoteLineFromJSON;
exports.CreateCreditNoteLineFromJSONTyped = CreateCreditNoteLineFromJSONTyped;
exports.CreateCreditNoteLineToJSON = CreateCreditNoteLineToJSON;
exports.CreateCreditNoteLineToJSONTyped = CreateCreditNoteLineToJSONTyped;
const CreditNoteLineType_1 = require("./CreditNoteLineType");
const CurrencyEnum_1 = require("./CurrencyEnum");
/**
 * Check if a given object implements the CreateCreditNoteLine interface.
 */
function instanceOfCreateCreditNoteLine(value) {
    if (!('amountAtom' in value) || value['amountAtom'] === undefined)
        return false;
    if (!('currency' in value) || value['currency'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    return true;
}
function CreateCreditNoteLineFromJSON(json) {
    return CreateCreditNoteLineFromJSONTyped(json, false);
}
function CreateCreditNoteLineFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amountAtom': json['amount_atom'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumFromJSON)(json['currency']),
        'invoiceItemId': json['invoice_item_id'] == null ? undefined : json['invoice_item_id'],
        'quantity': json['quantity'] == null ? undefined : json['quantity'],
        'type': (0, CreditNoteLineType_1.CreditNoteLineTypeFromJSON)(json['type']),
        'unitAmountAtom': json['unit_amount_atom'] == null ? undefined : json['unit_amount_atom'],
    };
}
function CreateCreditNoteLineToJSON(json) {
    return CreateCreditNoteLineToJSONTyped(json, false);
}
function CreateCreditNoteLineToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'amount_atom': value['amountAtom'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumToJSON)(value['currency']),
        'invoice_item_id': value['invoiceItemId'],
        'quantity': value['quantity'],
        'type': (0, CreditNoteLineType_1.CreditNoteLineTypeToJSON)(value['type']),
        'unit_amount_atom': value['unitAmountAtom'],
    };
}
