// @getopenpay/client/models/CreateDisputeRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateDisputeRequest = instanceOfCreateDisputeRequest;
exports.CreateDisputeRequestFromJSON = CreateDisputeRequestFromJSON;
exports.CreateDisputeRequestFromJSONTyped = CreateDisputeRequestFromJSONTyped;
exports.CreateDisputeRequestToJSON = CreateDisputeRequestToJSON;
exports.CreateDisputeRequestToJSONTyped = CreateDisputeRequestToJSONTyped;
/**
 * Check if a given object implements the CreateDisputeRequest interface.
 */
function instanceOfCreateDisputeRequest(value) {
    return true;
}
function CreateDisputeRequestFromJSON(json) {
    return CreateDisputeRequestFromJSONTyped(json, false);
}
function CreateDisputeRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'meta': json['meta'] == null ? undefined : json['meta'],
        'paymentIntentId': json['payment_intent_id'] == null ? undefined : json['payment_intent_id'],
        'reason': json['reason'] == null ? undefined : json['reason'],
        'theirDisputeId': json['their_dispute_id'] == null ? undefined : json['their_dispute_id'],
        'theirPaymentIntentId': json['their_payment_intent_id'] == null ? undefined : json['their_payment_intent_id'],
    };
}
function CreateDisputeRequestToJSON(json) {
    return CreateDisputeRequestToJSONTyped(json, false);
}
function CreateDisputeRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'meta': value['meta'],
        'payment_intent_id': value['paymentIntentId'],
        'reason': value['reason'],
        'their_dispute_id': value['theirDisputeId'],
        'their_payment_intent_id': value['theirPaymentIntentId'],
    };
}
