// @getopenpay/client/models/CreateCheckoutLineItem.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateCheckoutLineItem = instanceOfCreateCheckoutLineItem;
exports.CreateCheckoutLineItemFromJSON = CreateCheckoutLineItemFromJSON;
exports.CreateCheckoutLineItemFromJSONTyped = CreateCheckoutLineItemFromJSONTyped;
exports.CreateCheckoutLineItemToJSON = CreateCheckoutLineItemToJSON;
exports.CreateCheckoutLineItemToJSONTyped = CreateCheckoutLineItemToJSONTyped;
/**
 * Check if a given object implements the CreateCheckoutLineItem interface.
 */
function instanceOfCreateCheckoutLineItem(value) {
    if (!('priceId' in value) || value['priceId'] === undefined)
        return false;
    if (!('quantity' in value) || value['quantity'] === undefined)
        return false;
    return true;
}
function CreateCheckoutLineItemFromJSON(json) {
    return CreateCheckoutLineItemFromJSONTyped(json, false);
}
function CreateCheckoutLineItemFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'description': json['description'] == null ? undefined : json['description'],
        'priceId': json['price_id'],
        'quantity': json['quantity'],
    };
}
function CreateCheckoutLineItemToJSON(json) {
    return CreateCheckoutLineItemToJSONTyped(json, false);
}
function CreateCheckoutLineItemToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'custom_fields': value['customFields'],
        'description': value['description'],
        'price_id': value['priceId'],
        'quantity': value['quantity'],
    };
}
