// @getopenpay/client/models/CreateTaxIntegrationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateTaxIntegrationRequest = instanceOfCreateTaxIntegrationRequest;
exports.CreateTaxIntegrationRequestFromJSON = CreateTaxIntegrationRequestFromJSON;
exports.CreateTaxIntegrationRequestFromJSONTyped = CreateTaxIntegrationRequestFromJSONTyped;
exports.CreateTaxIntegrationRequestToJSON = CreateTaxIntegrationRequestToJSON;
exports.CreateTaxIntegrationRequestToJSONTyped = CreateTaxIntegrationRequestToJSONTyped;
const TaxIntegrationApiName_1 = require("./TaxIntegrationApiName");
/**
 * Check if a given object implements the CreateTaxIntegrationRequest interface.
 */
function instanceOfCreateTaxIntegrationRequest(value) {
    if (!('accountingTz' in value) || value['accountingTz'] === undefined)
        return false;
    if (!('apiKeys' in value) || value['apiKeys'] === undefined)
        return false;
    if (!('apiName' in value) || value['apiName'] === undefined)
        return false;
    return true;
}
function CreateTaxIntegrationRequestFromJSON(json) {
    return CreateTaxIntegrationRequestFromJSONTyped(json, false);
}
function CreateTaxIntegrationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'accountingTz': json['accounting_tz'],
        'apiKeys': json['api_keys'],
        'apiName': (0, TaxIntegrationApiName_1.TaxIntegrationApiNameFromJSON)(json['api_name']),
    };
}
function CreateTaxIntegrationRequestToJSON(json) {
    return CreateTaxIntegrationRequestToJSONTyped(json, false);
}
function CreateTaxIntegrationRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'accounting_tz': value['accountingTz'],
        'api_keys': value['apiKeys'],
        'api_name': (0, TaxIntegrationApiName_1.TaxIntegrationApiNameToJSON)(value['apiName']),
    };
}
