// @getopenpay/client/models/CreateInvoiceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateInvoiceRequest = instanceOfCreateInvoiceRequest;
exports.CreateInvoiceRequestFromJSON = CreateInvoiceRequestFromJSON;
exports.CreateInvoiceRequestFromJSONTyped = CreateInvoiceRequestFromJSONTyped;
exports.CreateInvoiceRequestToJSON = CreateInvoiceRequestToJSON;
exports.CreateInvoiceRequestToJSONTyped = CreateInvoiceRequestToJSONTyped;
const CollectionMethodEnum_1 = require("./CollectionMethodEnum");
const SelectedPriceQuantity_1 = require("./SelectedPriceQuantity");
const InvoiceType_1 = require("./InvoiceType");
/**
 * Check if a given object implements the CreateInvoiceRequest interface.
 */
function instanceOfCreateInvoiceRequest(value) {
    if (!('customerId' in value) || value['customerId'] === undefined)
        return false;
    return true;
}
function CreateInvoiceRequestFromJSON(json) {
    return CreateInvoiceRequestFromJSONTyped(json, false);
}
function CreateInvoiceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'collectionMethod': json['collection_method'] == null ? undefined : (0, CollectionMethodEnum_1.CollectionMethodEnumFromJSON)(json['collection_method']),
        'couponId': json['coupon_id'] == null ? undefined : json['coupon_id'],
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'customerId': json['customer_id'],
        'description': json['description'] == null ? undefined : json['description'],
        'emailInvoiceOnFinalization': json['email_invoice_on_finalization'] == null ? undefined : json['email_invoice_on_finalization'],
        'finalizeInvoiceImmediately': json['finalize_invoice_immediately'] == null ? undefined : json['finalize_invoice_immediately'],
        'invoiceType': json['invoice_type'] == null ? undefined : (0, InvoiceType_1.InvoiceTypeFromJSON)(json['invoice_type']),
        'isPreview': json['is_preview'] == null ? undefined : json['is_preview'],
        'netD': json['net_d'] == null ? undefined : json['net_d'],
        'paymentMethodId': json['payment_method_id'] == null ? undefined : json['payment_method_id'],
        'selectedProductPriceQuantity': json['selected_product_price_quantity'] == null ? undefined : (json['selected_product_price_quantity'].map(SelectedPriceQuantity_1.SelectedPriceQuantityFromJSON)),
        'subscriptionId': json['subscription_id'] == null ? undefined : json['subscription_id'],
    };
}
function CreateInvoiceRequestToJSON(json) {
    return CreateInvoiceRequestToJSONTyped(json, false);
}
function CreateInvoiceRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'collection_method': (0, CollectionMethodEnum_1.CollectionMethodEnumToJSON)(value['collectionMethod']),
        'coupon_id': value['couponId'],
        'custom_fields': value['customFields'],
        'customer_id': value['customerId'],
        'description': value['description'],
        'email_invoice_on_finalization': value['emailInvoiceOnFinalization'],
        'finalize_invoice_immediately': value['finalizeInvoiceImmediately'],
        'invoice_type': (0, InvoiceType_1.InvoiceTypeToJSON)(value['invoiceType']),
        'is_preview': value['isPreview'],
        'net_d': value['netD'],
        'payment_method_id': value['paymentMethodId'],
        'selected_product_price_quantity': value['selectedProductPriceQuantity'] == null ? undefined : (value['selectedProductPriceQuantity'].map(SelectedPriceQuantity_1.SelectedPriceQuantityToJSON)),
        'subscription_id': value['subscriptionId'],
    };
}
