// @getopenpay/client/models/CreateSubscriptionResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateSubscriptionResponse = instanceOfCreateSubscriptionResponse;
exports.CreateSubscriptionResponseFromJSON = CreateSubscriptionResponseFromJSON;
exports.CreateSubscriptionResponseFromJSONTyped = CreateSubscriptionResponseFromJSONTyped;
exports.CreateSubscriptionResponseToJSON = CreateSubscriptionResponseToJSON;
exports.CreateSubscriptionResponseToJSONTyped = CreateSubscriptionResponseToJSONTyped;
const InvoiceExternal_1 = require("./InvoiceExternal");
const SubscriptionExternal_1 = require("./SubscriptionExternal");
/**
 * Check if a given object implements the CreateSubscriptionResponse interface.
 */
function instanceOfCreateSubscriptionResponse(value) {
    if (!('created' in value) || value['created'] === undefined)
        return false;
    if (!('invoices' in value) || value['invoices'] === undefined)
        return false;
    if (!('processorsUsed' in value) || value['processorsUsed'] === undefined)
        return false;
    return true;
}
function CreateSubscriptionResponseFromJSON(json) {
    return CreateSubscriptionResponseFromJSONTyped(json, false);
}
function CreateSubscriptionResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'created': (json['created'].map(SubscriptionExternal_1.SubscriptionExternalFromJSON)),
        'invoices': (json['invoices'].map(InvoiceExternal_1.InvoiceExternalFromJSON)),
        'processorsUsed': json['processors_used'],
    };
}
function CreateSubscriptionResponseToJSON(json) {
    return CreateSubscriptionResponseToJSONTyped(json, false);
}
function CreateSubscriptionResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created': (value['created'].map(SubscriptionExternal_1.SubscriptionExternalToJSON)),
        'invoices': (value['invoices'].map(InvoiceExternal_1.InvoiceExternalToJSON)),
        'processors_used': value['processorsUsed'],
    };
}
