// @getopenpay/client/models/CreateSubscriptionItemRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateSubscriptionItemRequest = instanceOfCreateSubscriptionItemRequest;
exports.CreateSubscriptionItemRequestFromJSON = CreateSubscriptionItemRequestFromJSON;
exports.CreateSubscriptionItemRequestFromJSONTyped = CreateSubscriptionItemRequestFromJSONTyped;
exports.CreateSubscriptionItemRequestToJSON = CreateSubscriptionItemRequestToJSON;
exports.CreateSubscriptionItemRequestToJSONTyped = CreateSubscriptionItemRequestToJSONTyped;
const ProrationEnum_1 = require("./ProrationEnum");
/**
 * Check if a given object implements the CreateSubscriptionItemRequest interface.
 */
function instanceOfCreateSubscriptionItemRequest(value) {
    if (!('priceId' in value) || value['priceId'] === undefined)
        return false;
    if (!('subscriptionId' in value) || value['subscriptionId'] === undefined)
        return false;
    return true;
}
function CreateSubscriptionItemRequestFromJSON(json) {
    return CreateSubscriptionItemRequestFromJSONTyped(json, false);
}
function CreateSubscriptionItemRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'addAtPeriodEnd': json['add_at_period_end'] == null ? undefined : json['add_at_period_end'],
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'description': json['description'] == null ? undefined : json['description'],
        'priceId': json['price_id'],
        'prorationBehavior': json['proration_behavior'] == null ? undefined : (0, ProrationEnum_1.ProrationEnumFromJSON)(json['proration_behavior']),
        'quantity': json['quantity'] == null ? undefined : json['quantity'],
        'subscriptionId': json['subscription_id'],
    };
}
function CreateSubscriptionItemRequestToJSON(json) {
    return CreateSubscriptionItemRequestToJSONTyped(json, false);
}
function CreateSubscriptionItemRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'add_at_period_end': value['addAtPeriodEnd'],
        'custom_fields': value['customFields'],
        'description': value['description'],
        'price_id': value['priceId'],
        'proration_behavior': (0, ProrationEnum_1.ProrationEnumToJSON)(value['prorationBehavior']),
        'quantity': value['quantity'],
        'subscription_id': value['subscriptionId'],
    };
}
