// @getopenpay/client/models/CreateBillingMeterEventRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateBillingMeterEventRequest = instanceOfCreateBillingMeterEventRequest;
exports.CreateBillingMeterEventRequestFromJSON = CreateBillingMeterEventRequestFromJSON;
exports.CreateBillingMeterEventRequestFromJSONTyped = CreateBillingMeterEventRequestFromJSONTyped;
exports.CreateBillingMeterEventRequestToJSON = CreateBillingMeterEventRequestToJSON;
exports.CreateBillingMeterEventRequestToJSONTyped = CreateBillingMeterEventRequestToJSONTyped;
/**
 * Check if a given object implements the CreateBillingMeterEventRequest interface.
 */
function instanceOfCreateBillingMeterEventRequest(value) {
    if (!('eventName' in value) || value['eventName'] === undefined)
        return false;
    if (!('payload' in value) || value['payload'] === undefined)
        return false;
    return true;
}
function CreateBillingMeterEventRequestFromJSON(json) {
    return CreateBillingMeterEventRequestFromJSONTyped(json, false);
}
function CreateBillingMeterEventRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'eventDatetime': json['event_datetime'] == null ? undefined : (new Date(json['event_datetime'])),
        'eventName': json['event_name'],
        'identifier': json['identifier'] == null ? undefined : json['identifier'],
        'payload': json['payload'],
    };
}
function CreateBillingMeterEventRequestToJSON(json) {
    return CreateBillingMeterEventRequestToJSONTyped(json, false);
}
function CreateBillingMeterEventRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'event_datetime': value['eventDatetime'] == null ? undefined : ((value['eventDatetime']).toISOString()),
        'event_name': value['eventName'],
        'identifier': value['identifier'],
        'payload': value['payload'],
    };
}
