// @getopenpay/client/models/CreateBillingMeterEventAdjustmentRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateBillingMeterEventAdjustmentRequest = instanceOfCreateBillingMeterEventAdjustmentRequest;
exports.CreateBillingMeterEventAdjustmentRequestFromJSON = CreateBillingMeterEventAdjustmentRequestFromJSON;
exports.CreateBillingMeterEventAdjustmentRequestFromJSONTyped = CreateBillingMeterEventAdjustmentRequestFromJSONTyped;
exports.CreateBillingMeterEventAdjustmentRequestToJSON = CreateBillingMeterEventAdjustmentRequestToJSON;
exports.CreateBillingMeterEventAdjustmentRequestToJSONTyped = CreateBillingMeterEventAdjustmentRequestToJSONTyped;
const BillingMeterAdjustmentType_1 = require("./BillingMeterAdjustmentType");
/**
 * Check if a given object implements the CreateBillingMeterEventAdjustmentRequest interface.
 */
function instanceOfCreateBillingMeterEventAdjustmentRequest(value) {
    if (!('cancelIdentifier' in value) || value['cancelIdentifier'] === undefined)
        return false;
    if (!('eventName' in value) || value['eventName'] === undefined)
        return false;
    return true;
}
function CreateBillingMeterEventAdjustmentRequestFromJSON(json) {
    return CreateBillingMeterEventAdjustmentRequestFromJSONTyped(json, false);
}
function CreateBillingMeterEventAdjustmentRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'cancelIdentifier': json['cancel_identifier'],
        'eventName': json['event_name'],
        'type': json['type'] == null ? undefined : (0, BillingMeterAdjustmentType_1.BillingMeterAdjustmentTypeFromJSON)(json['type']),
    };
}
function CreateBillingMeterEventAdjustmentRequestToJSON(json) {
    return CreateBillingMeterEventAdjustmentRequestToJSONTyped(json, false);
}
function CreateBillingMeterEventAdjustmentRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'cancel_identifier': value['cancelIdentifier'],
        'event_name': value['eventName'],
        'type': (0, BillingMeterAdjustmentType_1.BillingMeterAdjustmentTypeToJSON)(value['type']),
    };
}
