// @getopenpay/client/models/CreditNoteReason.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreditNoteReason = void 0;
exports.instanceOfCreditNoteReason = instanceOfCreditNoteReason;
exports.CreditNoteReasonFromJSON = CreditNoteReasonFromJSON;
exports.CreditNoteReasonFromJSONTyped = CreditNoteReasonFromJSONTyped;
exports.CreditNoteReasonToJSON = CreditNoteReasonToJSON;
exports.CreditNoteReasonToJSONTyped = CreditNoteReasonToJSONTyped;
/**
 *
 * @export
 */
exports.CreditNoteReason = {
    Duplicate: 'duplicate',
    Fraudulent: 'fraudulent',
    OrderChange: 'order_change',
    ProductUnsatisfactory: 'product_unsatisfactory'
};
function instanceOfCreditNoteReason(value) {
    for (const key in exports.CreditNoteReason) {
        if (Object.prototype.hasOwnProperty.call(exports.CreditNoteReason, key)) {
            if (exports.CreditNoteReason[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CreditNoteReasonFromJSON(json) {
    return CreditNoteReasonFromJSONTyped(json, false);
}
function CreditNoteReasonFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CreditNoteReasonToJSON(value) {
    return value;
}
function CreditNoteReasonToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
