// @getopenpay/client/models/CurrencyFilterType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CurrencyFilterType = void 0;
exports.instanceOfCurrencyFilterType = instanceOfCurrencyFilterType;
exports.CurrencyFilterTypeFromJSON = CurrencyFilterTypeFromJSON;
exports.CurrencyFilterTypeFromJSONTyped = CurrencyFilterTypeFromJSONTyped;
exports.CurrencyFilterTypeToJSON = CurrencyFilterTypeToJSON;
exports.CurrencyFilterTypeToJSONTyped = CurrencyFilterTypeToJSONTyped;
/**
 *
 * @export
 */
exports.CurrencyFilterType = {
    Include: 'include',
    Exclude: 'exclude'
};
function instanceOfCurrencyFilterType(value) {
    for (const key in exports.CurrencyFilterType) {
        if (Object.prototype.hasOwnProperty.call(exports.CurrencyFilterType, key)) {
            if (exports.CurrencyFilterType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CurrencyFilterTypeFromJSON(json) {
    return CurrencyFilterTypeFromJSONTyped(json, false);
}
function CurrencyFilterTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CurrencyFilterTypeToJSON(value) {
    return value;
}
function CurrencyFilterTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
