// @getopenpay/client/models/CustomerStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerStatus = void 0;
exports.instanceOfCustomerStatus = instanceOfCustomerStatus;
exports.CustomerStatusFromJSON = CustomerStatusFromJSON;
exports.CustomerStatusFromJSONTyped = CustomerStatusFromJSONTyped;
exports.CustomerStatusToJSON = CustomerStatusToJSON;
exports.CustomerStatusToJSONTyped = CustomerStatusToJSONTyped;
/**
 *
 * @export
 */
exports.CustomerStatus = {
    Active: 'active',
    Inactive: 'inactive',
    Trialing: 'trialing',
    Cancelled: 'cancelled'
};
function instanceOfCustomerStatus(value) {
    for (const key in exports.CustomerStatus) {
        if (Object.prototype.hasOwnProperty.call(exports.CustomerStatus, key)) {
            if (exports.CustomerStatus[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CustomerStatusFromJSON(json) {
    return CustomerStatusFromJSONTyped(json, false);
}
function CustomerStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CustomerStatusToJSON(value) {
    return value;
}
function CustomerStatusToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
