// @getopenpay/client/models/CustomerBalanceTransactionExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCustomerBalanceTransactionExternal = instanceOfCustomerBalanceTransactionExternal;
exports.CustomerBalanceTransactionExternalFromJSON = CustomerBalanceTransactionExternalFromJSON;
exports.CustomerBalanceTransactionExternalFromJSONTyped = CustomerBalanceTransactionExternalFromJSONTyped;
exports.CustomerBalanceTransactionExternalToJSON = CustomerBalanceTransactionExternalToJSON;
exports.CustomerBalanceTransactionExternalToJSONTyped = CustomerBalanceTransactionExternalToJSONTyped;
const CustomerBalanceTransactionType_1 = require("./CustomerBalanceTransactionType");
const CurrencyEnum_1 = require("./CurrencyEnum");
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the CustomerBalanceTransactionExternal interface.
 */
function instanceOfCustomerBalanceTransactionExternal(value) {
    if (!('amountAtom' in value) || value['amountAtom'] === undefined)
        return false;
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('creditNoteId' in value) || value['creditNoteId'] === undefined)
        return false;
    if (!('currency' in value) || value['currency'] === undefined)
        return false;
    if (!('description' in value) || value['description'] === undefined)
        return false;
    if (!('endingBalanceAmountAtom' in value) || value['endingBalanceAmountAtom'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('invoiceId' in value) || value['invoiceId'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function CustomerBalanceTransactionExternalFromJSON(json) {
    return CustomerBalanceTransactionExternalFromJSONTyped(json, false);
}
function CustomerBalanceTransactionExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amountAtom': json['amount_atom'],
        'createdAt': (new Date(json['created_at'])),
        'creditNoteId': json['credit_note_id'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumFromJSON)(json['currency']),
        'description': json['description'],
        'endingBalanceAmountAtom': json['ending_balance_amount_atom'],
        'id': json['id'],
        'invoiceId': json['invoice_id'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'type': (0, CustomerBalanceTransactionType_1.CustomerBalanceTransactionTypeFromJSON)(json['type']),
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function CustomerBalanceTransactionExternalToJSON(json) {
    return CustomerBalanceTransactionExternalToJSONTyped(json, false);
}
function CustomerBalanceTransactionExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'amount_atom': value['amountAtom'],
        'created_at': ((value['createdAt']).toISOString()),
        'credit_note_id': value['creditNoteId'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumToJSON)(value['currency']),
        'description': value['description'],
        'ending_balance_amount_atom': value['endingBalanceAmountAtom'],
        'id': value['id'],
        'invoice_id': value['invoiceId'],
        'is_deleted': value['isDeleted'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'type': (0, CustomerBalanceTransactionType_1.CustomerBalanceTransactionTypeToJSON)(value['type']),
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
