// @getopenpay/client/models/CustomerBalanceTransactionType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerBalanceTransactionType = void 0;
exports.instanceOfCustomerBalanceTransactionType = instanceOfCustomerBalanceTransactionType;
exports.CustomerBalanceTransactionTypeFromJSON = CustomerBalanceTransactionTypeFromJSON;
exports.CustomerBalanceTransactionTypeFromJSONTyped = CustomerBalanceTransactionTypeFromJSONTyped;
exports.CustomerBalanceTransactionTypeToJSON = CustomerBalanceTransactionTypeToJSON;
exports.CustomerBalanceTransactionTypeToJSONTyped = CustomerBalanceTransactionTypeToJSONTyped;
/**
 * refer https://stripe.com/docs/billing/customer/balance#types
 * @export
 */
exports.CustomerBalanceTransactionType = {
    Adjustment: 'adjustment',
    AppliedToInvoice: 'applied_to_invoice',
    CreditNote: 'credit_note',
    Initial: 'initial',
    InvoiceOverpaid: 'invoice_overpaid',
    InvoiceTooLarge: 'invoice_too_large',
    InvoiceTooSmall: 'invoice_too_small',
    UnspentReceiverCredit: 'unspent_receiver_credit',
    UnappliedFromInvoice: 'unapplied_from_invoice'
};
function instanceOfCustomerBalanceTransactionType(value) {
    for (const key in exports.CustomerBalanceTransactionType) {
        if (Object.prototype.hasOwnProperty.call(exports.CustomerBalanceTransactionType, key)) {
            if (exports.CustomerBalanceTransactionType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CustomerBalanceTransactionTypeFromJSON(json) {
    return CustomerBalanceTransactionTypeFromJSONTyped(json, false);
}
function CustomerBalanceTransactionTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CustomerBalanceTransactionTypeToJSON(value) {
    return value;
}
function CustomerBalanceTransactionTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
