// @getopenpay/client/models/CustomerQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCustomerQueryParams = instanceOfCustomerQueryParams;
exports.CustomerQueryParamsFromJSON = CustomerQueryParamsFromJSON;
exports.CustomerQueryParamsFromJSONTyped = CustomerQueryParamsFromJSONTyped;
exports.CustomerQueryParamsToJSON = CustomerQueryParamsToJSON;
exports.CustomerQueryParamsToJSONTyped = CustomerQueryParamsToJSONTyped;
const DateTimeFilter_1 = require("./DateTimeFilter");
const SearchFilter_1 = require("./SearchFilter");
/**
 * Check if a given object implements the CustomerQueryParams interface.
 */
function instanceOfCustomerQueryParams(value) {
    return true;
}
function CustomerQueryParamsFromJSON(json) {
    return CustomerQueryParamsFromJSONTyped(json, false);
}
function CustomerQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'couponId': json['coupon_id'] == null ? undefined : json['coupon_id'],
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'email': json['email'] == null ? undefined : json['email'],
        'emails': json['emails'] == null ? undefined : json['emails'],
        'expand': json['expand'] == null ? undefined : json['expand'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'search': json['search'] == null ? undefined : (0, SearchFilter_1.SearchFilterFromJSON)(json['search']),
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function CustomerQueryParamsToJSON(json) {
    return CustomerQueryParamsToJSONTyped(json, false);
}
function CustomerQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'coupon_id': value['couponId'],
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'email': value['email'],
        'emails': value['emails'],
        'expand': value['expand'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'search': (0, SearchFilter_1.SearchFilterToJSON)(value['search']),
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
