// @getopenpay/client/models/DateTimeFilter.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDateTimeFilter = instanceOfDateTimeFilter;
exports.DateTimeFilterFromJSON = DateTimeFilterFromJSON;
exports.DateTimeFilterFromJSONTyped = DateTimeFilterFromJSONTyped;
exports.DateTimeFilterToJSON = DateTimeFilterToJSON;
exports.DateTimeFilterToJSONTyped = DateTimeFilterToJSONTyped;
/**
 * Check if a given object implements the DateTimeFilter interface.
 */
function instanceOfDateTimeFilter(value) {
    return true;
}
function DateTimeFilterFromJSON(json) {
    return DateTimeFilterFromJSONTyped(json, false);
}
function DateTimeFilterFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'eq': json['eq'] == null ? undefined : (new Date(json['eq'])),
        'gt': json['gt'] == null ? undefined : (new Date(json['gt'])),
        'gte': json['gte'] == null ? undefined : (new Date(json['gte'])),
        'lt': json['lt'] == null ? undefined : (new Date(json['lt'])),
        'lte': json['lte'] == null ? undefined : (new Date(json['lte'])),
    };
}
function DateTimeFilterToJSON(json) {
    return DateTimeFilterToJSONTyped(json, false);
}
function DateTimeFilterToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'eq': value['eq'] == null ? undefined : (value['eq'].toISOString()),
        'gt': value['gt'] == null ? undefined : (value['gt'].toISOString()),
        'gte': value['gte'] == null ? undefined : (value['gte'].toISOString()),
        'lt': value['lt'] == null ? undefined : (value['lt'].toISOString()),
        'lte': value['lte'] == null ? undefined : (value['lte'].toISOString()),
    };
}
