// @getopenpay/client/models/DeleteSubscriptionItemRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteSubscriptionItemRequest = instanceOfDeleteSubscriptionItemRequest;
exports.DeleteSubscriptionItemRequestFromJSON = DeleteSubscriptionItemRequestFromJSON;
exports.DeleteSubscriptionItemRequestFromJSONTyped = DeleteSubscriptionItemRequestFromJSONTyped;
exports.DeleteSubscriptionItemRequestToJSON = DeleteSubscriptionItemRequestToJSON;
exports.DeleteSubscriptionItemRequestToJSONTyped = DeleteSubscriptionItemRequestToJSONTyped;
const ProrationEnum_1 = require("./ProrationEnum");
/**
 * Check if a given object implements the DeleteSubscriptionItemRequest interface.
 */
function instanceOfDeleteSubscriptionItemRequest(value) {
    return true;
}
function DeleteSubscriptionItemRequestFromJSON(json) {
    return DeleteSubscriptionItemRequestFromJSONTyped(json, false);
}
function DeleteSubscriptionItemRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'dropAtEnd': json['drop_at_end'] == null ? undefined : json['drop_at_end'],
        'prorationBehavior': json['proration_behavior'] == null ? undefined : (0, ProrationEnum_1.ProrationEnumFromJSON)(json['proration_behavior']),
    };
}
function DeleteSubscriptionItemRequestToJSON(json) {
    return DeleteSubscriptionItemRequestToJSONTyped(json, false);
}
function DeleteSubscriptionItemRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'drop_at_end': value['dropAtEnd'],
        'proration_behavior': (0, ProrationEnum_1.ProrationEnumToJSON)(value['prorationBehavior']),
    };
}
