// @getopenpay/client/models/DeleteSubscriptionItemResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteSubscriptionItemResponse = instanceOfDeleteSubscriptionItemResponse;
exports.DeleteSubscriptionItemResponseFromJSON = DeleteSubscriptionItemResponseFromJSON;
exports.DeleteSubscriptionItemResponseFromJSONTyped = DeleteSubscriptionItemResponseFromJSONTyped;
exports.DeleteSubscriptionItemResponseToJSON = DeleteSubscriptionItemResponseToJSON;
exports.DeleteSubscriptionItemResponseToJSONTyped = DeleteSubscriptionItemResponseToJSONTyped;
/**
 * Check if a given object implements the DeleteSubscriptionItemResponse interface.
 */
function instanceOfDeleteSubscriptionItemResponse(value) {
    if (!('deletedAt' in value) || value['deletedAt'] === undefined)
        return false;
    if (!('subscriptionItemId' in value) || value['subscriptionItemId'] === undefined)
        return false;
    return true;
}
function DeleteSubscriptionItemResponseFromJSON(json) {
    return DeleteSubscriptionItemResponseFromJSONTyped(json, false);
}
function DeleteSubscriptionItemResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'deletedAt': (json['deleted_at'] == null ? null : new Date(json['deleted_at'])),
        'dropAtEnd': json['drop_at_end'] == null ? undefined : json['drop_at_end'],
        'message': json['message'] == null ? undefined : json['message'],
        'subscriptionItemId': json['subscription_item_id'],
    };
}
function DeleteSubscriptionItemResponseToJSON(json) {
    return DeleteSubscriptionItemResponseToJSONTyped(json, false);
}
function DeleteSubscriptionItemResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'deleted_at': (value['deletedAt'] == null ? null : value['deletedAt'].toISOString()),
        'drop_at_end': value['dropAtEnd'],
        'message': value['message'],
        'subscription_item_id': value['subscriptionItemId'],
    };
}
