// @getopenpay/client/models/DeleteSubscriptionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteSubscriptionRequest = instanceOfDeleteSubscriptionRequest;
exports.DeleteSubscriptionRequestFromJSON = DeleteSubscriptionRequestFromJSON;
exports.DeleteSubscriptionRequestFromJSONTyped = DeleteSubscriptionRequestFromJSONTyped;
exports.DeleteSubscriptionRequestToJSON = DeleteSubscriptionRequestToJSON;
exports.DeleteSubscriptionRequestToJSONTyped = DeleteSubscriptionRequestToJSONTyped;
const SubscriptionCancellationDetails_1 = require("./SubscriptionCancellationDetails");
/**
 * Check if a given object implements the DeleteSubscriptionRequest interface.
 */
function instanceOfDeleteSubscriptionRequest(value) {
    return true;
}
function DeleteSubscriptionRequestFromJSON(json) {
    return DeleteSubscriptionRequestFromJSONTyped(json, false);
}
function DeleteSubscriptionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'cancelUnpaidInvoices': json['cancel_unpaid_invoices'] == null ? undefined : json['cancel_unpaid_invoices'],
        'cancellationDetails': json['cancellation_details'] == null ? undefined : (0, SubscriptionCancellationDetails_1.SubscriptionCancellationDetailsFromJSON)(json['cancellation_details']),
        'fullRefund': json['full_refund'] == null ? undefined : json['full_refund'],
        'isPreview': json['is_preview'] == null ? undefined : json['is_preview'],
        'prorate': json['prorate'] == null ? undefined : json['prorate'],
    };
}
function DeleteSubscriptionRequestToJSON(json) {
    return DeleteSubscriptionRequestToJSONTyped(json, false);
}
function DeleteSubscriptionRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'cancel_unpaid_invoices': value['cancelUnpaidInvoices'],
        'cancellation_details': (0, SubscriptionCancellationDetails_1.SubscriptionCancellationDetailsToJSON)(value['cancellationDetails']),
        'full_refund': value['fullRefund'],
        'is_preview': value['isPreview'],
        'prorate': value['prorate'],
    };
}
