// @getopenpay/client/models/DeleteInvoiceItemResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteInvoiceItemResponse = instanceOfDeleteInvoiceItemResponse;
exports.DeleteInvoiceItemResponseFromJSON = DeleteInvoiceItemResponseFromJSON;
exports.DeleteInvoiceItemResponseFromJSONTyped = DeleteInvoiceItemResponseFromJSONTyped;
exports.DeleteInvoiceItemResponseToJSON = DeleteInvoiceItemResponseToJSON;
exports.DeleteInvoiceItemResponseToJSONTyped = DeleteInvoiceItemResponseToJSONTyped;
/**
 * Check if a given object implements the DeleteInvoiceItemResponse interface.
 */
function instanceOfDeleteInvoiceItemResponse(value) {
    if (!('invoiceItemId' in value) || value['invoiceItemId'] === undefined)
        return false;
    return true;
}
function DeleteInvoiceItemResponseFromJSON(json) {
    return DeleteInvoiceItemResponseFromJSONTyped(json, false);
}
function DeleteInvoiceItemResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'invoiceItemId': json['invoice_item_id'],
        'message': json['message'] == null ? undefined : json['message'],
    };
}
function DeleteInvoiceItemResponseToJSON(json) {
    return DeleteInvoiceItemResponseToJSONTyped(json, false);
}
function DeleteInvoiceItemResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'invoice_item_id': value['invoiceItemId'],
        'message': value['message'],
    };
}
