// @getopenpay/client/models/EventSearchParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfEventSearchParams = instanceOfEventSearchParams;
exports.EventSearchParamsFromJSON = EventSearchParamsFromJSON;
exports.EventSearchParamsFromJSONTyped = EventSearchParamsFromJSONTyped;
exports.EventSearchParamsToJSON = EventSearchParamsToJSON;
exports.EventSearchParamsToJSONTyped = EventSearchParamsToJSONTyped;
const DateTimeFilter_1 = require("./DateTimeFilter");
const EventType_1 = require("./EventType");
/**
 * Check if a given object implements the EventSearchParams interface.
 */
function instanceOfEventSearchParams(value) {
    if (!('objectId' in value) || value['objectId'] === undefined)
        return false;
    return true;
}
function EventSearchParamsFromJSON(json) {
    return EventSearchParamsFromJSONTyped(json, false);
}
function EventSearchParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'expand': json['expand'] == null ? undefined : json['expand'],
        'limit': json['limit'] == null ? undefined : json['limit'],
        'objectId': json['object_id'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'types': json['types'] == null ? undefined : (json['types'].map(EventType_1.EventTypeFromJSON)),
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function EventSearchParamsToJSON(json) {
    return EventSearchParamsToJSONTyped(json, false);
}
function EventSearchParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'expand': value['expand'],
        'limit': value['limit'],
        'object_id': value['objectId'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'types': value['types'] == null ? undefined : (value['types'].map(EventType_1.EventTypeToJSON)),
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
