// @getopenpay/client/models/EventRetriggerRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfEventRetriggerRequest = instanceOfEventRetriggerRequest;
exports.EventRetriggerRequestFromJSON = EventRetriggerRequestFromJSON;
exports.EventRetriggerRequestFromJSONTyped = EventRetriggerRequestFromJSONTyped;
exports.EventRetriggerRequestToJSON = EventRetriggerRequestToJSON;
exports.EventRetriggerRequestToJSONTyped = EventRetriggerRequestToJSONTyped;
/**
 * Check if a given object implements the EventRetriggerRequest interface.
 */
function instanceOfEventRetriggerRequest(value) {
    if (!('eventId' in value) || value['eventId'] === undefined)
        return false;
    return true;
}
function EventRetriggerRequestFromJSON(json) {
    return EventRetriggerRequestFromJSONTyped(json, false);
}
function EventRetriggerRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'eventId': json['event_id'],
        'selectedWebhooks': json['selected_webhooks'] == null ? undefined : json['selected_webhooks'],
    };
}
function EventRetriggerRequestToJSON(json) {
    return EventRetriggerRequestToJSONTyped(json, false);
}
function EventRetriggerRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'event_id': value['eventId'],
        'selected_webhooks': value['selectedWebhooks'],
    };
}
