// @getopenpay/client/models/InvoiceItemDiscountAmountsPublic.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfInvoiceItemDiscountAmountsPublic = instanceOfInvoiceItemDiscountAmountsPublic;
exports.InvoiceItemDiscountAmountsPublicFromJSON = InvoiceItemDiscountAmountsPublicFromJSON;
exports.InvoiceItemDiscountAmountsPublicFromJSONTyped = InvoiceItemDiscountAmountsPublicFromJSONTyped;
exports.InvoiceItemDiscountAmountsPublicToJSON = InvoiceItemDiscountAmountsPublicToJSON;
exports.InvoiceItemDiscountAmountsPublicToJSONTyped = InvoiceItemDiscountAmountsPublicToJSONTyped;
/**
 * Check if a given object implements the InvoiceItemDiscountAmountsPublic interface.
 */
function instanceOfInvoiceItemDiscountAmountsPublic(value) {
    if (!('amountAtom' in value) || value['amountAtom'] === undefined)
        return false;
    if (!('couponName' in value) || value['couponName'] === undefined)
        return false;
    return true;
}
function InvoiceItemDiscountAmountsPublicFromJSON(json) {
    return InvoiceItemDiscountAmountsPublicFromJSONTyped(json, false);
}
function InvoiceItemDiscountAmountsPublicFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amountAtom': json['amount_atom'],
        'couponDescription': json['coupon_description'] == null ? undefined : json['coupon_description'],
        'couponName': json['coupon_name'],
    };
}
function InvoiceItemDiscountAmountsPublicToJSON(json) {
    return InvoiceItemDiscountAmountsPublicToJSONTyped(json, false);
}
function InvoiceItemDiscountAmountsPublicToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'amount_atom': value['amountAtom'],
        'coupon_description': value['couponDescription'],
        'coupon_name': value['couponName'],
    };
}
