// @getopenpay/client/models/InvoiceSettings.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfInvoiceSettings = instanceOfInvoiceSettings;
exports.InvoiceSettingsFromJSON = InvoiceSettingsFromJSON;
exports.InvoiceSettingsFromJSONTyped = InvoiceSettingsFromJSONTyped;
exports.InvoiceSettingsToJSON = InvoiceSettingsToJSON;
exports.InvoiceSettingsToJSONTyped = InvoiceSettingsToJSONTyped;
/**
 * Check if a given object implements the InvoiceSettings interface.
 */
function instanceOfInvoiceSettings(value) {
    return true;
}
function InvoiceSettingsFromJSON(json) {
    return InvoiceSettingsFromJSONTyped(json, false);
}
function InvoiceSettingsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'defaultNetD': json['default_net_d'] == null ? undefined : json['default_net_d'],
    };
}
function InvoiceSettingsToJSON(json) {
    return InvoiceSettingsToJSONTyped(json, false);
}
function InvoiceSettingsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'default_net_d': value['defaultNetD'],
    };
}
