// @getopenpay/client/models/InvoiceStatusEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceStatusEnum = void 0;
exports.instanceOfInvoiceStatusEnum = instanceOfInvoiceStatusEnum;
exports.InvoiceStatusEnumFromJSON = InvoiceStatusEnumFromJSON;
exports.InvoiceStatusEnumFromJSONTyped = InvoiceStatusEnumFromJSONTyped;
exports.InvoiceStatusEnumToJSON = InvoiceStatusEnumToJSON;
exports.InvoiceStatusEnumToJSONTyped = InvoiceStatusEnumToJSONTyped;
/**
 *
 * @export
 */
exports.InvoiceStatusEnum = {
    Draft: 'draft',
    Open: 'open',
    Void: 'void',
    Paid: 'paid',
    PastDue: 'past_due',
    Uncollectible: 'uncollectible'
};
function instanceOfInvoiceStatusEnum(value) {
    for (const key in exports.InvoiceStatusEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.InvoiceStatusEnum, key)) {
            if (exports.InvoiceStatusEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function InvoiceStatusEnumFromJSON(json) {
    return InvoiceStatusEnumFromJSONTyped(json, false);
}
function InvoiceStatusEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function InvoiceStatusEnumToJSON(value) {
    return value;
}
function InvoiceStatusEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
