// @getopenpay/client/models/InvoiceItemExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfInvoiceItemExternal = instanceOfInvoiceItemExternal;
exports.InvoiceItemExternalFromJSON = InvoiceItemExternalFromJSON;
exports.InvoiceItemExternalFromJSONTyped = InvoiceItemExternalFromJSONTyped;
exports.InvoiceItemExternalToJSON = InvoiceItemExternalToJSON;
exports.InvoiceItemExternalToJSONTyped = InvoiceItemExternalToJSONTyped;
const InvoiceItemDiscountAmountsExternal_1 = require("./InvoiceItemDiscountAmountsExternal");
const PriceTypeEnum_1 = require("./PriceTypeEnum");
const CurrencyEnum_1 = require("./CurrencyEnum");
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the InvoiceItemExternal interface.
 */
function instanceOfInvoiceItemExternal(value) {
    if (!('accountId' in value) || value['accountId'] === undefined)
        return false;
    if (!('amountAtom' in value) || value['amountAtom'] === undefined)
        return false;
    if (!('amountAtomConsideringDiscountApplied' in value) || value['amountAtomConsideringDiscountApplied'] === undefined)
        return false;
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('currency' in value) || value['currency'] === undefined)
        return false;
    if (!('customerId' in value) || value['customerId'] === undefined)
        return false;
    if (!('description' in value) || value['description'] === undefined)
        return false;
    if (!('discountAmountAtoms' in value) || value['discountAmountAtoms'] === undefined)
        return false;
    if (!('discounts' in value) || value['discounts'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('invoiceId' in value) || value['invoiceId'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('periodEnd' in value) || value['periodEnd'] === undefined)
        return false;
    if (!('periodStart' in value) || value['periodStart'] === undefined)
        return false;
    if (!('priceId' in value) || value['priceId'] === undefined)
        return false;
    if (!('priceType' in value) || value['priceType'] === undefined)
        return false;
    if (!('productId' in value) || value['productId'] === undefined)
        return false;
    if (!('proration' in value) || value['proration'] === undefined)
        return false;
    if (!('prorationDetailsDebitInvoiceItem' in value) || value['prorationDetailsDebitInvoiceItem'] === undefined)
        return false;
    if (!('quantity' in value) || value['quantity'] === undefined)
        return false;
    if (!('subscriptionId' in value) || value['subscriptionId'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function InvoiceItemExternalFromJSON(json) {
    return InvoiceItemExternalFromJSONTyped(json, false);
}
function InvoiceItemExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'accountId': json['account_id'],
        'amountAtom': json['amount_atom'],
        'amountAtomConsideringDiscountApplied': json['amount_atom_considering_discount_applied'],
        'createdAt': (new Date(json['created_at'])),
        'currency': (0, CurrencyEnum_1.CurrencyEnumFromJSON)(json['currency']),
        'customerId': json['customer_id'],
        'description': json['description'],
        'discountAmountAtoms': (json['discount_amount_atoms'].map(InvoiceItemDiscountAmountsExternal_1.InvoiceItemDiscountAmountsExternalFromJSON)),
        'discounts': json['discounts'],
        'id': json['id'],
        'invoiceId': json['invoice_id'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'name': json['name'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'periodEnd': (new Date(json['period_end'])),
        'periodStart': (new Date(json['period_start'])),
        'priceId': json['price_id'],
        'priceType': (0, PriceTypeEnum_1.PriceTypeEnumFromJSON)(json['price_type']),
        'productId': json['product_id'],
        'proration': json['proration'],
        'prorationDetailsDebitInvoiceItem': json['proration_details_debit_invoice_item'],
        'quantity': json['quantity'],
        'subscriptionCancelledAt': json['subscription_cancelled_at'] == null ? undefined : (new Date(json['subscription_cancelled_at'])),
        'subscriptionId': json['subscription_id'],
        'subscriptionItemDescription': json['subscription_item_description'] == null ? undefined : json['subscription_item_description'],
        'subscriptionItemId': json['subscription_item_id'] == null ? undefined : json['subscription_item_id'],
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function InvoiceItemExternalToJSON(json) {
    return InvoiceItemExternalToJSONTyped(json, false);
}
function InvoiceItemExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'account_id': value['accountId'],
        'amount_atom': value['amountAtom'],
        'amount_atom_considering_discount_applied': value['amountAtomConsideringDiscountApplied'],
        'created_at': ((value['createdAt']).toISOString()),
        'currency': (0, CurrencyEnum_1.CurrencyEnumToJSON)(value['currency']),
        'customer_id': value['customerId'],
        'description': value['description'],
        'discount_amount_atoms': (value['discountAmountAtoms'].map(InvoiceItemDiscountAmountsExternal_1.InvoiceItemDiscountAmountsExternalToJSON)),
        'discounts': value['discounts'],
        'id': value['id'],
        'invoice_id': value['invoiceId'],
        'is_deleted': value['isDeleted'],
        'name': value['name'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'period_end': ((value['periodEnd']).toISOString()),
        'period_start': ((value['periodStart']).toISOString()),
        'price_id': value['priceId'],
        'price_type': (0, PriceTypeEnum_1.PriceTypeEnumToJSON)(value['priceType']),
        'product_id': value['productId'],
        'proration': value['proration'],
        'proration_details_debit_invoice_item': value['prorationDetailsDebitInvoiceItem'],
        'quantity': value['quantity'],
        'subscription_cancelled_at': value['subscriptionCancelledAt'] == null ? undefined : (value['subscriptionCancelledAt'].toISOString()),
        'subscription_id': value['subscriptionId'],
        'subscription_item_description': value['subscriptionItemDescription'],
        'subscription_item_id': value['subscriptionItemId'],
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
