// @getopenpay/client/models/ListResponseProductFamilyExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseProductFamilyExternal = instanceOfListResponseProductFamilyExternal;
exports.ListResponseProductFamilyExternalFromJSON = ListResponseProductFamilyExternalFromJSON;
exports.ListResponseProductFamilyExternalFromJSONTyped = ListResponseProductFamilyExternalFromJSONTyped;
exports.ListResponseProductFamilyExternalToJSON = ListResponseProductFamilyExternalToJSON;
exports.ListResponseProductFamilyExternalToJSONTyped = ListResponseProductFamilyExternalToJSONTyped;
const ProductFamilyExternal_1 = require("./ProductFamilyExternal");
/**
 * Check if a given object implements the ListResponseProductFamilyExternal interface.
 */
function instanceOfListResponseProductFamilyExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseProductFamilyExternalFromJSON(json) {
    return ListResponseProductFamilyExternalFromJSONTyped(json, false);
}
function ListResponseProductFamilyExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(ProductFamilyExternal_1.ProductFamilyExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseProductFamilyExternalToJSON(json) {
    return ListResponseProductFamilyExternalToJSONTyped(json, false);
}
function ListResponseProductFamilyExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(ProductFamilyExternal_1.ProductFamilyExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
