// @getopenpay/client/models/ListActiveSubParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListActiveSubParams = instanceOfListActiveSubParams;
exports.ListActiveSubParamsFromJSON = ListActiveSubParamsFromJSON;
exports.ListActiveSubParamsFromJSONTyped = ListActiveSubParamsFromJSONTyped;
exports.ListActiveSubParamsToJSON = ListActiveSubParamsToJSON;
exports.ListActiveSubParamsToJSONTyped = ListActiveSubParamsToJSONTyped;
/**
 * Check if a given object implements the ListActiveSubParams interface.
 */
function instanceOfListActiveSubParams(value) {
    return true;
}
function ListActiveSubParamsFromJSON(json) {
    return ListActiveSubParamsFromJSONTyped(json, false);
}
function ListActiveSubParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'priceId': json['price_id'] == null ? undefined : json['price_id'],
        'productId': json['product_id'] == null ? undefined : json['product_id'],
    };
}
function ListActiveSubParamsToJSON(json) {
    return ListActiveSubParamsToJSONTyped(json, false);
}
function ListActiveSubParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'price_id': value['priceId'],
        'product_id': value['productId'],
    };
}
