// @getopenpay/client/models/ListResponseChargeExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseChargeExternal = instanceOfListResponseChargeExternal;
exports.ListResponseChargeExternalFromJSON = ListResponseChargeExternalFromJSON;
exports.ListResponseChargeExternalFromJSONTyped = ListResponseChargeExternalFromJSONTyped;
exports.ListResponseChargeExternalToJSON = ListResponseChargeExternalToJSON;
exports.ListResponseChargeExternalToJSONTyped = ListResponseChargeExternalToJSONTyped;
const ChargeExternal_1 = require("./ChargeExternal");
/**
 * Check if a given object implements the ListResponseChargeExternal interface.
 */
function instanceOfListResponseChargeExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseChargeExternalFromJSON(json) {
    return ListResponseChargeExternalFromJSONTyped(json, false);
}
function ListResponseChargeExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(ChargeExternal_1.ChargeExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseChargeExternalToJSON(json) {
    return ListResponseChargeExternalToJSONTyped(json, false);
}
function ListResponseChargeExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(ChargeExternal_1.ChargeExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
