// @getopenpay/client/models/ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner = instanceOfListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSON = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSON;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSONTyped = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSONTyped;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSON = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSON;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSONTyped = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSONTyped;
const PaymentProviderType_1 = require("./PaymentProviderType");
const PaymentMethodMappingExternal_1 = require("./PaymentMethodMappingExternal");
const CompleteAddress_1 = require("./CompleteAddress");
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner interface.
 */
function instanceOfListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner(value) {
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('displayName' in value) || value['displayName'] === undefined)
        return false;
    if (!('expiryDate' in value) || value['expiryDate'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('isFullDetailsKnown' in value) || value['isFullDetailsKnown'] === undefined)
        return false;
    if (!('provider' in value) || value['provider'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSON(json) {
    return ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSONTyped(json, false);
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'billingAddress': json['billing_address'] == null ? undefined : (0, CompleteAddress_1.CompleteAddressFromJSON)(json['billing_address']),
        'cardBrand': json['card_brand'] == null ? undefined : json['card_brand'],
        'cardCountry': json['card_country'] == null ? undefined : json['card_country'],
        'cardFingerprint': json['card_fingerprint'] == null ? undefined : json['card_fingerprint'],
        'cardIin': json['card_iin'] == null ? undefined : json['card_iin'],
        'cardIssuer': json['card_issuer'] == null ? undefined : json['card_issuer'],
        'cardType': json['card_type'] == null ? undefined : json['card_type'],
        'cdeExternalId': json['cde_external_id'] == null ? undefined : json['cde_external_id'],
        'createdAt': (new Date(json['created_at'])),
        'customerId': json['customer_id'] == null ? undefined : json['customer_id'],
        'displayName': json['display_name'],
        'expiryDate': json['expiry_date'],
        'id': json['id'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'isFullDetailsKnown': json['is_full_details_known'],
        'lastFour': json['last_four'] == null ? undefined : json['last_four'],
        'mappings': json['mappings'] == null ? undefined : (json['mappings'].map(PaymentMethodMappingExternal_1.PaymentMethodMappingExternalFromJSON)),
        'metadata': json['metadata'] == null ? undefined : json['metadata'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'provider': (0, PaymentProviderType_1.PaymentProviderTypeFromJSON)(json['provider']),
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSON(json) {
    return ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSONTyped(json, false);
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'billing_address': (0, CompleteAddress_1.CompleteAddressToJSON)(value['billingAddress']),
        'card_brand': value['cardBrand'],
        'card_country': value['cardCountry'],
        'card_fingerprint': value['cardFingerprint'],
        'card_iin': value['cardIin'],
        'card_issuer': value['cardIssuer'],
        'card_type': value['cardType'],
        'cde_external_id': value['cdeExternalId'],
        'created_at': ((value['createdAt']).toISOString()),
        'customer_id': value['customerId'],
        'display_name': value['displayName'],
        'expiry_date': value['expiryDate'],
        'id': value['id'],
        'is_deleted': value['isDeleted'],
        'is_full_details_known': value['isFullDetailsKnown'],
        'last_four': value['lastFour'],
        'mappings': value['mappings'] == null ? undefined : (value['mappings'].map(PaymentMethodMappingExternal_1.PaymentMethodMappingExternalToJSON)),
        'metadata': value['metadata'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'provider': (0, PaymentProviderType_1.PaymentProviderTypeToJSON)(value['provider']),
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
