// @getopenpay/client/models/ListResponsePriceExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponsePriceExternal = instanceOfListResponsePriceExternal;
exports.ListResponsePriceExternalFromJSON = ListResponsePriceExternalFromJSON;
exports.ListResponsePriceExternalFromJSONTyped = ListResponsePriceExternalFromJSONTyped;
exports.ListResponsePriceExternalToJSON = ListResponsePriceExternalToJSON;
exports.ListResponsePriceExternalToJSONTyped = ListResponsePriceExternalToJSONTyped;
const PriceExternal_1 = require("./PriceExternal");
/**
 * Check if a given object implements the ListResponsePriceExternal interface.
 */
function instanceOfListResponsePriceExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponsePriceExternalFromJSON(json) {
    return ListResponsePriceExternalFromJSONTyped(json, false);
}
function ListResponsePriceExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(PriceExternal_1.PriceExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponsePriceExternalToJSON(json) {
    return ListResponsePriceExternalToJSONTyped(json, false);
}
function ListResponsePriceExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(PriceExternal_1.PriceExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
