// @getopenpay/client/models/ListResponseInvoiceItemExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseInvoiceItemExternal = instanceOfListResponseInvoiceItemExternal;
exports.ListResponseInvoiceItemExternalFromJSON = ListResponseInvoiceItemExternalFromJSON;
exports.ListResponseInvoiceItemExternalFromJSONTyped = ListResponseInvoiceItemExternalFromJSONTyped;
exports.ListResponseInvoiceItemExternalToJSON = ListResponseInvoiceItemExternalToJSON;
exports.ListResponseInvoiceItemExternalToJSONTyped = ListResponseInvoiceItemExternalToJSONTyped;
const InvoiceItemExternal_1 = require("./InvoiceItemExternal");
/**
 * Check if a given object implements the ListResponseInvoiceItemExternal interface.
 */
function instanceOfListResponseInvoiceItemExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseInvoiceItemExternalFromJSON(json) {
    return ListResponseInvoiceItemExternalFromJSONTyped(json, false);
}
function ListResponseInvoiceItemExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(InvoiceItemExternal_1.InvoiceItemExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseInvoiceItemExternalToJSON(json) {
    return ListResponseInvoiceItemExternalToJSONTyped(json, false);
}
function ListResponseInvoiceItemExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(InvoiceItemExternal_1.InvoiceItemExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
