// @getopenpay/client/models/ListResponsePaymentLinkExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponsePaymentLinkExternal = instanceOfListResponsePaymentLinkExternal;
exports.ListResponsePaymentLinkExternalFromJSON = ListResponsePaymentLinkExternalFromJSON;
exports.ListResponsePaymentLinkExternalFromJSONTyped = ListResponsePaymentLinkExternalFromJSONTyped;
exports.ListResponsePaymentLinkExternalToJSON = ListResponsePaymentLinkExternalToJSON;
exports.ListResponsePaymentLinkExternalToJSONTyped = ListResponsePaymentLinkExternalToJSONTyped;
const PaymentLinkExternal_1 = require("./PaymentLinkExternal");
/**
 * Check if a given object implements the ListResponsePaymentLinkExternal interface.
 */
function instanceOfListResponsePaymentLinkExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponsePaymentLinkExternalFromJSON(json) {
    return ListResponsePaymentLinkExternalFromJSONTyped(json, false);
}
function ListResponsePaymentLinkExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(PaymentLinkExternal_1.PaymentLinkExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponsePaymentLinkExternalToJSON(json) {
    return ListResponsePaymentLinkExternalToJSONTyped(json, false);
}
function ListResponsePaymentLinkExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(PaymentLinkExternal_1.PaymentLinkExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
