// @getopenpay/client/models/MerchantTaxIdSetting.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantTaxIdSetting = instanceOfMerchantTaxIdSetting;
exports.MerchantTaxIdSettingFromJSON = MerchantTaxIdSettingFromJSON;
exports.MerchantTaxIdSettingFromJSONTyped = MerchantTaxIdSettingFromJSONTyped;
exports.MerchantTaxIdSettingToJSON = MerchantTaxIdSettingToJSON;
exports.MerchantTaxIdSettingToJSONTyped = MerchantTaxIdSettingToJSONTyped;
/**
 * Check if a given object implements the MerchantTaxIdSetting interface.
 */
function instanceOfMerchantTaxIdSetting(value) {
    if (!('isDefault' in value) || value['isDefault'] === undefined)
        return false;
    if (!('taxId' in value) || value['taxId'] === undefined)
        return false;
    if (!('taxIdType' in value) || value['taxIdType'] === undefined)
        return false;
    return true;
}
function MerchantTaxIdSettingFromJSON(json) {
    return MerchantTaxIdSettingFromJSONTyped(json, false);
}
function MerchantTaxIdSettingFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'isDefault': json['is_default'],
        'taxId': json['tax_id'],
        'taxIdType': json['tax_id_type'],
    };
}
function MerchantTaxIdSettingToJSON(json) {
    return MerchantTaxIdSettingToJSONTyped(json, false);
}
function MerchantTaxIdSettingToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'is_default': value['isDefault'],
        'tax_id': value['taxId'],
        'tax_id_type': value['taxIdType'],
    };
}
