// @getopenpay/client/models/NonPciIntegrationEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NonPciIntegrationEnum = void 0;
exports.instanceOfNonPciIntegrationEnum = instanceOfNonPciIntegrationEnum;
exports.NonPciIntegrationEnumFromJSON = NonPciIntegrationEnumFromJSON;
exports.NonPciIntegrationEnumFromJSONTyped = NonPciIntegrationEnumFromJSONTyped;
exports.NonPciIntegrationEnumToJSON = NonPciIntegrationEnumToJSON;
exports.NonPciIntegrationEnumToJSONTyped = NonPciIntegrationEnumToJSONTyped;
/**
 *
 * @export
 */
exports.NonPciIntegrationEnum = {
    Chartmogul: 'chartmogul',
    Intercom: 'intercom'
};
function instanceOfNonPciIntegrationEnum(value) {
    for (const key in exports.NonPciIntegrationEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.NonPciIntegrationEnum, key)) {
            if (exports.NonPciIntegrationEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function NonPciIntegrationEnumFromJSON(json) {
    return NonPciIntegrationEnumFromJSONTyped(json, false);
}
function NonPciIntegrationEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function NonPciIntegrationEnumToJSON(value) {
    return value;
}
function NonPciIntegrationEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
