// @getopenpay/client/models/NumberConfigField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NumberConfigFieldTypeEnum = void 0;
exports.instanceOfNumberConfigField = instanceOfNumberConfigField;
exports.NumberConfigFieldFromJSON = NumberConfigFieldFromJSON;
exports.NumberConfigFieldFromJSONTyped = NumberConfigFieldFromJSONTyped;
exports.NumberConfigFieldToJSON = NumberConfigFieldToJSON;
exports.NumberConfigFieldToJSONTyped = NumberConfigFieldToJSONTyped;
/**
 * @export
 */
exports.NumberConfigFieldTypeEnum = {
    Number: 'number'
};
/**
 * Check if a given object implements the NumberConfigField interface.
 */
function instanceOfNumberConfigField(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('value' in value) || value['value'] === undefined)
        return false;
    return true;
}
function NumberConfigFieldFromJSON(json) {
    return NumberConfigFieldFromJSONTyped(json, false);
}
function NumberConfigFieldFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'defaultValue': json['default_value'] == null ? undefined : json['default_value'],
        'description': json['description'] == null ? undefined : json['description'],
        'key': json['key'],
        'maxValue': json['max_value'] == null ? undefined : json['max_value'],
        'minValue': json['min_value'] == null ? undefined : json['min_value'],
        'name': json['name'],
        'type': json['type'] == null ? undefined : json['type'],
        'value': json['value'],
    };
}
function NumberConfigFieldToJSON(json) {
    return NumberConfigFieldToJSONTyped(json, false);
}
function NumberConfigFieldToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'default_value': value['defaultValue'],
        'description': value['description'],
        'key': value['key'],
        'max_value': value['maxValue'],
        'min_value': value['minValue'],
        'name': value['name'],
        'type': value['type'],
        'value': value['value'],
    };
}
