// @getopenpay/client/models/PaymentProcessorName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentProcessorName = void 0;
exports.instanceOfPaymentProcessorName = instanceOfPaymentProcessorName;
exports.PaymentProcessorNameFromJSON = PaymentProcessorNameFromJSON;
exports.PaymentProcessorNameFromJSONTyped = PaymentProcessorNameFromJSONTyped;
exports.PaymentProcessorNameToJSON = PaymentProcessorNameToJSON;
exports.PaymentProcessorNameToJSONTyped = PaymentProcessorNameToJSONTyped;
/**
 * PaymentProcessors are api's that we support and can charge.
 * @export
 */
exports.PaymentProcessorName = {
    Adyen: 'adyen',
    Airwallex: 'airwallex',
    AuthorizeNet: 'authorize_net',
    CheckoutCom: 'checkout_com',
    Braintree: 'braintree',
    Stripe: 'stripe',
    Foobar: 'foobar',
    Pockyt: 'pockyt',
    Cybersource: 'cybersource',
    Loop: 'loop'
};
function instanceOfPaymentProcessorName(value) {
    for (const key in exports.PaymentProcessorName) {
        if (Object.prototype.hasOwnProperty.call(exports.PaymentProcessorName, key)) {
            if (exports.PaymentProcessorName[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function PaymentProcessorNameFromJSON(json) {
    return PaymentProcessorNameFromJSONTyped(json, false);
}
function PaymentProcessorNameFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function PaymentProcessorNameToJSON(value) {
    return value;
}
function PaymentProcessorNameToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
