// @getopenpay/client/models/PaymentMethodExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPaymentMethodExternal = instanceOfPaymentMethodExternal;
exports.PaymentMethodExternalFromJSON = PaymentMethodExternalFromJSON;
exports.PaymentMethodExternalFromJSONTyped = PaymentMethodExternalFromJSONTyped;
exports.PaymentMethodExternalToJSON = PaymentMethodExternalToJSON;
exports.PaymentMethodExternalToJSONTyped = PaymentMethodExternalToJSONTyped;
const PaymentProviderType_1 = require("./PaymentProviderType");
const PaymentMethodMappingExternal_1 = require("./PaymentMethodMappingExternal");
const CompleteAddress_1 = require("./CompleteAddress");
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the PaymentMethodExternal interface.
 */
function instanceOfPaymentMethodExternal(value) {
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('displayName' in value) || value['displayName'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('provider' in value) || value['provider'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function PaymentMethodExternalFromJSON(json) {
    return PaymentMethodExternalFromJSONTyped(json, false);
}
function PaymentMethodExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'billingAddress': json['billing_address'] == null ? undefined : (0, CompleteAddress_1.CompleteAddressFromJSON)(json['billing_address']),
        'cardBrand': json['card_brand'] == null ? undefined : json['card_brand'],
        'cardType': json['card_type'] == null ? undefined : json['card_type'],
        'createdAt': (new Date(json['created_at'])),
        'customerId': json['customer_id'] == null ? undefined : json['customer_id'],
        'displayName': json['display_name'],
        'id': json['id'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'lastFour': json['last_four'] == null ? undefined : json['last_four'],
        'mappings': json['mappings'] == null ? undefined : (json['mappings'].map(PaymentMethodMappingExternal_1.PaymentMethodMappingExternalFromJSON)),
        'metadata': json['metadata'] == null ? undefined : json['metadata'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'provider': (0, PaymentProviderType_1.PaymentProviderTypeFromJSON)(json['provider']),
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function PaymentMethodExternalToJSON(json) {
    return PaymentMethodExternalToJSONTyped(json, false);
}
function PaymentMethodExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'billing_address': (0, CompleteAddress_1.CompleteAddressToJSON)(value['billingAddress']),
        'card_brand': value['cardBrand'],
        'card_type': value['cardType'],
        'created_at': ((value['createdAt']).toISOString()),
        'customer_id': value['customerId'],
        'display_name': value['displayName'],
        'id': value['id'],
        'is_deleted': value['isDeleted'],
        'last_four': value['lastFour'],
        'mappings': value['mappings'] == null ? undefined : (value['mappings'].map(PaymentMethodMappingExternal_1.PaymentMethodMappingExternalToJSON)),
        'metadata': value['metadata'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'provider': (0, PaymentProviderType_1.PaymentProviderTypeToJSON)(value['provider']),
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
