// @getopenpay/client/models/PriceTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriceTypeEnum = void 0;
exports.instanceOfPriceTypeEnum = instanceOfPriceTypeEnum;
exports.PriceTypeEnumFromJSON = PriceTypeEnumFromJSON;
exports.PriceTypeEnumFromJSONTyped = PriceTypeEnumFromJSONTyped;
exports.PriceTypeEnumToJSON = PriceTypeEnumToJSON;
exports.PriceTypeEnumToJSONTyped = PriceTypeEnumToJSONTyped;
/**
 *
 * @export
 */
exports.PriceTypeEnum = {
    OneTime: 'one_time',
    Recurring: 'recurring'
};
function instanceOfPriceTypeEnum(value) {
    for (const key in exports.PriceTypeEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.PriceTypeEnum, key)) {
            if (exports.PriceTypeEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function PriceTypeEnumFromJSON(json) {
    return PriceTypeEnumFromJSONTyped(json, false);
}
function PriceTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function PriceTypeEnumToJSON(value) {
    return value;
}
function PriceTypeEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
