// @getopenpay/client/models/ProductExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfProductExternal = instanceOfProductExternal;
exports.ProductExternalFromJSON = ProductExternalFromJSON;
exports.ProductExternalFromJSONTyped = ProductExternalFromJSONTyped;
exports.ProductExternalToJSON = ProductExternalToJSON;
exports.ProductExternalToJSONTyped = ProductExternalToJSONTyped;
const PriceExternal_1 = require("./PriceExternal");
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the ProductExternal interface.
 */
function instanceOfProductExternal(value) {
    if (!('accountSku' in value) || value['accountSku'] === undefined)
        return false;
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('defaultPrice' in value) || value['defaultPrice'] === undefined)
        return false;
    if (!('description' in value) || value['description'] === undefined)
        return false;
    if (!('features' in value) || value['features'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('isActive' in value) || value['isActive'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('prices' in value) || value['prices'] === undefined)
        return false;
    if (!('unitLabel' in value) || value['unitLabel'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function ProductExternalFromJSON(json) {
    return ProductExternalFromJSONTyped(json, false);
}
function ProductExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'accountSku': json['account_sku'],
        'createdAt': (new Date(json['created_at'])),
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'defaultPrice': json['default_price'],
        'description': json['description'],
        'features': json['features'],
        'id': json['id'],
        'isActive': json['is_active'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'metadata': json['metadata'] == null ? undefined : json['metadata'],
        'name': json['name'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'prices': (json['prices'].map(PriceExternal_1.PriceExternalFromJSON)),
        'unitLabel': json['unit_label'],
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function ProductExternalToJSON(json) {
    return ProductExternalToJSONTyped(json, false);
}
function ProductExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'account_sku': value['accountSku'],
        'created_at': ((value['createdAt']).toISOString()),
        'custom_fields': value['customFields'],
        'default_price': value['defaultPrice'],
        'description': value['description'],
        'features': value['features'],
        'id': value['id'],
        'is_active': value['isActive'],
        'is_deleted': value['isDeleted'],
        'metadata': value['metadata'],
        'name': value['name'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'prices': (value['prices'].map(PriceExternal_1.PriceExternalToJSON)),
        'unit_label': value['unitLabel'],
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
