// @getopenpay/client/models/ProrationEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProrationEnum = void 0;
exports.instanceOfProrationEnum = instanceOfProrationEnum;
exports.ProrationEnumFromJSON = ProrationEnumFromJSON;
exports.ProrationEnumFromJSONTyped = ProrationEnumFromJSONTyped;
exports.ProrationEnumToJSON = ProrationEnumToJSON;
exports.ProrationEnumToJSONTyped = ProrationEnumToJSONTyped;
/**
 *
 * @export
 */
exports.ProrationEnum = {
    CreateProrations: 'create_prorations',
    None: 'none',
    AlwaysInvoice: 'always_invoice'
};
function instanceOfProrationEnum(value) {
    for (const key in exports.ProrationEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.ProrationEnum, key)) {
            if (exports.ProrationEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function ProrationEnumFromJSON(json) {
    return ProrationEnumFromJSONTyped(json, false);
}
function ProrationEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function ProrationEnumToJSON(value) {
    return value;
}
function ProrationEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
