// @getopenpay/client/models/RecurringDetails.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfRecurringDetails = instanceOfRecurringDetails;
exports.RecurringDetailsFromJSON = RecurringDetailsFromJSON;
exports.RecurringDetailsFromJSONTyped = RecurringDetailsFromJSONTyped;
exports.RecurringDetailsToJSON = RecurringDetailsToJSON;
exports.RecurringDetailsToJSONTyped = RecurringDetailsToJSONTyped;
const UsageAggMethodEnum_1 = require("./UsageAggMethodEnum");
const UsageTypeEnum_1 = require("./UsageTypeEnum");
/**
 * Check if a given object implements the RecurringDetails interface.
 */
function instanceOfRecurringDetails(value) {
    if (!('usageType' in value) || value['usageType'] === undefined)
        return false;
    return true;
}
function RecurringDetailsFromJSON(json) {
    return RecurringDetailsFromJSONTyped(json, false);
}
function RecurringDetailsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'aggregateUsage': json['aggregate_usage'] == null ? undefined : (0, UsageAggMethodEnum_1.UsageAggMethodEnumFromJSON)(json['aggregate_usage']),
        'trialPeriodDays': json['trial_period_days'] == null ? undefined : json['trial_period_days'],
        'usageType': (0, UsageTypeEnum_1.UsageTypeEnumFromJSON)(json['usage_type']),
    };
}
function RecurringDetailsToJSON(json) {
    return RecurringDetailsToJSONTyped(json, false);
}
function RecurringDetailsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'aggregate_usage': (0, UsageAggMethodEnum_1.UsageAggMethodEnumToJSON)(value['aggregateUsage']),
        'trial_period_days': value['trialPeriodDays'],
        'usage_type': (0, UsageTypeEnum_1.UsageTypeEnumToJSON)(value['usageType']),
    };
}
