// @getopenpay/client/models/RefundReasonEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefundReasonEnum = void 0;
exports.instanceOfRefundReasonEnum = instanceOfRefundReasonEnum;
exports.RefundReasonEnumFromJSON = RefundReasonEnumFromJSON;
exports.RefundReasonEnumFromJSONTyped = RefundReasonEnumFromJSONTyped;
exports.RefundReasonEnumToJSON = RefundReasonEnumToJSON;
exports.RefundReasonEnumToJSONTyped = RefundReasonEnumToJSONTyped;
/**
 *
 * @export
 */
exports.RefundReasonEnum = {
    Manual: 'manual',
    Duplicate: 'duplicate',
    Fraudulent: 'fraudulent',
    RequestedByCustomer: 'requested_by_customer',
    ExpiredUncapturedCharge: 'expired_uncaptured_charge'
};
function instanceOfRefundReasonEnum(value) {
    for (const key in exports.RefundReasonEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.RefundReasonEnum, key)) {
            if (exports.RefundReasonEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function RefundReasonEnumFromJSON(json) {
    return RefundReasonEnumFromJSONTyped(json, false);
}
function RefundReasonEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function RefundReasonEnumToJSON(value) {
    return value;
}
function RefundReasonEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
