// @getopenpay/client/models/ResumeSubscriptionResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfResumeSubscriptionResponse = instanceOfResumeSubscriptionResponse;
exports.ResumeSubscriptionResponseFromJSON = ResumeSubscriptionResponseFromJSON;
exports.ResumeSubscriptionResponseFromJSONTyped = ResumeSubscriptionResponseFromJSONTyped;
exports.ResumeSubscriptionResponseToJSON = ResumeSubscriptionResponseToJSON;
exports.ResumeSubscriptionResponseToJSONTyped = ResumeSubscriptionResponseToJSONTyped;
const InvoiceExternal_1 = require("./InvoiceExternal");
const SubscriptionExternal_1 = require("./SubscriptionExternal");
/**
 * Check if a given object implements the ResumeSubscriptionResponse interface.
 */
function instanceOfResumeSubscriptionResponse(value) {
    if (!('subscription' in value) || value['subscription'] === undefined)
        return false;
    return true;
}
function ResumeSubscriptionResponseFromJSON(json) {
    return ResumeSubscriptionResponseFromJSONTyped(json, false);
}
function ResumeSubscriptionResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'invoice': json['invoice'] == null ? undefined : (0, InvoiceExternal_1.InvoiceExternalFromJSON)(json['invoice']),
        'subscription': (0, SubscriptionExternal_1.SubscriptionExternalFromJSON)(json['subscription']),
    };
}
function ResumeSubscriptionResponseToJSON(json) {
    return ResumeSubscriptionResponseToJSONTyped(json, false);
}
function ResumeSubscriptionResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'invoice': (0, InvoiceExternal_1.InvoiceExternalToJSON)(value['invoice']),
        'subscription': (0, SubscriptionExternal_1.SubscriptionExternalToJSON)(value['subscription']),
    };
}
