// @getopenpay/client/models/SearchFilter.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSearchFilter = instanceOfSearchFilter;
exports.SearchFilterFromJSON = SearchFilterFromJSON;
exports.SearchFilterFromJSONTyped = SearchFilterFromJSONTyped;
exports.SearchFilterToJSON = SearchFilterToJSON;
exports.SearchFilterToJSONTyped = SearchFilterToJSONTyped;
/**
 * Check if a given object implements the SearchFilter interface.
 */
function instanceOfSearchFilter(value) {
    return true;
}
function SearchFilterFromJSON(json) {
    return SearchFilterFromJSONTyped(json, false);
}
function SearchFilterFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'eq': json['eq'] == null ? undefined : json['eq'],
        'ilike': json['ilike'] == null ? undefined : json['ilike'],
    };
}
function SearchFilterToJSON(json) {
    return SearchFilterToJSONTyped(json, false);
}
function SearchFilterToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'eq': value['eq'],
        'ilike': value['ilike'],
    };
}
