// @getopenpay/client/models/SelectedPriceQuantity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSelectedPriceQuantity = instanceOfSelectedPriceQuantity;
exports.SelectedPriceQuantityFromJSON = SelectedPriceQuantityFromJSON;
exports.SelectedPriceQuantityFromJSONTyped = SelectedPriceQuantityFromJSONTyped;
exports.SelectedPriceQuantityToJSON = SelectedPriceQuantityToJSON;
exports.SelectedPriceQuantityToJSONTyped = SelectedPriceQuantityToJSONTyped;
/**
 * Check if a given object implements the SelectedPriceQuantity interface.
 */
function instanceOfSelectedPriceQuantity(value) {
    if (!('priceId' in value) || value['priceId'] === undefined)
        return false;
    if (!('quantity' in value) || value['quantity'] === undefined)
        return false;
    return true;
}
function SelectedPriceQuantityFromJSON(json) {
    return SelectedPriceQuantityFromJSONTyped(json, false);
}
function SelectedPriceQuantityFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'priceId': json['price_id'],
        'quantity': json['quantity'],
    };
}
function SelectedPriceQuantityToJSON(json) {
    return SelectedPriceQuantityToJSONTyped(json, false);
}
function SelectedPriceQuantityToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'price_id': value['priceId'],
        'quantity': value['quantity'],
    };
}
