// @getopenpay/client/models/SignupQuestionnaireStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSignupQuestionnaireStatus = instanceOfSignupQuestionnaireStatus;
exports.SignupQuestionnaireStatusFromJSON = SignupQuestionnaireStatusFromJSON;
exports.SignupQuestionnaireStatusFromJSONTyped = SignupQuestionnaireStatusFromJSONTyped;
exports.SignupQuestionnaireStatusToJSON = SignupQuestionnaireStatusToJSON;
exports.SignupQuestionnaireStatusToJSONTyped = SignupQuestionnaireStatusToJSONTyped;
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the SignupQuestionnaireStatus interface.
 */
function instanceOfSignupQuestionnaireStatus(value) {
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('status' in value) || value['status'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function SignupQuestionnaireStatusFromJSON(json) {
    return SignupQuestionnaireStatusFromJSONTyped(json, false);
}
function SignupQuestionnaireStatusFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'createdAt': (new Date(json['created_at'])),
        'id': json['id'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'status': json['status'],
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function SignupQuestionnaireStatusToJSON(json) {
    return SignupQuestionnaireStatusToJSONTyped(json, false);
}
function SignupQuestionnaireStatusToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created_at': ((value['createdAt']).toISOString()),
        'id': value['id'],
        'is_deleted': value['isDeleted'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'status': value['status'],
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
