// @getopenpay/client/models/SubscriptionResumeRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSubscriptionResumeRequest = instanceOfSubscriptionResumeRequest;
exports.SubscriptionResumeRequestFromJSON = SubscriptionResumeRequestFromJSON;
exports.SubscriptionResumeRequestFromJSONTyped = SubscriptionResumeRequestFromJSONTyped;
exports.SubscriptionResumeRequestToJSON = SubscriptionResumeRequestToJSON;
exports.SubscriptionResumeRequestToJSONTyped = SubscriptionResumeRequestToJSONTyped;
const ProrationEnum_1 = require("./ProrationEnum");
/**
 * Check if a given object implements the SubscriptionResumeRequest interface.
 */
function instanceOfSubscriptionResumeRequest(value) {
    return true;
}
function SubscriptionResumeRequestFromJSON(json) {
    return SubscriptionResumeRequestFromJSONTyped(json, false);
}
function SubscriptionResumeRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'isPreview': json['is_preview'] == null ? undefined : json['is_preview'],
        'prorationBehavior': json['proration_behavior'] == null ? undefined : (0, ProrationEnum_1.ProrationEnumFromJSON)(json['proration_behavior']),
    };
}
function SubscriptionResumeRequestToJSON(json) {
    return SubscriptionResumeRequestToJSONTyped(json, false);
}
function SubscriptionResumeRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'is_preview': value['isPreview'],
        'proration_behavior': (0, ProrationEnum_1.ProrationEnumToJSON)(value['prorationBehavior']),
    };
}
