// @getopenpay/client/models/TaxIntegrationApiName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxIntegrationApiName = void 0;
exports.instanceOfTaxIntegrationApiName = instanceOfTaxIntegrationApiName;
exports.TaxIntegrationApiNameFromJSON = TaxIntegrationApiNameFromJSON;
exports.TaxIntegrationApiNameFromJSONTyped = TaxIntegrationApiNameFromJSONTyped;
exports.TaxIntegrationApiNameToJSON = TaxIntegrationApiNameToJSON;
exports.TaxIntegrationApiNameToJSONTyped = TaxIntegrationApiNameToJSONTyped;
/**
 *
 * @export
 */
exports.TaxIntegrationApiName = {
    AnrokV1: 'anrok_v1'
};
function instanceOfTaxIntegrationApiName(value) {
    for (const key in exports.TaxIntegrationApiName) {
        if (Object.prototype.hasOwnProperty.call(exports.TaxIntegrationApiName, key)) {
            if (exports.TaxIntegrationApiName[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function TaxIntegrationApiNameFromJSON(json) {
    return TaxIntegrationApiNameFromJSONTyped(json, false);
}
function TaxIntegrationApiNameFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function TaxIntegrationApiNameToJSON(value) {
    return value;
}
function TaxIntegrationApiNameToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
