// @getopenpay/client/models/UpdatePaymentIntent.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdatePaymentIntent = instanceOfUpdatePaymentIntent;
exports.UpdatePaymentIntentFromJSON = UpdatePaymentIntentFromJSON;
exports.UpdatePaymentIntentFromJSONTyped = UpdatePaymentIntentFromJSONTyped;
exports.UpdatePaymentIntentToJSON = UpdatePaymentIntentToJSON;
exports.UpdatePaymentIntentToJSONTyped = UpdatePaymentIntentToJSONTyped;
/**
 * Check if a given object implements the UpdatePaymentIntent interface.
 */
function instanceOfUpdatePaymentIntent(value) {
    return true;
}
function UpdatePaymentIntentFromJSON(json) {
    return UpdatePaymentIntentFromJSONTyped(json, false);
}
function UpdatePaymentIntentFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
    };
}
function UpdatePaymentIntentToJSON(json) {
    return UpdatePaymentIntentToJSONTyped(json, false);
}
function UpdatePaymentIntentToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'custom_fields': value['customFields'],
    };
}
