// @getopenpay/client/models/UpdateChargeRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateChargeRequest = instanceOfUpdateChargeRequest;
exports.UpdateChargeRequestFromJSON = UpdateChargeRequestFromJSON;
exports.UpdateChargeRequestFromJSONTyped = UpdateChargeRequestFromJSONTyped;
exports.UpdateChargeRequestToJSON = UpdateChargeRequestToJSON;
exports.UpdateChargeRequestToJSONTyped = UpdateChargeRequestToJSONTyped;
/**
 * Check if a given object implements the UpdateChargeRequest interface.
 */
function instanceOfUpdateChargeRequest(value) {
    return true;
}
function UpdateChargeRequestFromJSON(json) {
    return UpdateChargeRequestFromJSONTyped(json, false);
}
function UpdateChargeRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
    };
}
function UpdateChargeRequestToJSON(json) {
    return UpdateChargeRequestToJSONTyped(json, false);
}
function UpdateChargeRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'custom_fields': value['customFields'],
    };
}
