// @getopenpay/client/models/UpdateInvoiceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateInvoiceRequest = instanceOfUpdateInvoiceRequest;
exports.UpdateInvoiceRequestFromJSON = UpdateInvoiceRequestFromJSON;
exports.UpdateInvoiceRequestFromJSONTyped = UpdateInvoiceRequestFromJSONTyped;
exports.UpdateInvoiceRequestToJSON = UpdateInvoiceRequestToJSON;
exports.UpdateInvoiceRequestToJSONTyped = UpdateInvoiceRequestToJSONTyped;
/**
 * Check if a given object implements the UpdateInvoiceRequest interface.
 */
function instanceOfUpdateInvoiceRequest(value) {
    return true;
}
function UpdateInvoiceRequestFromJSON(json) {
    return UpdateInvoiceRequestFromJSONTyped(json, false);
}
function UpdateInvoiceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'couponId': json['coupon_id'] == null ? undefined : json['coupon_id'],
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'description': json['description'] == null ? undefined : json['description'],
        'emailInvoiceOnFinalization': json['email_invoice_on_finalization'] == null ? undefined : json['email_invoice_on_finalization'],
        'paymentMethodId': json['payment_method_id'] == null ? undefined : json['payment_method_id'],
    };
}
function UpdateInvoiceRequestToJSON(json) {
    return UpdateInvoiceRequestToJSONTyped(json, false);
}
function UpdateInvoiceRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'coupon_id': value['couponId'],
        'custom_fields': value['customFields'],
        'description': value['description'],
        'email_invoice_on_finalization': value['emailInvoiceOnFinalization'],
        'payment_method_id': value['paymentMethodId'],
    };
}
