// @getopenpay/client/apis/PaymentMethodsApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodsApi = void 0;
const runtime = __importStar(require("../runtime"));
const index_1 = require("../models/index");
/**
 *
 */
class PaymentMethodsApi extends runtime.BaseAPI {
    /**
     * Attaches the payment_method to a customer. A Payment_method must be attached to a customer before it can be used for payment. Once attached to a customer, it can not be reattached to a different customer.
     * Attach Payment Method
     */
    attachPaymentMethodRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['paymentMethodId'] == null) {
                throw new runtime.RequiredError('paymentMethodId', 'Required parameter "paymentMethodId" was null or undefined when calling attachPaymentMethod().');
            }
            if (requestParameters['attachPaymentMethodRequest'] == null) {
                throw new runtime.RequiredError('attachPaymentMethodRequest', 'Required parameter "attachPaymentMethodRequest" was null or undefined when calling attachPaymentMethod().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/payment-methods/{payment_method_id}/attach`.replace(`{${"payment_method_id"}}`, encodeURIComponent(String(requestParameters['paymentMethodId']))),
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.AttachPaymentMethodRequestToJSON)(requestParameters['attachPaymentMethodRequest']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PaymentMethodExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Attaches the payment_method to a customer. A Payment_method must be attached to a customer before it can be used for payment. Once attached to a customer, it can not be reattached to a different customer.
     * Attach Payment Method
     */
    attachPaymentMethod(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.attachPaymentMethodRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Authorize Payment Method
     */
    authorizePaymentMethodRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['authorizePaymentMethodRequest'] == null) {
                throw new runtime.RequiredError('authorizePaymentMethodRequest', 'Required parameter "authorizePaymentMethodRequest" was null or undefined when calling authorizePaymentMethod().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/payment-methods/authorize`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.AuthorizePaymentMethodRequestToJSON)(requestParameters['authorizePaymentMethodRequest']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PaymentMethodExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Authorize Payment Method
     */
    authorizePaymentMethod(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.authorizePaymentMethodRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Detaches a payment method from a customer.  A detached payment method can no longer be used for payment, nor can it be reattached to the same or a different customer.  A payment method cannot be detached if it is the default payment method for at least one customer, invoice, or active subscription.
     * Detach Payment Method
     */
    detachPaymentMethodRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['paymentMethodId'] == null) {
                throw new runtime.RequiredError('paymentMethodId', 'Required parameter "paymentMethodId" was null or undefined when calling detachPaymentMethod().');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/payment-methods/{payment_method_id}/detach`.replace(`{${"payment_method_id"}}`, encodeURIComponent(String(requestParameters['paymentMethodId']))),
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PaymentMethodExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Detaches a payment method from a customer.  A detached payment method can no longer be used for payment, nor can it be reattached to the same or a different customer.  A payment method cannot be detached if it is the default payment method for at least one customer, invoice, or active subscription.
     * Detach Payment Method
     */
    detachPaymentMethod(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.detachPaymentMethodRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Get Payment Method
     */
    getPaymentMethodRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['paymentMethodId'] == null) {
                throw new runtime.RequiredError('paymentMethodId', 'Required parameter "paymentMethodId" was null or undefined when calling getPaymentMethod().');
            }
            const queryParameters = {};
            if (requestParameters['expand'] != null) {
                queryParameters['expand'] = requestParameters['expand'];
            }
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/payment-methods/{payment_method_id}`.replace(`{${"payment_method_id"}}`, encodeURIComponent(String(requestParameters['paymentMethodId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.ResponseGetPaymentMethodPaymentMethodsPaymentMethodIdGetFromJSON)(jsonValue));
        });
    }
    /**
     * Get Payment Method
     */
    getPaymentMethod(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getPaymentMethodRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
}
exports.PaymentMethodsApi = PaymentMethodsApi;
