// @getopenpay/client/apis/PricesApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricesApi = void 0;
const runtime = __importStar(require("../runtime"));
const index_1 = require("../models/index");
/**
 *
 */
class PricesApi extends runtime.BaseAPI {
    /**
     * Create Price For Product
     */
    createPriceForProductRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['createPriceRequest'] == null) {
                throw new runtime.RequiredError('createPriceRequest', 'Required parameter "createPriceRequest" was null or undefined when calling createPriceForProduct().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/prices/`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.CreatePriceRequestToJSON)(requestParameters['createPriceRequest']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PriceExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Create Price For Product
     */
    createPriceForProduct(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.createPriceForProductRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Delete Price For Product
     */
    deletePriceForProductRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['priceId'] == null) {
                throw new runtime.RequiredError('priceId', 'Required parameter "priceId" was null or undefined when calling deletePriceForProduct().');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/prices/{price_id}`.replace(`{${"price_id"}}`, encodeURIComponent(String(requestParameters['priceId']))),
                method: 'DELETE',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.DeletePriceResponseFromJSON)(jsonValue));
        });
    }
    /**
     * Delete Price For Product
     */
    deletePriceForProduct(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.deletePriceForProductRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Returns list of prices which can only be purchased with the given price.
     * Get Price Add Ons
     */
    getPriceAddOnsRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['priceId'] == null) {
                throw new runtime.RequiredError('priceId', 'Required parameter "priceId" was null or undefined when calling getPriceAddOns().');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/prices/{price_id}/add-ons`.replace(`{${"price_id"}}`, encodeURIComponent(String(requestParameters['priceId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(index_1.PriceExternalFromJSON));
        });
    }
    /**
     * Returns list of prices which can only be purchased with the given price.
     * Get Price Add Ons
     */
    getPriceAddOns(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getPriceAddOnsRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Get Price For Product
     */
    getPriceForProductRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['priceId'] == null) {
                throw new runtime.RequiredError('priceId', 'Required parameter "priceId" was null or undefined when calling getPriceForProduct().');
            }
            const queryParameters = {};
            if (requestParameters['expand'] != null) {
                queryParameters['expand'] = requestParameters['expand'];
            }
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/prices/{price_id}`.replace(`{${"price_id"}}`, encodeURIComponent(String(requestParameters['priceId']))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PriceExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Get Price For Product
     */
    getPriceForProduct(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getPriceForProductRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * List Prices
     */
    listPricesRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['priceQueryParams'] == null) {
                throw new runtime.RequiredError('priceQueryParams', 'Required parameter "priceQueryParams" was null or undefined when calling listPrices().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/prices/list`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.PriceQueryParamsToJSON)(requestParameters['priceQueryParams']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.ListResponsePriceExternalFromJSON)(jsonValue));
        });
    }
    /**
     * List Prices
     */
    listPrices(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.listPricesRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
    /**
     * Update Price For Product
     */
    updatePriceForProductRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters['priceId'] == null) {
                throw new runtime.RequiredError('priceId', 'Required parameter "priceId" was null or undefined when calling updatePriceForProduct().');
            }
            if (requestParameters['updatePriceRequest'] == null) {
                throw new runtime.RequiredError('updatePriceRequest', 'Required parameter "updatePriceRequest" was null or undefined when calling updatePriceForProduct().');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("HTTPBearer", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/prices/{price_id}`.replace(`{${"price_id"}}`, encodeURIComponent(String(requestParameters['priceId']))),
                method: 'PUT',
                headers: headerParameters,
                query: queryParameters,
                body: (0, index_1.UpdatePriceRequestToJSON)(requestParameters['updatePriceRequest']),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.PriceExternalFromJSON)(jsonValue));
        });
    }
    /**
     * Update Price For Product
     */
    updatePriceForProduct(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.updatePriceForProductRaw(requestParameters, initOverrides);
            return yield response.value();
        });
    }
}
exports.PricesApi = PricesApi;
