// @getopenpay/client/models/BillingSchemeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingSchemeEnum = void 0;
exports.instanceOfBillingSchemeEnum = instanceOfBillingSchemeEnum;
exports.BillingSchemeEnumFromJSON = BillingSchemeEnumFromJSON;
exports.BillingSchemeEnumFromJSONTyped = BillingSchemeEnumFromJSONTyped;
exports.BillingSchemeEnumToJSON = BillingSchemeEnumToJSON;
exports.BillingSchemeEnumToJSONTyped = BillingSchemeEnumToJSONTyped;
/**
 *
 * @export
 */
exports.BillingSchemeEnum = {
    PerUnit: 'per_unit',
    Tiered: 'tiered'
};
function instanceOfBillingSchemeEnum(value) {
    for (const key in exports.BillingSchemeEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.BillingSchemeEnum, key)) {
            if (exports.BillingSchemeEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function BillingSchemeEnumFromJSON(json) {
    return BillingSchemeEnumFromJSONTyped(json, false);
}
function BillingSchemeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function BillingSchemeEnumToJSON(value) {
    return value;
}
function BillingSchemeEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
