// @getopenpay/client/models/BillingMeterQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBillingMeterQueryParams = instanceOfBillingMeterQueryParams;
exports.BillingMeterQueryParamsFromJSON = BillingMeterQueryParamsFromJSON;
exports.BillingMeterQueryParamsFromJSONTyped = BillingMeterQueryParamsFromJSONTyped;
exports.BillingMeterQueryParamsToJSON = BillingMeterQueryParamsToJSON;
exports.BillingMeterQueryParamsToJSONTyped = BillingMeterQueryParamsToJSONTyped;
const DateTimeFilter_1 = require("./DateTimeFilter");
/**
 * Check if a given object implements the BillingMeterQueryParams interface.
 */
function instanceOfBillingMeterQueryParams(value) {
    if (!('isActive' in value) || value['isActive'] === undefined)
        return false;
    return true;
}
function BillingMeterQueryParamsFromJSON(json) {
    return BillingMeterQueryParamsFromJSONTyped(json, false);
}
function BillingMeterQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'expand': json['expand'] == null ? undefined : json['expand'],
        'isActive': json['is_active'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function BillingMeterQueryParamsToJSON(json) {
    return BillingMeterQueryParamsToJSONTyped(json, false);
}
function BillingMeterQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'expand': value['expand'],
        'is_active': value['isActive'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
