// @getopenpay/client/models/BooleanConfigField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BooleanConfigFieldTypeEnum = void 0;
exports.instanceOfBooleanConfigField = instanceOfBooleanConfigField;
exports.BooleanConfigFieldFromJSON = BooleanConfigFieldFromJSON;
exports.BooleanConfigFieldFromJSONTyped = BooleanConfigFieldFromJSONTyped;
exports.BooleanConfigFieldToJSON = BooleanConfigFieldToJSON;
exports.BooleanConfigFieldToJSONTyped = BooleanConfigFieldToJSONTyped;
/**
 * @export
 */
exports.BooleanConfigFieldTypeEnum = {
    Boolean: 'boolean'
};
/**
 * Check if a given object implements the BooleanConfigField interface.
 */
function instanceOfBooleanConfigField(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('value' in value) || value['value'] === undefined)
        return false;
    return true;
}
function BooleanConfigFieldFromJSON(json) {
    return BooleanConfigFieldFromJSONTyped(json, false);
}
function BooleanConfigFieldFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'defaultValue': json['default_value'] == null ? undefined : json['default_value'],
        'description': json['description'] == null ? undefined : json['description'],
        'key': json['key'],
        'name': json['name'],
        'type': json['type'] == null ? undefined : json['type'],
        'value': json['value'],
    };
}
function BooleanConfigFieldToJSON(json) {
    return BooleanConfigFieldToJSONTyped(json, false);
}
function BooleanConfigFieldToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'default_value': value['defaultValue'],
        'description': value['description'],
        'key': value['key'],
        'name': value['name'],
        'type': value['type'],
        'value': value['value'],
    };
}
