// @getopenpay/client/models/ChargeStatusEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChargeStatusEnum = void 0;
exports.instanceOfChargeStatusEnum = instanceOfChargeStatusEnum;
exports.ChargeStatusEnumFromJSON = ChargeStatusEnumFromJSON;
exports.ChargeStatusEnumFromJSONTyped = ChargeStatusEnumFromJSONTyped;
exports.ChargeStatusEnumToJSON = ChargeStatusEnumToJSON;
exports.ChargeStatusEnumToJSONTyped = ChargeStatusEnumToJSONTyped;
/**
 *
 * @export
 */
exports.ChargeStatusEnum = {
    Succeeded: 'succeeded',
    Failed: 'failed',
    Pending: 'pending'
};
function instanceOfChargeStatusEnum(value) {
    for (const key in exports.ChargeStatusEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.ChargeStatusEnum, key)) {
            if (exports.ChargeStatusEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function ChargeStatusEnumFromJSON(json) {
    return ChargeStatusEnumFromJSONTyped(json, false);
}
function ChargeStatusEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function ChargeStatusEnumToJSON(value) {
    return value;
}
function ChargeStatusEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
