// @getopenpay/client/models/CheckoutMode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutMode = void 0;
exports.instanceOfCheckoutMode = instanceOfCheckoutMode;
exports.CheckoutModeFromJSON = CheckoutModeFromJSON;
exports.CheckoutModeFromJSONTyped = CheckoutModeFromJSONTyped;
exports.CheckoutModeToJSON = CheckoutModeToJSON;
exports.CheckoutModeToJSONTyped = CheckoutModeToJSONTyped;
/**
 *
 * @export
 */
exports.CheckoutMode = {
    Payment: 'payment',
    Setup: 'setup',
    Subscription: 'subscription'
};
function instanceOfCheckoutMode(value) {
    for (const key in exports.CheckoutMode) {
        if (Object.prototype.hasOwnProperty.call(exports.CheckoutMode, key)) {
            if (exports.CheckoutMode[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CheckoutModeFromJSON(json) {
    return CheckoutModeFromJSONTyped(json, false);
}
function CheckoutModeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CheckoutModeToJSON(value) {
    return value;
}
function CheckoutModeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
