// @getopenpay/client/models/CheckoutSessionStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutSessionStatus = void 0;
exports.instanceOfCheckoutSessionStatus = instanceOfCheckoutSessionStatus;
exports.CheckoutSessionStatusFromJSON = CheckoutSessionStatusFromJSON;
exports.CheckoutSessionStatusFromJSONTyped = CheckoutSessionStatusFromJSONTyped;
exports.CheckoutSessionStatusToJSON = CheckoutSessionStatusToJSON;
exports.CheckoutSessionStatusToJSONTyped = CheckoutSessionStatusToJSONTyped;
/**
 *
 * @export
 */
exports.CheckoutSessionStatus = {
    Open: 'open',
    Expired: 'expired',
    Complete: 'complete'
};
function instanceOfCheckoutSessionStatus(value) {
    for (const key in exports.CheckoutSessionStatus) {
        if (Object.prototype.hasOwnProperty.call(exports.CheckoutSessionStatus, key)) {
            if (exports.CheckoutSessionStatus[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CheckoutSessionStatusFromJSON(json) {
    return CheckoutSessionStatusFromJSONTyped(json, false);
}
function CheckoutSessionStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CheckoutSessionStatusToJSON(value) {
    return value;
}
function CheckoutSessionStatusToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
