// @getopenpay/client/models/CollectionMethodEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CollectionMethodEnum = void 0;
exports.instanceOfCollectionMethodEnum = instanceOfCollectionMethodEnum;
exports.CollectionMethodEnumFromJSON = CollectionMethodEnumFromJSON;
exports.CollectionMethodEnumFromJSONTyped = CollectionMethodEnumFromJSONTyped;
exports.CollectionMethodEnumToJSON = CollectionMethodEnumToJSON;
exports.CollectionMethodEnumToJSONTyped = CollectionMethodEnumToJSONTyped;
/**
 *
 * @export
 */
exports.CollectionMethodEnum = {
    ChargeAutomatically: 'charge_automatically',
    SendInvoice: 'send_invoice'
};
function instanceOfCollectionMethodEnum(value) {
    for (const key in exports.CollectionMethodEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.CollectionMethodEnum, key)) {
            if (exports.CollectionMethodEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CollectionMethodEnumFromJSON(json) {
    return CollectionMethodEnumFromJSONTyped(json, false);
}
function CollectionMethodEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CollectionMethodEnumToJSON(value) {
    return value;
}
function CollectionMethodEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
