// @getopenpay/client/models/CreateCustomerBalanceTransactionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateCustomerBalanceTransactionRequest = instanceOfCreateCustomerBalanceTransactionRequest;
exports.CreateCustomerBalanceTransactionRequestFromJSON = CreateCustomerBalanceTransactionRequestFromJSON;
exports.CreateCustomerBalanceTransactionRequestFromJSONTyped = CreateCustomerBalanceTransactionRequestFromJSONTyped;
exports.CreateCustomerBalanceTransactionRequestToJSON = CreateCustomerBalanceTransactionRequestToJSON;
exports.CreateCustomerBalanceTransactionRequestToJSONTyped = CreateCustomerBalanceTransactionRequestToJSONTyped;
const CurrencyEnum_1 = require("./CurrencyEnum");
/**
 * Check if a given object implements the CreateCustomerBalanceTransactionRequest interface.
 */
function instanceOfCreateCustomerBalanceTransactionRequest(value) {
    if (!('amountAtom' in value) || value['amountAtom'] === undefined)
        return false;
    return true;
}
function CreateCustomerBalanceTransactionRequestFromJSON(json) {
    return CreateCustomerBalanceTransactionRequestFromJSONTyped(json, false);
}
function CreateCustomerBalanceTransactionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amountAtom': json['amount_atom'],
        'currency': json['currency'] == null ? undefined : (0, CurrencyEnum_1.CurrencyEnumFromJSON)(json['currency']),
        'description': json['description'] == null ? undefined : json['description'],
    };
}
function CreateCustomerBalanceTransactionRequestToJSON(json) {
    return CreateCustomerBalanceTransactionRequestToJSONTyped(json, false);
}
function CreateCustomerBalanceTransactionRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'amount_atom': value['amountAtom'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumToJSON)(value['currency']),
        'description': value['description'],
    };
}
