// @getopenpay/client/models/CreditNoteLineType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreditNoteLineType = void 0;
exports.instanceOfCreditNoteLineType = instanceOfCreditNoteLineType;
exports.CreditNoteLineTypeFromJSON = CreditNoteLineTypeFromJSON;
exports.CreditNoteLineTypeFromJSONTyped = CreditNoteLineTypeFromJSONTyped;
exports.CreditNoteLineTypeToJSON = CreditNoteLineTypeToJSON;
exports.CreditNoteLineTypeToJSONTyped = CreditNoteLineTypeToJSONTyped;
/**
 *
 * @export
 */
exports.CreditNoteLineType = {
    InvoiceLineItem: 'invoice_line_item',
    CustomLineItem: 'custom_line_item'
};
function instanceOfCreditNoteLineType(value) {
    for (const key in exports.CreditNoteLineType) {
        if (Object.prototype.hasOwnProperty.call(exports.CreditNoteLineType, key)) {
            if (exports.CreditNoteLineType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CreditNoteLineTypeFromJSON(json) {
    return CreditNoteLineTypeFromJSONTyped(json, false);
}
function CreditNoteLineTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CreditNoteLineTypeToJSON(value) {
    return value;
}
function CreditNoteLineTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
