// @getopenpay/client/models/CreditNoteQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreditNoteQueryParams = instanceOfCreditNoteQueryParams;
exports.CreditNoteQueryParamsFromJSON = CreditNoteQueryParamsFromJSON;
exports.CreditNoteQueryParamsFromJSONTyped = CreditNoteQueryParamsFromJSONTyped;
exports.CreditNoteQueryParamsToJSON = CreditNoteQueryParamsToJSON;
exports.CreditNoteQueryParamsToJSONTyped = CreditNoteQueryParamsToJSONTyped;
const DateTimeFilter_1 = require("./DateTimeFilter");
/**
 * Check if a given object implements the CreditNoteQueryParams interface.
 */
function instanceOfCreditNoteQueryParams(value) {
    return true;
}
function CreditNoteQueryParamsFromJSON(json) {
    return CreditNoteQueryParamsFromJSONTyped(json, false);
}
function CreditNoteQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'customerId': json['customer_id'] == null ? undefined : json['customer_id'],
        'expand': json['expand'] == null ? undefined : json['expand'],
        'invoiceId': json['invoice_id'] == null ? undefined : json['invoice_id'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function CreditNoteQueryParamsToJSON(json) {
    return CreditNoteQueryParamsToJSONTyped(json, false);
}
function CreditNoteQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'customer_id': value['customerId'],
        'expand': value['expand'],
        'invoice_id': value['invoiceId'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
