// @getopenpay/client/models/CreditNoteItemExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreditNoteItemExternal = instanceOfCreditNoteItemExternal;
exports.CreditNoteItemExternalFromJSON = CreditNoteItemExternalFromJSON;
exports.CreditNoteItemExternalFromJSONTyped = CreditNoteItemExternalFromJSONTyped;
exports.CreditNoteItemExternalToJSON = CreditNoteItemExternalToJSON;
exports.CreditNoteItemExternalToJSONTyped = CreditNoteItemExternalToJSONTyped;
const CreditNoteLineType_1 = require("./CreditNoteLineType");
const CurrencyEnum_1 = require("./CurrencyEnum");
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the CreditNoteItemExternal interface.
 */
function instanceOfCreditNoteItemExternal(value) {
    if (!('amountAtom' in value) || value['amountAtom'] === undefined)
        return false;
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('currency' in value) || value['currency'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function CreditNoteItemExternalFromJSON(json) {
    return CreditNoteItemExternalFromJSONTyped(json, false);
}
function CreditNoteItemExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amountAtom': json['amount_atom'],
        'createdAt': (new Date(json['created_at'])),
        'currency': (0, CurrencyEnum_1.CurrencyEnumFromJSON)(json['currency']),
        'id': json['id'],
        'invoiceItemId': json['invoice_item_id'] == null ? undefined : json['invoice_item_id'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'quantity': json['quantity'] == null ? undefined : json['quantity'],
        'type': (0, CreditNoteLineType_1.CreditNoteLineTypeFromJSON)(json['type']),
        'unitAmountAtom': json['unit_amount_atom'] == null ? undefined : json['unit_amount_atom'],
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function CreditNoteItemExternalToJSON(json) {
    return CreditNoteItemExternalToJSONTyped(json, false);
}
function CreditNoteItemExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'amount_atom': value['amountAtom'],
        'created_at': ((value['createdAt']).toISOString()),
        'currency': (0, CurrencyEnum_1.CurrencyEnumToJSON)(value['currency']),
        'id': value['id'],
        'invoice_item_id': value['invoiceItemId'],
        'is_deleted': value['isDeleted'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'quantity': value['quantity'],
        'type': (0, CreditNoteLineType_1.CreditNoteLineTypeToJSON)(value['type']),
        'unit_amount_atom': value['unitAmountAtom'],
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
