// @getopenpay/client/models/CurrencyEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CurrencyEnum = void 0;
exports.instanceOfCurrencyEnum = instanceOfCurrencyEnum;
exports.CurrencyEnumFromJSON = CurrencyEnumFromJSON;
exports.CurrencyEnumFromJSONTyped = CurrencyEnumFromJSONTyped;
exports.CurrencyEnumToJSON = CurrencyEnumToJSON;
exports.CurrencyEnumToJSONTyped = CurrencyEnumToJSONTyped;
/**
 *
 * @export
 */
exports.CurrencyEnum = {
    Usd: 'usd',
    Brl: 'brl',
    Eur: 'eur',
    Gbp: 'gbp',
    Aud: 'aud',
    Cad: 'cad',
    Nok: 'nok',
    Sek: 'sek',
    Mxn: 'mxn',
    Inr: 'inr',
    Ars: 'ars',
    Cop: 'cop',
    Clp: 'clp',
    Nzd: 'nzd',
    Jpy: 'jpy',
    Chf: 'chf',
    Cny: 'cny',
    Hkd: 'hkd',
    Sgd: 'sgd',
    Pln: 'pln',
    Dkk: 'dkk',
    Zar: 'zar',
    Thb: 'thb',
    Myr: 'myr',
    Ils: 'ils',
    Php: 'php',
    Try: 'try',
    Idr: 'idr',
    Czk: 'czk',
    Aed: 'aed',
    Sar: 'sar',
    Rub: 'rub',
    All: 'all',
    Amd: 'amd',
    Ang: 'ang',
    Aoa: 'aoa',
    Awg: 'awg',
    Azn: 'azn',
    Bam: 'bam',
    Bbd: 'bbd',
    Bdt: 'bdt',
    Bgn: 'bgn',
    Bif: 'bif',
    Bmd: 'bmd',
    Bnd: 'bnd',
    Bob: 'bob',
    Bsd: 'bsd',
    Bwp: 'bwp',
    Bzd: 'bzd',
    Crc: 'crc',
    Cve: 'cve',
    Djf: 'djf',
    Dop: 'dop',
    Dzd: 'dzd',
    Egp: 'egp',
    Etb: 'etb',
    Fjd: 'fjd',
    Fkp: 'fkp',
    Gel: 'gel',
    Gip: 'gip',
    Gmd: 'gmd',
    Gnf: 'gnf',
    Gtq: 'gtq',
    Gyd: 'gyd',
    Hnl: 'hnl',
    Huf: 'huf',
    Jmd: 'jmd',
    Kes: 'kes',
    Kgs: 'kgs',
    Khr: 'khr',
    Kmf: 'kmf',
    Krw: 'krw',
    Kyd: 'kyd',
    Kzt: 'kzt',
    Lak: 'lak',
    Lbp: 'lbp',
    Lkr: 'lkr',
    Lrd: 'lrd',
    Lsl: 'lsl',
    Mad: 'mad',
    Mdl: 'mdl',
    Mkd: 'mkd',
    Mnt: 'mnt',
    Mop: 'mop',
    Mur: 'mur',
    Mvr: 'mvr',
    Mwk: 'mwk',
    Nad: 'nad',
    Ngn: 'ngn',
    Nio: 'nio',
    Npr: 'npr',
    Pab: 'pab',
    Pen: 'pen',
    Pgk: 'pgk',
    Pkr: 'pkr',
    Pyg: 'pyg',
    Qar: 'qar',
    Ron: 'ron',
    Rsd: 'rsd',
    Rwf: 'rwf',
    Sbd: 'sbd',
    Scr: 'scr',
    Shp: 'shp',
    Srd: 'srd',
    Std: 'std',
    Szl: 'szl',
    Tjs: 'tjs',
    Top: 'top',
    Ttd: 'ttd',
    Twd: 'twd',
    Tzs: 'tzs',
    Uah: 'uah',
    Ugx: 'ugx',
    Uyu: 'uyu',
    Uzs: 'uzs',
    Vnd: 'vnd',
    Vuv: 'vuv',
    Wst: 'wst',
    Xaf: 'xaf',
    Xcd: 'xcd',
    Xof: 'xof',
    Xpf: 'xpf',
    Zmw: 'zmw'
};
function instanceOfCurrencyEnum(value) {
    for (const key in exports.CurrencyEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.CurrencyEnum, key)) {
            if (exports.CurrencyEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CurrencyEnumFromJSON(json) {
    return CurrencyEnumFromJSONTyped(json, false);
}
function CurrencyEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CurrencyEnumToJSON(value) {
    return value;
}
function CurrencyEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
