// @getopenpay/client/models/CustomerLanguage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerLanguage = void 0;
exports.instanceOfCustomerLanguage = instanceOfCustomerLanguage;
exports.CustomerLanguageFromJSON = CustomerLanguageFromJSON;
exports.CustomerLanguageFromJSONTyped = CustomerLanguageFromJSONTyped;
exports.CustomerLanguageToJSON = CustomerLanguageToJSON;
exports.CustomerLanguageToJSONTyped = CustomerLanguageToJSONTyped;
/**
 *
 * @export
 */
exports.CustomerLanguage = {
    En: 'en',
    EnUs: 'en_us',
    Pt: 'pt',
    Es: 'es',
    De: 'de',
    Fr: 'fr',
    It: 'it',
    Nl: 'nl',
    Ro: 'ro',
    Zh: 'zh',
    ZhHk: 'zh_hk',
    ZhTw: 'zh_tw',
    Ru: 'ru',
    Ja: 'ja',
    Ko: 'ko',
    Ar: 'ar',
    Tr: 'tr',
    Hi: 'hi'
};
function instanceOfCustomerLanguage(value) {
    for (const key in exports.CustomerLanguage) {
        if (Object.prototype.hasOwnProperty.call(exports.CustomerLanguage, key)) {
            if (exports.CustomerLanguage[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function CustomerLanguageFromJSON(json) {
    return CustomerLanguageFromJSONTyped(json, false);
}
function CustomerLanguageFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function CustomerLanguageToJSON(value) {
    return value;
}
function CustomerLanguageToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
