// @getopenpay/client/models/DeleteProductResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteProductResponse = instanceOfDeleteProductResponse;
exports.DeleteProductResponseFromJSON = DeleteProductResponseFromJSON;
exports.DeleteProductResponseFromJSONTyped = DeleteProductResponseFromJSONTyped;
exports.DeleteProductResponseToJSON = DeleteProductResponseToJSON;
exports.DeleteProductResponseToJSONTyped = DeleteProductResponseToJSONTyped;
/**
 * Check if a given object implements the DeleteProductResponse interface.
 */
function instanceOfDeleteProductResponse(value) {
    if (!('productId' in value) || value['productId'] === undefined)
        return false;
    return true;
}
function DeleteProductResponseFromJSON(json) {
    return DeleteProductResponseFromJSONTyped(json, false);
}
function DeleteProductResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'] == null ? undefined : json['message'],
        'productId': json['product_id'],
    };
}
function DeleteProductResponseToJSON(json) {
    return DeleteProductResponseToJSONTyped(json, false);
}
function DeleteProductResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'product_id': value['productId'],
    };
}
