// @getopenpay/client/models/DeletePriceResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeletePriceResponse = instanceOfDeletePriceResponse;
exports.DeletePriceResponseFromJSON = DeletePriceResponseFromJSON;
exports.DeletePriceResponseFromJSONTyped = DeletePriceResponseFromJSONTyped;
exports.DeletePriceResponseToJSON = DeletePriceResponseToJSON;
exports.DeletePriceResponseToJSONTyped = DeletePriceResponseToJSONTyped;
/**
 * Check if a given object implements the DeletePriceResponse interface.
 */
function instanceOfDeletePriceResponse(value) {
    if (!('priceId' in value) || value['priceId'] === undefined)
        return false;
    return true;
}
function DeletePriceResponseFromJSON(json) {
    return DeletePriceResponseFromJSONTyped(json, false);
}
function DeletePriceResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'] == null ? undefined : json['message'],
        'priceId': json['price_id'],
    };
}
function DeletePriceResponseToJSON(json) {
    return DeletePriceResponseToJSONTyped(json, false);
}
function DeletePriceResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'price_id': value['priceId'],
    };
}
