// @getopenpay/client/models/EnumConfigField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumConfigFieldTypeEnum = void 0;
exports.instanceOfEnumConfigField = instanceOfEnumConfigField;
exports.EnumConfigFieldFromJSON = EnumConfigFieldFromJSON;
exports.EnumConfigFieldFromJSONTyped = EnumConfigFieldFromJSONTyped;
exports.EnumConfigFieldToJSON = EnumConfigFieldToJSON;
exports.EnumConfigFieldToJSONTyped = EnumConfigFieldToJSONTyped;
/**
 * @export
 */
exports.EnumConfigFieldTypeEnum = {
    Enum: 'enum'
};
/**
 * Check if a given object implements the EnumConfigField interface.
 */
function instanceOfEnumConfigField(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('options' in value) || value['options'] === undefined)
        return false;
    if (!('value' in value) || value['value'] === undefined)
        return false;
    return true;
}
function EnumConfigFieldFromJSON(json) {
    return EnumConfigFieldFromJSONTyped(json, false);
}
function EnumConfigFieldFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'defaultValue': json['default_value'] == null ? undefined : json['default_value'],
        'description': json['description'] == null ? undefined : json['description'],
        'key': json['key'],
        'name': json['name'],
        'options': json['options'],
        'type': json['type'] == null ? undefined : json['type'],
        'value': json['value'],
    };
}
function EnumConfigFieldToJSON(json) {
    return EnumConfigFieldToJSONTyped(json, false);
}
function EnumConfigFieldToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'default_value': value['defaultValue'],
        'description': value['description'],
        'key': value['key'],
        'name': value['name'],
        'options': value['options'],
        'type': value['type'],
        'value': value['value'],
    };
}
