// @getopenpay/client/models/EventRetriggerResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfEventRetriggerResponse = instanceOfEventRetriggerResponse;
exports.EventRetriggerResponseFromJSON = EventRetriggerResponseFromJSON;
exports.EventRetriggerResponseFromJSONTyped = EventRetriggerResponseFromJSONTyped;
exports.EventRetriggerResponseToJSON = EventRetriggerResponseToJSON;
exports.EventRetriggerResponseToJSONTyped = EventRetriggerResponseToJSONTyped;
const runtime_1 = require("../runtime");
const WebhookDeliveryResult_1 = require("./WebhookDeliveryResult");
/**
 * Check if a given object implements the EventRetriggerResponse interface.
 */
function instanceOfEventRetriggerResponse(value) {
    if (!('webhookDeliveryResults' in value) || value['webhookDeliveryResults'] === undefined)
        return false;
    return true;
}
function EventRetriggerResponseFromJSON(json) {
    return EventRetriggerResponseFromJSONTyped(json, false);
}
function EventRetriggerResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'webhookDeliveryResults': ((0, runtime_1.mapValues)(json['webhook_delivery_results'], WebhookDeliveryResult_1.WebhookDeliveryResultFromJSON)),
    };
}
function EventRetriggerResponseToJSON(json) {
    return EventRetriggerResponseToJSONTyped(json, false);
}
function EventRetriggerResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'webhook_delivery_results': ((0, runtime_1.mapValues)(value['webhookDeliveryResults'], WebhookDeliveryResult_1.WebhookDeliveryResultToJSON)),
    };
}
