// @getopenpay/client/models/FallbackConfigurationOutput.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFallbackConfigurationOutput = instanceOfFallbackConfigurationOutput;
exports.FallbackConfigurationOutputFromJSON = FallbackConfigurationOutputFromJSON;
exports.FallbackConfigurationOutputFromJSONTyped = FallbackConfigurationOutputFromJSONTyped;
exports.FallbackConfigurationOutputToJSON = FallbackConfigurationOutputToJSON;
exports.FallbackConfigurationOutputToJSONTyped = FallbackConfigurationOutputToJSONTyped;
const CheckoutProcessorsPreferences_1 = require("./CheckoutProcessorsPreferences");
const SelectedPriceQuantity_1 = require("./SelectedPriceQuantity");
/**
 * Check if a given object implements the FallbackConfigurationOutput interface.
 */
function instanceOfFallbackConfigurationOutput(value) {
    return true;
}
function FallbackConfigurationOutputFromJSON(json) {
    return FallbackConfigurationOutputFromJSONTyped(json, false);
}
function FallbackConfigurationOutputFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fallbackProcessorPreferences': json['fallback_processor_preferences'] == null ? undefined : (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesFromJSON)(json['fallback_processor_preferences']),
        'priceGroups': json['price_groups'] == null ? undefined : (json['price_groups'].map(SelectedPriceQuantity_1.SelectedPriceQuantityFromJSON)),
    };
}
function FallbackConfigurationOutputToJSON(json) {
    return FallbackConfigurationOutputToJSONTyped(json, false);
}
function FallbackConfigurationOutputToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'fallback_processor_preferences': (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesToJSON)(value['fallbackProcessorPreferences']),
        'price_groups': value['priceGroups'] == null ? undefined : (value['priceGroups'].map(SelectedPriceQuantity_1.SelectedPriceQuantityToJSON)),
    };
}
