// @getopenpay/client/models/FallbackConfigurationInput.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFallbackConfigurationInput = instanceOfFallbackConfigurationInput;
exports.FallbackConfigurationInputFromJSON = FallbackConfigurationInputFromJSON;
exports.FallbackConfigurationInputFromJSONTyped = FallbackConfigurationInputFromJSONTyped;
exports.FallbackConfigurationInputToJSON = FallbackConfigurationInputToJSON;
exports.FallbackConfigurationInputToJSONTyped = FallbackConfigurationInputToJSONTyped;
const CheckoutProcessorsPreferences_1 = require("./CheckoutProcessorsPreferences");
const SelectedPriceQuantity_1 = require("./SelectedPriceQuantity");
/**
 * Check if a given object implements the FallbackConfigurationInput interface.
 */
function instanceOfFallbackConfigurationInput(value) {
    return true;
}
function FallbackConfigurationInputFromJSON(json) {
    return FallbackConfigurationInputFromJSONTyped(json, false);
}
function FallbackConfigurationInputFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fallbackProcessorPreferences': json['fallback_processor_preferences'] == null ? undefined : (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesFromJSON)(json['fallback_processor_preferences']),
        'priceGroups': json['price_groups'] == null ? undefined : (json['price_groups'].map(SelectedPriceQuantity_1.SelectedPriceQuantityFromJSON)),
    };
}
function FallbackConfigurationInputToJSON(json) {
    return FallbackConfigurationInputToJSONTyped(json, false);
}
function FallbackConfigurationInputToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'fallback_processor_preferences': (0, CheckoutProcessorsPreferences_1.CheckoutProcessorsPreferencesToJSON)(value['fallbackProcessorPreferences']),
        'price_groups': value['priceGroups'] == null ? undefined : (value['priceGroups'].map(SelectedPriceQuantity_1.SelectedPriceQuantityToJSON)),
    };
}
