// @getopenpay/client/models/IntRangeFilter.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfIntRangeFilter = instanceOfIntRangeFilter;
exports.IntRangeFilterFromJSON = IntRangeFilterFromJSON;
exports.IntRangeFilterFromJSONTyped = IntRangeFilterFromJSONTyped;
exports.IntRangeFilterToJSON = IntRangeFilterToJSON;
exports.IntRangeFilterToJSONTyped = IntRangeFilterToJSONTyped;
/**
 * Check if a given object implements the IntRangeFilter interface.
 */
function instanceOfIntRangeFilter(value) {
    return true;
}
function IntRangeFilterFromJSON(json) {
    return IntRangeFilterFromJSONTyped(json, false);
}
function IntRangeFilterFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'eq': json['eq'] == null ? undefined : json['eq'],
        'gt': json['gt'] == null ? undefined : json['gt'],
        'gte': json['gte'] == null ? undefined : json['gte'],
        'lt': json['lt'] == null ? undefined : json['lt'],
        'lte': json['lte'] == null ? undefined : json['lte'],
    };
}
function IntRangeFilterToJSON(json) {
    return IntRangeFilterToJSONTyped(json, false);
}
function IntRangeFilterToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'eq': value['eq'],
        'gt': value['gt'],
        'gte': value['gte'],
        'lt': value['lt'],
        'lte': value['lte'],
    };
}
