// @getopenpay/client/models/InvoiceItemPublic.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfInvoiceItemPublic = instanceOfInvoiceItemPublic;
exports.InvoiceItemPublicFromJSON = InvoiceItemPublicFromJSON;
exports.InvoiceItemPublicFromJSONTyped = InvoiceItemPublicFromJSONTyped;
exports.InvoiceItemPublicToJSON = InvoiceItemPublicToJSON;
exports.InvoiceItemPublicToJSONTyped = InvoiceItemPublicToJSONTyped;
const InvoiceItemDiscountAmountsPublic_1 = require("./InvoiceItemDiscountAmountsPublic");
const CurrencyEnum_1 = require("./CurrencyEnum");
const PriceTierPublic_1 = require("./PriceTierPublic");
/**
 * Check if a given object implements the InvoiceItemPublic interface.
 */
function instanceOfInvoiceItemPublic(value) {
    if (!('amountAtom' in value) || value['amountAtom'] === undefined)
        return false;
    if (!('amountAtomConsideringDiscountApplied' in value) || value['amountAtomConsideringDiscountApplied'] === undefined)
        return false;
    if (!('currency' in value) || value['currency'] === undefined)
        return false;
    if (!('discountAmountAtoms' in value) || value['discountAmountAtoms'] === undefined)
        return false;
    if (!('discounts' in value) || value['discounts'] === undefined)
        return false;
    if (!('invoiceItemDescription' in value) || value['invoiceItemDescription'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('periodEnd' in value) || value['periodEnd'] === undefined)
        return false;
    if (!('quantity' in value) || value['quantity'] === undefined)
        return false;
    return true;
}
function InvoiceItemPublicFromJSON(json) {
    return InvoiceItemPublicFromJSONTyped(json, false);
}
function InvoiceItemPublicFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amountAtom': json['amount_atom'],
        'amountAtomConsideringDiscountApplied': json['amount_atom_considering_discount_applied'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumFromJSON)(json['currency']),
        'discountAmountAtoms': (json['discount_amount_atoms'].map(InvoiceItemDiscountAmountsPublic_1.InvoiceItemDiscountAmountsPublicFromJSON)),
        'discounts': json['discounts'],
        'invoiceItemDescription': json['invoice_item_description'],
        'name': json['name'],
        'periodEnd': (new Date(json['period_end'])),
        'priceTiers': json['price_tiers'] == null ? undefined : (json['price_tiers'].map(PriceTierPublic_1.PriceTierPublicFromJSON)),
        'quantity': json['quantity'],
        'subscriptionItemDescription': json['subscription_item_description'] == null ? undefined : json['subscription_item_description'],
    };
}
function InvoiceItemPublicToJSON(json) {
    return InvoiceItemPublicToJSONTyped(json, false);
}
function InvoiceItemPublicToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'amount_atom': value['amountAtom'],
        'amount_atom_considering_discount_applied': value['amountAtomConsideringDiscountApplied'],
        'currency': (0, CurrencyEnum_1.CurrencyEnumToJSON)(value['currency']),
        'discount_amount_atoms': (value['discountAmountAtoms'].map(InvoiceItemDiscountAmountsPublic_1.InvoiceItemDiscountAmountsPublicToJSON)),
        'discounts': value['discounts'],
        'invoice_item_description': value['invoiceItemDescription'],
        'name': value['name'],
        'period_end': ((value['periodEnd']).toISOString()),
        'price_tiers': value['priceTiers'] == null ? undefined : (value['priceTiers'].map(PriceTierPublic_1.PriceTierPublicToJSON)),
        'quantity': value['quantity'],
        'subscription_item_description': value['subscriptionItemDescription'],
    };
}
