// @getopenpay/client/models/InvoiceType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceType = void 0;
exports.instanceOfInvoiceType = instanceOfInvoiceType;
exports.InvoiceTypeFromJSON = InvoiceTypeFromJSON;
exports.InvoiceTypeFromJSONTyped = InvoiceTypeFromJSONTyped;
exports.InvoiceTypeToJSON = InvoiceTypeToJSON;
exports.InvoiceTypeToJSONTyped = InvoiceTypeToJSONTyped;
/**
 *
 * @export
 */
exports.InvoiceType = {
    Standard: 'standard',
    OneOff: 'one_off'
};
function instanceOfInvoiceType(value) {
    for (const key in exports.InvoiceType) {
        if (Object.prototype.hasOwnProperty.call(exports.InvoiceType, key)) {
            if (exports.InvoiceType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function InvoiceTypeFromJSON(json) {
    return InvoiceTypeFromJSONTyped(json, false);
}
function InvoiceTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function InvoiceTypeToJSON(value) {
    return value;
}
function InvoiceTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
