// @getopenpay/client/models/InvoiceItemsQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfInvoiceItemsQueryParams = instanceOfInvoiceItemsQueryParams;
exports.InvoiceItemsQueryParamsFromJSON = InvoiceItemsQueryParamsFromJSON;
exports.InvoiceItemsQueryParamsFromJSONTyped = InvoiceItemsQueryParamsFromJSONTyped;
exports.InvoiceItemsQueryParamsToJSON = InvoiceItemsQueryParamsToJSON;
exports.InvoiceItemsQueryParamsToJSONTyped = InvoiceItemsQueryParamsToJSONTyped;
const DateTimeFilter_1 = require("./DateTimeFilter");
/**
 * Check if a given object implements the InvoiceItemsQueryParams interface.
 */
function instanceOfInvoiceItemsQueryParams(value) {
    return true;
}
function InvoiceItemsQueryParamsFromJSON(json) {
    return InvoiceItemsQueryParamsFromJSONTyped(json, false);
}
function InvoiceItemsQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'couponId': json['coupon_id'] == null ? undefined : json['coupon_id'],
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'customerId': json['customer_id'] == null ? undefined : json['customer_id'],
        'expand': json['expand'] == null ? undefined : json['expand'],
        'invoiceId': json['invoice_id'] == null ? undefined : json['invoice_id'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function InvoiceItemsQueryParamsToJSON(json) {
    return InvoiceItemsQueryParamsToJSONTyped(json, false);
}
function InvoiceItemsQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'coupon_id': value['couponId'],
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'customer_id': value['customerId'],
        'expand': value['expand'],
        'invoice_id': value['invoiceId'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
