// @getopenpay/client/models/ListResponseCreditNoteExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseCreditNoteExternal = instanceOfListResponseCreditNoteExternal;
exports.ListResponseCreditNoteExternalFromJSON = ListResponseCreditNoteExternalFromJSON;
exports.ListResponseCreditNoteExternalFromJSONTyped = ListResponseCreditNoteExternalFromJSONTyped;
exports.ListResponseCreditNoteExternalToJSON = ListResponseCreditNoteExternalToJSON;
exports.ListResponseCreditNoteExternalToJSONTyped = ListResponseCreditNoteExternalToJSONTyped;
const CreditNoteExternal_1 = require("./CreditNoteExternal");
/**
 * Check if a given object implements the ListResponseCreditNoteExternal interface.
 */
function instanceOfListResponseCreditNoteExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseCreditNoteExternalFromJSON(json) {
    return ListResponseCreditNoteExternalFromJSONTyped(json, false);
}
function ListResponseCreditNoteExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(CreditNoteExternal_1.CreditNoteExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseCreditNoteExternalToJSON(json) {
    return ListResponseCreditNoteExternalToJSONTyped(json, false);
}
function ListResponseCreditNoteExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(CreditNoteExternal_1.CreditNoteExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
