// @getopenpay/client/models/ListResponseInvoiceExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseInvoiceExternal = instanceOfListResponseInvoiceExternal;
exports.ListResponseInvoiceExternalFromJSON = ListResponseInvoiceExternalFromJSON;
exports.ListResponseInvoiceExternalFromJSONTyped = ListResponseInvoiceExternalFromJSONTyped;
exports.ListResponseInvoiceExternalToJSON = ListResponseInvoiceExternalToJSON;
exports.ListResponseInvoiceExternalToJSONTyped = ListResponseInvoiceExternalToJSONTyped;
const InvoiceExternal_1 = require("./InvoiceExternal");
/**
 * Check if a given object implements the ListResponseInvoiceExternal interface.
 */
function instanceOfListResponseInvoiceExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseInvoiceExternalFromJSON(json) {
    return ListResponseInvoiceExternalFromJSONTyped(json, false);
}
function ListResponseInvoiceExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(InvoiceExternal_1.InvoiceExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseInvoiceExternalToJSON(json) {
    return ListResponseInvoiceExternalToJSONTyped(json, false);
}
function ListResponseInvoiceExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(InvoiceExternal_1.InvoiceExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
