// @getopenpay/client/models/ListResponseBillingMeterExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseBillingMeterExternal = instanceOfListResponseBillingMeterExternal;
exports.ListResponseBillingMeterExternalFromJSON = ListResponseBillingMeterExternalFromJSON;
exports.ListResponseBillingMeterExternalFromJSONTyped = ListResponseBillingMeterExternalFromJSONTyped;
exports.ListResponseBillingMeterExternalToJSON = ListResponseBillingMeterExternalToJSON;
exports.ListResponseBillingMeterExternalToJSONTyped = ListResponseBillingMeterExternalToJSONTyped;
const BillingMeterExternal_1 = require("./BillingMeterExternal");
/**
 * Check if a given object implements the ListResponseBillingMeterExternal interface.
 */
function instanceOfListResponseBillingMeterExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseBillingMeterExternalFromJSON(json) {
    return ListResponseBillingMeterExternalFromJSONTyped(json, false);
}
function ListResponseBillingMeterExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(BillingMeterExternal_1.BillingMeterExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseBillingMeterExternalToJSON(json) {
    return ListResponseBillingMeterExternalToJSONTyped(json, false);
}
function ListResponseBillingMeterExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(BillingMeterExternal_1.BillingMeterExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
