// @getopenpay/client/models/ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseUnionTokenizedCreditCardExternalPaymentMethodExternal = instanceOfListResponseUnionTokenizedCreditCardExternalPaymentMethodExternal;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalFromJSON = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalFromJSON;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalFromJSONTyped = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalFromJSONTyped;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalToJSON = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalToJSON;
exports.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalToJSONTyped = ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalToJSONTyped;
const ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner_1 = require("./ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner");
/**
 * Check if a given object implements the ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternal interface.
 */
function instanceOfListResponseUnionTokenizedCreditCardExternalPaymentMethodExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalFromJSON(json) {
    return ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalFromJSONTyped(json, false);
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner_1.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalToJSON(json) {
    return ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalToJSONTyped(json, false);
}
function ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInner_1.ListResponseUnionTokenizedCreditCardExternalPaymentMethodExternalDataInnerToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
