// @getopenpay/client/models/ListResponseBillingMeterSummary.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseBillingMeterSummary = instanceOfListResponseBillingMeterSummary;
exports.ListResponseBillingMeterSummaryFromJSON = ListResponseBillingMeterSummaryFromJSON;
exports.ListResponseBillingMeterSummaryFromJSONTyped = ListResponseBillingMeterSummaryFromJSONTyped;
exports.ListResponseBillingMeterSummaryToJSON = ListResponseBillingMeterSummaryToJSON;
exports.ListResponseBillingMeterSummaryToJSONTyped = ListResponseBillingMeterSummaryToJSONTyped;
const BillingMeterSummary_1 = require("./BillingMeterSummary");
/**
 * Check if a given object implements the ListResponseBillingMeterSummary interface.
 */
function instanceOfListResponseBillingMeterSummary(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseBillingMeterSummaryFromJSON(json) {
    return ListResponseBillingMeterSummaryFromJSONTyped(json, false);
}
function ListResponseBillingMeterSummaryFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(BillingMeterSummary_1.BillingMeterSummaryFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseBillingMeterSummaryToJSON(json) {
    return ListResponseBillingMeterSummaryToJSONTyped(json, false);
}
function ListResponseBillingMeterSummaryToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(BillingMeterSummary_1.BillingMeterSummaryToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
