// @getopenpay/client/models/ListResponseEventExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseEventExternal = instanceOfListResponseEventExternal;
exports.ListResponseEventExternalFromJSON = ListResponseEventExternalFromJSON;
exports.ListResponseEventExternalFromJSONTyped = ListResponseEventExternalFromJSONTyped;
exports.ListResponseEventExternalToJSON = ListResponseEventExternalToJSON;
exports.ListResponseEventExternalToJSONTyped = ListResponseEventExternalToJSONTyped;
const EventExternal_1 = require("./EventExternal");
/**
 * Check if a given object implements the ListResponseEventExternal interface.
 */
function instanceOfListResponseEventExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseEventExternalFromJSON(json) {
    return ListResponseEventExternalFromJSONTyped(json, false);
}
function ListResponseEventExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(EventExternal_1.EventExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseEventExternalToJSON(json) {
    return ListResponseEventExternalToJSONTyped(json, false);
}
function ListResponseEventExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(EventExternal_1.EventExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
