// @getopenpay/client/models/ListResponseSubscriptionItemExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseSubscriptionItemExternal = instanceOfListResponseSubscriptionItemExternal;
exports.ListResponseSubscriptionItemExternalFromJSON = ListResponseSubscriptionItemExternalFromJSON;
exports.ListResponseSubscriptionItemExternalFromJSONTyped = ListResponseSubscriptionItemExternalFromJSONTyped;
exports.ListResponseSubscriptionItemExternalToJSON = ListResponseSubscriptionItemExternalToJSON;
exports.ListResponseSubscriptionItemExternalToJSONTyped = ListResponseSubscriptionItemExternalToJSONTyped;
const SubscriptionItemExternal_1 = require("./SubscriptionItemExternal");
/**
 * Check if a given object implements the ListResponseSubscriptionItemExternal interface.
 */
function instanceOfListResponseSubscriptionItemExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseSubscriptionItemExternalFromJSON(json) {
    return ListResponseSubscriptionItemExternalFromJSONTyped(json, false);
}
function ListResponseSubscriptionItemExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(SubscriptionItemExternal_1.SubscriptionItemExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseSubscriptionItemExternalToJSON(json) {
    return ListResponseSubscriptionItemExternalToJSONTyped(json, false);
}
function ListResponseSubscriptionItemExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(SubscriptionItemExternal_1.SubscriptionItemExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
