// @getopenpay/client/models/ListResponsePromotionCodeExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponsePromotionCodeExternal = instanceOfListResponsePromotionCodeExternal;
exports.ListResponsePromotionCodeExternalFromJSON = ListResponsePromotionCodeExternalFromJSON;
exports.ListResponsePromotionCodeExternalFromJSONTyped = ListResponsePromotionCodeExternalFromJSONTyped;
exports.ListResponsePromotionCodeExternalToJSON = ListResponsePromotionCodeExternalToJSON;
exports.ListResponsePromotionCodeExternalToJSONTyped = ListResponsePromotionCodeExternalToJSONTyped;
const PromotionCodeExternal_1 = require("./PromotionCodeExternal");
/**
 * Check if a given object implements the ListResponsePromotionCodeExternal interface.
 */
function instanceOfListResponsePromotionCodeExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponsePromotionCodeExternalFromJSON(json) {
    return ListResponsePromotionCodeExternalFromJSONTyped(json, false);
}
function ListResponsePromotionCodeExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(PromotionCodeExternal_1.PromotionCodeExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponsePromotionCodeExternalToJSON(json) {
    return ListResponsePromotionCodeExternalToJSONTyped(json, false);
}
function ListResponsePromotionCodeExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(PromotionCodeExternal_1.PromotionCodeExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
