// @getopenpay/client/models/ListResponseRefundExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseRefundExternal = instanceOfListResponseRefundExternal;
exports.ListResponseRefundExternalFromJSON = ListResponseRefundExternalFromJSON;
exports.ListResponseRefundExternalFromJSONTyped = ListResponseRefundExternalFromJSONTyped;
exports.ListResponseRefundExternalToJSON = ListResponseRefundExternalToJSON;
exports.ListResponseRefundExternalToJSONTyped = ListResponseRefundExternalToJSONTyped;
const RefundExternal_1 = require("./RefundExternal");
/**
 * Check if a given object implements the ListResponseRefundExternal interface.
 */
function instanceOfListResponseRefundExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseRefundExternalFromJSON(json) {
    return ListResponseRefundExternalFromJSONTyped(json, false);
}
function ListResponseRefundExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(RefundExternal_1.RefundExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseRefundExternalToJSON(json) {
    return ListResponseRefundExternalToJSONTyped(json, false);
}
function ListResponseRefundExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(RefundExternal_1.RefundExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
