// @getopenpay/client/models/ListResponseCheckoutSessionExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseCheckoutSessionExternal = instanceOfListResponseCheckoutSessionExternal;
exports.ListResponseCheckoutSessionExternalFromJSON = ListResponseCheckoutSessionExternalFromJSON;
exports.ListResponseCheckoutSessionExternalFromJSONTyped = ListResponseCheckoutSessionExternalFromJSONTyped;
exports.ListResponseCheckoutSessionExternalToJSON = ListResponseCheckoutSessionExternalToJSON;
exports.ListResponseCheckoutSessionExternalToJSONTyped = ListResponseCheckoutSessionExternalToJSONTyped;
const CheckoutSessionExternal_1 = require("./CheckoutSessionExternal");
/**
 * Check if a given object implements the ListResponseCheckoutSessionExternal interface.
 */
function instanceOfListResponseCheckoutSessionExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseCheckoutSessionExternalFromJSON(json) {
    return ListResponseCheckoutSessionExternalFromJSONTyped(json, false);
}
function ListResponseCheckoutSessionExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(CheckoutSessionExternal_1.CheckoutSessionExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseCheckoutSessionExternalToJSON(json) {
    return ListResponseCheckoutSessionExternalToJSONTyped(json, false);
}
function ListResponseCheckoutSessionExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(CheckoutSessionExternal_1.CheckoutSessionExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
