// @getopenpay/client/models/ListResponseCouponExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponseCouponExternal = instanceOfListResponseCouponExternal;
exports.ListResponseCouponExternalFromJSON = ListResponseCouponExternalFromJSON;
exports.ListResponseCouponExternalFromJSONTyped = ListResponseCouponExternalFromJSONTyped;
exports.ListResponseCouponExternalToJSON = ListResponseCouponExternalToJSON;
exports.ListResponseCouponExternalToJSONTyped = ListResponseCouponExternalToJSONTyped;
const CouponExternal_1 = require("./CouponExternal");
/**
 * Check if a given object implements the ListResponseCouponExternal interface.
 */
function instanceOfListResponseCouponExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponseCouponExternalFromJSON(json) {
    return ListResponseCouponExternalFromJSONTyped(json, false);
}
function ListResponseCouponExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(CouponExternal_1.CouponExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponseCouponExternalToJSON(json) {
    return ListResponseCouponExternalToJSONTyped(json, false);
}
function ListResponseCouponExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(CouponExternal_1.CouponExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
