// @getopenpay/client/models/ListResponsePaymentIntentExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListResponsePaymentIntentExternal = instanceOfListResponsePaymentIntentExternal;
exports.ListResponsePaymentIntentExternalFromJSON = ListResponsePaymentIntentExternalFromJSON;
exports.ListResponsePaymentIntentExternalFromJSONTyped = ListResponsePaymentIntentExternalFromJSONTyped;
exports.ListResponsePaymentIntentExternalToJSON = ListResponsePaymentIntentExternalToJSON;
exports.ListResponsePaymentIntentExternalToJSONTyped = ListResponsePaymentIntentExternalToJSONTyped;
const PaymentIntentExternal_1 = require("./PaymentIntentExternal");
/**
 * Check if a given object implements the ListResponsePaymentIntentExternal interface.
 */
function instanceOfListResponsePaymentIntentExternal(value) {
    if (!('data' in value) || value['data'] === undefined)
        return false;
    if (!('pageNumber' in value) || value['pageNumber'] === undefined)
        return false;
    if (!('pageSize' in value) || value['pageSize'] === undefined)
        return false;
    if (!('totalObjects' in value) || value['totalObjects'] === undefined)
        return false;
    return true;
}
function ListResponsePaymentIntentExternalFromJSON(json) {
    return ListResponsePaymentIntentExternalFromJSONTyped(json, false);
}
function ListResponsePaymentIntentExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'data': (json['data'].map(PaymentIntentExternal_1.PaymentIntentExternalFromJSON)),
        'pageNumber': json['page_number'],
        'pageSize': json['page_size'],
        'totalObjects': json['total_objects'],
    };
}
function ListResponsePaymentIntentExternalToJSON(json) {
    return ListResponsePaymentIntentExternalToJSONTyped(json, false);
}
function ListResponsePaymentIntentExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'data': (value['data'].map(PaymentIntentExternal_1.PaymentIntentExternalToJSON)),
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'total_objects': value['totalObjects'],
    };
}
