// @getopenpay/client/models/MarkVoidRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMarkVoidRequest = instanceOfMarkVoidRequest;
exports.MarkVoidRequestFromJSON = MarkVoidRequestFromJSON;
exports.MarkVoidRequestFromJSONTyped = MarkVoidRequestFromJSONTyped;
exports.MarkVoidRequestToJSON = MarkVoidRequestToJSON;
exports.MarkVoidRequestToJSONTyped = MarkVoidRequestToJSONTyped;
/**
 * Check if a given object implements the MarkVoidRequest interface.
 */
function instanceOfMarkVoidRequest(value) {
    return true;
}
function MarkVoidRequestFromJSON(json) {
    return MarkVoidRequestFromJSONTyped(json, false);
}
function MarkVoidRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'comment': json['comment'] == null ? undefined : json['comment'],
    };
}
function MarkVoidRequestToJSON(json) {
    return MarkVoidRequestToJSONTyped(json, false);
}
function MarkVoidRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'comment': value['comment'],
    };
}
