// @getopenpay/client/models/MarkUncollectibleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMarkUncollectibleRequest = instanceOfMarkUncollectibleRequest;
exports.MarkUncollectibleRequestFromJSON = MarkUncollectibleRequestFromJSON;
exports.MarkUncollectibleRequestFromJSONTyped = MarkUncollectibleRequestFromJSONTyped;
exports.MarkUncollectibleRequestToJSON = MarkUncollectibleRequestToJSON;
exports.MarkUncollectibleRequestToJSONTyped = MarkUncollectibleRequestToJSONTyped;
/**
 * Check if a given object implements the MarkUncollectibleRequest interface.
 */
function instanceOfMarkUncollectibleRequest(value) {
    return true;
}
function MarkUncollectibleRequestFromJSON(json) {
    return MarkUncollectibleRequestFromJSONTyped(json, false);
}
function MarkUncollectibleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'comment': json['comment'] == null ? undefined : json['comment'],
    };
}
function MarkUncollectibleRequestToJSON(json) {
    return MarkUncollectibleRequestToJSONTyped(json, false);
}
function MarkUncollectibleRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'comment': value['comment'],
    };
}
