// @getopenpay/client/models/MeterEventValueGroupingWindow.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeterEventValueGroupingWindow = void 0;
exports.instanceOfMeterEventValueGroupingWindow = instanceOfMeterEventValueGroupingWindow;
exports.MeterEventValueGroupingWindowFromJSON = MeterEventValueGroupingWindowFromJSON;
exports.MeterEventValueGroupingWindowFromJSONTyped = MeterEventValueGroupingWindowFromJSONTyped;
exports.MeterEventValueGroupingWindowToJSON = MeterEventValueGroupingWindowToJSON;
exports.MeterEventValueGroupingWindowToJSONTyped = MeterEventValueGroupingWindowToJSONTyped;
/**
 *
 * @export
 */
exports.MeterEventValueGroupingWindow = {
    Day: 'day',
    Hour: 'hour',
    None: 'none'
};
function instanceOfMeterEventValueGroupingWindow(value) {
    for (const key in exports.MeterEventValueGroupingWindow) {
        if (Object.prototype.hasOwnProperty.call(exports.MeterEventValueGroupingWindow, key)) {
            if (exports.MeterEventValueGroupingWindow[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function MeterEventValueGroupingWindowFromJSON(json) {
    return MeterEventValueGroupingWindowFromJSONTyped(json, false);
}
function MeterEventValueGroupingWindowFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function MeterEventValueGroupingWindowToJSON(value) {
    return value;
}
function MeterEventValueGroupingWindowToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
