// @getopenpay/client/models/NonPciIntegrationConfigurationInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NonPciIntegrationConfigurationInnerTypeEnum = void 0;
exports.instanceOfNonPciIntegrationConfigurationInner = instanceOfNonPciIntegrationConfigurationInner;
exports.NonPciIntegrationConfigurationInnerFromJSON = NonPciIntegrationConfigurationInnerFromJSON;
exports.NonPciIntegrationConfigurationInnerFromJSONTyped = NonPciIntegrationConfigurationInnerFromJSONTyped;
exports.NonPciIntegrationConfigurationInnerToJSON = NonPciIntegrationConfigurationInnerToJSON;
exports.NonPciIntegrationConfigurationInnerToJSONTyped = NonPciIntegrationConfigurationInnerToJSONTyped;
/**
 * @export
 */
exports.NonPciIntegrationConfigurationInnerTypeEnum = {
    String: 'string',
    Number: 'number',
    Boolean: 'boolean',
    Enum: 'enum'
};
/**
 * Check if a given object implements the NonPciIntegrationConfigurationInner interface.
 */
function instanceOfNonPciIntegrationConfigurationInner(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('value' in value) || value['value'] === undefined)
        return false;
    if (!('options' in value) || value['options'] === undefined)
        return false;
    return true;
}
function NonPciIntegrationConfigurationInnerFromJSON(json) {
    return NonPciIntegrationConfigurationInnerFromJSONTyped(json, false);
}
function NonPciIntegrationConfigurationInnerFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'defaultValue': json['default_value'] == null ? undefined : json['default_value'],
        'description': json['description'] == null ? undefined : json['description'],
        'key': json['key'],
        'maxLength': json['max_length'] == null ? undefined : json['max_length'],
        'minLength': json['min_length'] == null ? undefined : json['min_length'],
        'name': json['name'],
        'type': json['type'] == null ? undefined : json['type'],
        'value': json['value'],
        'maxValue': json['max_value'] == null ? undefined : json['max_value'],
        'minValue': json['min_value'] == null ? undefined : json['min_value'],
        'options': json['options'],
    };
}
function NonPciIntegrationConfigurationInnerToJSON(json) {
    return NonPciIntegrationConfigurationInnerToJSONTyped(json, false);
}
function NonPciIntegrationConfigurationInnerToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'default_value': value['defaultValue'],
        'description': value['description'],
        'key': value['key'],
        'max_length': value['maxLength'],
        'min_length': value['minLength'],
        'name': value['name'],
        'type': value['type'],
        'value': value['value'],
        'max_value': value['maxValue'],
        'min_value': value['minValue'],
        'options': value['options'],
    };
}
