// @getopenpay/client/models/PayInvoiceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPayInvoiceRequest = instanceOfPayInvoiceRequest;
exports.PayInvoiceRequestFromJSON = PayInvoiceRequestFromJSON;
exports.PayInvoiceRequestFromJSONTyped = PayInvoiceRequestFromJSONTyped;
exports.PayInvoiceRequestToJSON = PayInvoiceRequestToJSON;
exports.PayInvoiceRequestToJSONTyped = PayInvoiceRequestToJSONTyped;
/**
 * Check if a given object implements the PayInvoiceRequest interface.
 */
function instanceOfPayInvoiceRequest(value) {
    return true;
}
function PayInvoiceRequestFromJSON(json) {
    return PayInvoiceRequestFromJSONTyped(json, false);
}
function PayInvoiceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'comment': json['comment'] == null ? undefined : json['comment'],
        'paidOutOfBand': json['paid_out_of_band'] == null ? undefined : json['paid_out_of_band'],
        'paymentMethodId': json['payment_method_id'] == null ? undefined : json['payment_method_id'],
    };
}
function PayInvoiceRequestToJSON(json) {
    return PayInvoiceRequestToJSONTyped(json, false);
}
function PayInvoiceRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'comment': value['comment'],
        'paid_out_of_band': value['paidOutOfBand'],
        'payment_method_id': value['paymentMethodId'],
    };
}
