// @getopenpay/client/models/PaymentLinkLineItemExternal.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPaymentLinkLineItemExternal = instanceOfPaymentLinkLineItemExternal;
exports.PaymentLinkLineItemExternalFromJSON = PaymentLinkLineItemExternalFromJSON;
exports.PaymentLinkLineItemExternalFromJSONTyped = PaymentLinkLineItemExternalFromJSONTyped;
exports.PaymentLinkLineItemExternalToJSON = PaymentLinkLineItemExternalToJSON;
exports.PaymentLinkLineItemExternalToJSONTyped = PaymentLinkLineItemExternalToJSONTyped;
const ObjectName_1 = require("./ObjectName");
/**
 * Check if a given object implements the PaymentLinkLineItemExternal interface.
 */
function instanceOfPaymentLinkLineItemExternal(value) {
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('paymentLinkId' in value) || value['paymentLinkId'] === undefined)
        return false;
    if (!('priceId' in value) || value['priceId'] === undefined)
        return false;
    if (!('quantity' in value) || value['quantity'] === undefined)
        return false;
    if (!('updatedAt' in value) || value['updatedAt'] === undefined)
        return false;
    return true;
}
function PaymentLinkLineItemExternalFromJSON(json) {
    return PaymentLinkLineItemExternalFromJSONTyped(json, false);
}
function PaymentLinkLineItemExternalFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'createdAt': (new Date(json['created_at'])),
        'id': json['id'],
        'isDeleted': json['is_deleted'] == null ? undefined : json['is_deleted'],
        'object': json['object'] == null ? undefined : (0, ObjectName_1.ObjectNameFromJSON)(json['object']),
        'paymentLinkId': json['payment_link_id'],
        'priceId': json['price_id'],
        'quantity': json['quantity'],
        'updatedAt': (new Date(json['updated_at'])),
    };
}
function PaymentLinkLineItemExternalToJSON(json) {
    return PaymentLinkLineItemExternalToJSONTyped(json, false);
}
function PaymentLinkLineItemExternalToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created_at': ((value['createdAt']).toISOString()),
        'id': value['id'],
        'is_deleted': value['isDeleted'],
        'object': (0, ObjectName_1.ObjectNameToJSON)(value['object']),
        'payment_link_id': value['paymentLinkId'],
        'price_id': value['priceId'],
        'quantity': value['quantity'],
        'updated_at': ((value['updatedAt']).toISOString()),
    };
}
