// @getopenpay/client/models/PaymentLinkQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPaymentLinkQueryParams = instanceOfPaymentLinkQueryParams;
exports.PaymentLinkQueryParamsFromJSON = PaymentLinkQueryParamsFromJSON;
exports.PaymentLinkQueryParamsFromJSONTyped = PaymentLinkQueryParamsFromJSONTyped;
exports.PaymentLinkQueryParamsToJSON = PaymentLinkQueryParamsToJSON;
exports.PaymentLinkQueryParamsToJSONTyped = PaymentLinkQueryParamsToJSONTyped;
const DateTimeFilter_1 = require("./DateTimeFilter");
/**
 * Check if a given object implements the PaymentLinkQueryParams interface.
 */
function instanceOfPaymentLinkQueryParams(value) {
    return true;
}
function PaymentLinkQueryParamsFromJSON(json) {
    return PaymentLinkQueryParamsFromJSONTyped(json, false);
}
function PaymentLinkQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'expand': json['expand'] == null ? undefined : json['expand'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function PaymentLinkQueryParamsToJSON(json) {
    return PaymentLinkQueryParamsToJSONTyped(json, false);
}
function PaymentLinkQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'expand': value['expand'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
