// @getopenpay/client/models/PaymentIntentStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentIntentStatus = void 0;
exports.instanceOfPaymentIntentStatus = instanceOfPaymentIntentStatus;
exports.PaymentIntentStatusFromJSON = PaymentIntentStatusFromJSON;
exports.PaymentIntentStatusFromJSONTyped = PaymentIntentStatusFromJSONTyped;
exports.PaymentIntentStatusToJSON = PaymentIntentStatusToJSON;
exports.PaymentIntentStatusToJSONTyped = PaymentIntentStatusToJSONTyped;
/**
 *
 * @export
 */
exports.PaymentIntentStatus = {
    RequiresConfirmation: 'requires_confirmation',
    RequiresCapture: 'requires_capture',
    RequiresAction: 'requires_action',
    RequiresPaymentMethod: 'requires_payment_method',
    Processing: 'processing',
    Canceled: 'canceled',
    Succeeded: 'succeeded',
    Exception: 'exception'
};
function instanceOfPaymentIntentStatus(value) {
    for (const key in exports.PaymentIntentStatus) {
        if (Object.prototype.hasOwnProperty.call(exports.PaymentIntentStatus, key)) {
            if (exports.PaymentIntentStatus[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function PaymentIntentStatusFromJSON(json) {
    return PaymentIntentStatusFromJSONTyped(json, false);
}
function PaymentIntentStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function PaymentIntentStatusToJSON(value) {
    return value;
}
function PaymentIntentStatusToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
