// @getopenpay/client/models/PricingModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingModel = void 0;
exports.instanceOfPricingModel = instanceOfPricingModel;
exports.PricingModelFromJSON = PricingModelFromJSON;
exports.PricingModelFromJSONTyped = PricingModelFromJSONTyped;
exports.PricingModelToJSON = PricingModelToJSON;
exports.PricingModelToJSONTyped = PricingModelToJSONTyped;
/**
 *
 * @export
 */
exports.PricingModel = {
    Standard: 'standard',
    Package: 'package',
    Graduated: 'graduated',
    Volume: 'volume'
};
function instanceOfPricingModel(value) {
    for (const key in exports.PricingModel) {
        if (Object.prototype.hasOwnProperty.call(exports.PricingModel, key)) {
            if (exports.PricingModel[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function PricingModelFromJSON(json) {
    return PricingModelFromJSONTyped(json, false);
}
function PricingModelFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function PricingModelToJSON(value) {
    return value;
}
function PricingModelToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
