// @getopenpay/client/models/PricingTiersEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingTiersEnum = void 0;
exports.instanceOfPricingTiersEnum = instanceOfPricingTiersEnum;
exports.PricingTiersEnumFromJSON = PricingTiersEnumFromJSON;
exports.PricingTiersEnumFromJSONTyped = PricingTiersEnumFromJSONTyped;
exports.PricingTiersEnumToJSON = PricingTiersEnumToJSON;
exports.PricingTiersEnumToJSONTyped = PricingTiersEnumToJSONTyped;
/**
 *
 * @export
 */
exports.PricingTiersEnum = {
    Graduated: 'graduated',
    Volume: 'volume'
};
function instanceOfPricingTiersEnum(value) {
    for (const key in exports.PricingTiersEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.PricingTiersEnum, key)) {
            if (exports.PricingTiersEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function PricingTiersEnumFromJSON(json) {
    return PricingTiersEnumFromJSONTyped(json, false);
}
function PricingTiersEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function PricingTiersEnumToJSON(value) {
    return value;
}
function PricingTiersEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
