// @getopenpay/client/models/RefundStatusEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefundStatusEnum = void 0;
exports.instanceOfRefundStatusEnum = instanceOfRefundStatusEnum;
exports.RefundStatusEnumFromJSON = RefundStatusEnumFromJSON;
exports.RefundStatusEnumFromJSONTyped = RefundStatusEnumFromJSONTyped;
exports.RefundStatusEnumToJSON = RefundStatusEnumToJSON;
exports.RefundStatusEnumToJSONTyped = RefundStatusEnumToJSONTyped;
/**
 *
 * @export
 */
exports.RefundStatusEnum = {
    Succeeded: 'succeeded',
    Failed: 'failed',
    Pending: 'pending',
    RequiresAction: 'requires_action',
    Canceled: 'canceled'
};
function instanceOfRefundStatusEnum(value) {
    for (const key in exports.RefundStatusEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.RefundStatusEnum, key)) {
            if (exports.RefundStatusEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function RefundStatusEnumFromJSON(json) {
    return RefundStatusEnumFromJSONTyped(json, false);
}
function RefundStatusEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function RefundStatusEnumToJSON(value) {
    return value;
}
function RefundStatusEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
