// @getopenpay/client/models/RefundQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfRefundQueryParams = instanceOfRefundQueryParams;
exports.RefundQueryParamsFromJSON = RefundQueryParamsFromJSON;
exports.RefundQueryParamsFromJSONTyped = RefundQueryParamsFromJSONTyped;
exports.RefundQueryParamsToJSON = RefundQueryParamsToJSON;
exports.RefundQueryParamsToJSONTyped = RefundQueryParamsToJSONTyped;
const RefundStatusEnum_1 = require("./RefundStatusEnum");
const DateTimeFilter_1 = require("./DateTimeFilter");
const IntRangeFilter_1 = require("./IntRangeFilter");
const RefundReasonEnum_1 = require("./RefundReasonEnum");
/**
 * Check if a given object implements the RefundQueryParams interface.
 */
function instanceOfRefundQueryParams(value) {
    return true;
}
function RefundQueryParamsFromJSON(json) {
    return RefundQueryParamsFromJSONTyped(json, false);
}
function RefundQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amountAtom': json['amount_atom'] == null ? undefined : (0, IntRangeFilter_1.IntRangeFilterFromJSON)(json['amount_atom']),
        'chargeId': json['charge_id'] == null ? undefined : json['charge_id'],
        'createdAt': json['created_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['created_at']),
        'customerId': json['customer_id'] == null ? undefined : json['customer_id'],
        'expand': json['expand'] == null ? undefined : json['expand'],
        'pageNumber': json['page_number'] == null ? undefined : json['page_number'],
        'pageSize': json['page_size'] == null ? undefined : json['page_size'],
        'paymentIntentId': json['payment_intent_id'] == null ? undefined : json['payment_intent_id'],
        'reason': json['reason'] == null ? undefined : (0, RefundReasonEnum_1.RefundReasonEnumFromJSON)(json['reason']),
        'sortDescending': json['sort_descending'] == null ? undefined : json['sort_descending'],
        'sortKey': json['sort_key'] == null ? undefined : json['sort_key'],
        'status': json['status'] == null ? undefined : (0, RefundStatusEnum_1.RefundStatusEnumFromJSON)(json['status']),
        'updatedAt': json['updated_at'] == null ? undefined : (0, DateTimeFilter_1.DateTimeFilterFromJSON)(json['updated_at']),
    };
}
function RefundQueryParamsToJSON(json) {
    return RefundQueryParamsToJSONTyped(json, false);
}
function RefundQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'amount_atom': (0, IntRangeFilter_1.IntRangeFilterToJSON)(value['amountAtom']),
        'charge_id': value['chargeId'],
        'created_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['createdAt']),
        'customer_id': value['customerId'],
        'expand': value['expand'],
        'page_number': value['pageNumber'],
        'page_size': value['pageSize'],
        'payment_intent_id': value['paymentIntentId'],
        'reason': (0, RefundReasonEnum_1.RefundReasonEnumToJSON)(value['reason']),
        'sort_descending': value['sortDescending'],
        'sort_key': value['sortKey'],
        'status': (0, RefundStatusEnum_1.RefundStatusEnumToJSON)(value['status']),
        'updated_at': (0, DateTimeFilter_1.DateTimeFilterToJSON)(value['updatedAt']),
    };
}
