// @getopenpay/client/models/SubscriptionStatusEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionStatusEnum = void 0;
exports.instanceOfSubscriptionStatusEnum = instanceOfSubscriptionStatusEnum;
exports.SubscriptionStatusEnumFromJSON = SubscriptionStatusEnumFromJSON;
exports.SubscriptionStatusEnumFromJSONTyped = SubscriptionStatusEnumFromJSONTyped;
exports.SubscriptionStatusEnumToJSON = SubscriptionStatusEnumToJSON;
exports.SubscriptionStatusEnumToJSONTyped = SubscriptionStatusEnumToJSONTyped;
/**
 *
 * @export
 */
exports.SubscriptionStatusEnum = {
    Incomplete: 'incomplete',
    Trialing: 'trialing',
    Active: 'active',
    Paused: 'paused',
    PastDue: 'past_due',
    Canceled: 'canceled',
    Scheduled: 'scheduled'
};
function instanceOfSubscriptionStatusEnum(value) {
    for (const key in exports.SubscriptionStatusEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.SubscriptionStatusEnum, key)) {
            if (exports.SubscriptionStatusEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function SubscriptionStatusEnumFromJSON(json) {
    return SubscriptionStatusEnumFromJSONTyped(json, false);
}
function SubscriptionStatusEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function SubscriptionStatusEnumToJSON(value) {
    return value;
}
function SubscriptionStatusEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
