// @getopenpay/client/models/SubscriptionItemDetails.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSubscriptionItemDetails = instanceOfSubscriptionItemDetails;
exports.SubscriptionItemDetailsFromJSON = SubscriptionItemDetailsFromJSON;
exports.SubscriptionItemDetailsFromJSONTyped = SubscriptionItemDetailsFromJSONTyped;
exports.SubscriptionItemDetailsToJSON = SubscriptionItemDetailsToJSON;
exports.SubscriptionItemDetailsToJSONTyped = SubscriptionItemDetailsToJSONTyped;
/**
 * Check if a given object implements the SubscriptionItemDetails interface.
 */
function instanceOfSubscriptionItemDetails(value) {
    return true;
}
function SubscriptionItemDetailsFromJSON(json) {
    return SubscriptionItemDetailsFromJSONTyped(json, false);
}
function SubscriptionItemDetailsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'description': json['description'] == null ? undefined : json['description'],
    };
}
function SubscriptionItemDetailsToJSON(json) {
    return SubscriptionItemDetailsToJSONTyped(json, false);
}
function SubscriptionItemDetailsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'custom_fields': value['customFields'],
        'description': value['description'],
    };
}
