// @getopenpay/client/models/SubscriptionItemQueryParams.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSubscriptionItemQueryParams = instanceOfSubscriptionItemQueryParams;
exports.SubscriptionItemQueryParamsFromJSON = SubscriptionItemQueryParamsFromJSON;
exports.SubscriptionItemQueryParamsFromJSONTyped = SubscriptionItemQueryParamsFromJSONTyped;
exports.SubscriptionItemQueryParamsToJSON = SubscriptionItemQueryParamsToJSON;
exports.SubscriptionItemQueryParamsToJSONTyped = SubscriptionItemQueryParamsToJSONTyped;
/**
 * Check if a given object implements the SubscriptionItemQueryParams interface.
 */
function instanceOfSubscriptionItemQueryParams(value) {
    if (!('subscriptionId' in value) || value['subscriptionId'] === undefined)
        return false;
    return true;
}
function SubscriptionItemQueryParamsFromJSON(json) {
    return SubscriptionItemQueryParamsFromJSONTyped(json, false);
}
function SubscriptionItemQueryParamsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'subscriptionId': json['subscription_id'],
    };
}
function SubscriptionItemQueryParamsToJSON(json) {
    return SubscriptionItemQueryParamsToJSONTyped(json, false);
}
function SubscriptionItemQueryParamsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'subscription_id': value['subscriptionId'],
    };
}
