// @getopenpay/client/models/SubscriptionCancellationDetails.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSubscriptionCancellationDetails = instanceOfSubscriptionCancellationDetails;
exports.SubscriptionCancellationDetailsFromJSON = SubscriptionCancellationDetailsFromJSON;
exports.SubscriptionCancellationDetailsFromJSONTyped = SubscriptionCancellationDetailsFromJSONTyped;
exports.SubscriptionCancellationDetailsToJSON = SubscriptionCancellationDetailsToJSON;
exports.SubscriptionCancellationDetailsToJSONTyped = SubscriptionCancellationDetailsToJSONTyped;
const SubscriptionCancelFeedbackEnum_1 = require("./SubscriptionCancelFeedbackEnum");
/**
 * Check if a given object implements the SubscriptionCancellationDetails interface.
 */
function instanceOfSubscriptionCancellationDetails(value) {
    return true;
}
function SubscriptionCancellationDetailsFromJSON(json) {
    return SubscriptionCancellationDetailsFromJSONTyped(json, false);
}
function SubscriptionCancellationDetailsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'comment': json['comment'] == null ? undefined : json['comment'],
        'feedback': json['feedback'] == null ? undefined : (0, SubscriptionCancelFeedbackEnum_1.SubscriptionCancelFeedbackEnumFromJSON)(json['feedback']),
        'reason': json['reason'] == null ? undefined : json['reason'],
    };
}
function SubscriptionCancellationDetailsToJSON(json) {
    return SubscriptionCancellationDetailsToJSONTyped(json, false);
}
function SubscriptionCancellationDetailsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'comment': value['comment'],
        'feedback': (0, SubscriptionCancelFeedbackEnum_1.SubscriptionCancelFeedbackEnumToJSON)(value['feedback']),
        'reason': value['reason'],
    };
}
