// @getopenpay/client/models/SubscriptionCancelFeedbackEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionCancelFeedbackEnum = void 0;
exports.instanceOfSubscriptionCancelFeedbackEnum = instanceOfSubscriptionCancelFeedbackEnum;
exports.SubscriptionCancelFeedbackEnumFromJSON = SubscriptionCancelFeedbackEnumFromJSON;
exports.SubscriptionCancelFeedbackEnumFromJSONTyped = SubscriptionCancelFeedbackEnumFromJSONTyped;
exports.SubscriptionCancelFeedbackEnumToJSON = SubscriptionCancelFeedbackEnumToJSON;
exports.SubscriptionCancelFeedbackEnumToJSONTyped = SubscriptionCancelFeedbackEnumToJSONTyped;
/**
 *
 * @export
 */
exports.SubscriptionCancelFeedbackEnum = {
    TooExpensive: 'too_expensive',
    MissingFeatures: 'missing_features',
    SwitchedService: 'switched_service',
    Unused: 'unused',
    CustomerService: 'customer_service',
    TooComplex: 'too_complex',
    LowQuality: 'low_quality',
    Other: 'other'
};
function instanceOfSubscriptionCancelFeedbackEnum(value) {
    for (const key in exports.SubscriptionCancelFeedbackEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.SubscriptionCancelFeedbackEnum, key)) {
            if (exports.SubscriptionCancelFeedbackEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function SubscriptionCancelFeedbackEnumFromJSON(json) {
    return SubscriptionCancelFeedbackEnumFromJSONTyped(json, false);
}
function SubscriptionCancelFeedbackEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function SubscriptionCancelFeedbackEnumToJSON(value) {
    return value;
}
function SubscriptionCancelFeedbackEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
