// @getopenpay/client/models/TaxIdSetting.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTaxIdSetting = instanceOfTaxIdSetting;
exports.TaxIdSettingFromJSON = TaxIdSettingFromJSON;
exports.TaxIdSettingFromJSONTyped = TaxIdSettingFromJSONTyped;
exports.TaxIdSettingToJSON = TaxIdSettingToJSON;
exports.TaxIdSettingToJSONTyped = TaxIdSettingToJSONTyped;
/**
 * Check if a given object implements the TaxIdSetting interface.
 */
function instanceOfTaxIdSetting(value) {
    if (!('taxId' in value) || value['taxId'] === undefined)
        return false;
    if (!('taxIdType' in value) || value['taxIdType'] === undefined)
        return false;
    return true;
}
function TaxIdSettingFromJSON(json) {
    return TaxIdSettingFromJSONTyped(json, false);
}
function TaxIdSettingFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'taxId': json['tax_id'],
        'taxIdType': json['tax_id_type'],
    };
}
function TaxIdSettingToJSON(json) {
    return TaxIdSettingToJSONTyped(json, false);
}
function TaxIdSettingToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'tax_id': value['taxId'],
        'tax_id_type': value['taxIdType'],
    };
}
