// @getopenpay/client/models/TaxIdTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxIdTypeEnum = void 0;
exports.instanceOfTaxIdTypeEnum = instanceOfTaxIdTypeEnum;
exports.TaxIdTypeEnumFromJSON = TaxIdTypeEnumFromJSON;
exports.TaxIdTypeEnumFromJSONTyped = TaxIdTypeEnumFromJSONTyped;
exports.TaxIdTypeEnumToJSON = TaxIdTypeEnumToJSON;
exports.TaxIdTypeEnumToJSONTyped = TaxIdTypeEnumToJSONTyped;
/**
 * Enum for all supported tax ID types.
 * @export
 */
exports.TaxIdTypeEnum = {
    AlTin: 'al_tin',
    AdNrt: 'ad_nrt',
    AoTin: 'ao_tin',
    ArCuit: 'ar_cuit',
    AmTin: 'am_tin',
    AuAbn: 'au_abn',
    AuArn: 'au_arn',
    EuVat: 'eu_vat',
    BsTin: 'bs_tin',
    BhVat: 'bh_vat',
    BbTin: 'bb_tin',
    ByTin: 'by_tin',
    BoTin: 'bo_tin',
    BaTin: 'ba_tin',
    BrCnpj: 'br_cnpj',
    BrCpf: 'br_cpf',
    BgUic: 'bg_uic',
    KhTin: 'kh_tin',
    CaBn: 'ca_bn',
    CaGstHst: 'ca_gst_hst',
    CaPstBc: 'ca_pst_bc',
    CaPstMb: 'ca_pst_mb',
    CaPstSk: 'ca_pst_sk',
    CaQst: 'ca_qst',
    ClTin: 'cl_tin',
    CnTin: 'cn_tin',
    CoNit: 'co_nit',
    CdNif: 'cd_nif',
    CrTin: 'cr_tin',
    HrOib: 'hr_oib',
    DoRcn: 'do_rcn',
    EcRuc: 'ec_ruc',
    EgTin: 'eg_tin',
    SvNit: 'sv_nit',
    EuOssVat: 'eu_oss_vat',
    DeStn: 'de_stn',
    GeVat: 'ge_vat',
    GnNif: 'gn_nif',
    HkBr: 'hk_br',
    HuTin: 'hu_tin',
    IsVat: 'is_vat',
    InGst: 'in_gst',
    IdNpwp: 'id_npwp',
    IlVat: 'il_vat',
    JpCn: 'jp_cn',
    JpRn: 'jp_rn',
    JpTrn: 'jp_trn',
    KzBin: 'kz_bin',
    KePin: 'ke_pin',
    LiUid: 'li_uid',
    LiVat: 'li_vat',
    MyFrp: 'my_frp',
    MyItn: 'my_itn',
    MySst: 'my_sst',
    MrNif: 'mr_nif',
    MxRfc: 'mx_rfc',
    MdVat: 'md_vat',
    MePib: 'me_pib',
    MaVat: 'ma_vat',
    NpPan: 'np_pan',
    NzGst: 'nz_gst',
    NgTin: 'ng_tin',
    MkVat: 'mk_vat',
    NoVat: 'no_vat',
    NoVoec: 'no_voec',
    OmVat: 'om_vat',
    PeRuc: 'pe_ruc',
    PhTin: 'ph_tin',
    RoTin: 'ro_tin',
    RuInn: 'ru_inn',
    RuKpp: 'ru_kpp',
    SaVat: 'sa_vat',
    SnNinea: 'sn_ninea',
    RsPib: 'rs_pib',
    SgGst: 'sg_gst',
    SgUen: 'sg_uen',
    SiTin: 'si_tin',
    ZaVat: 'za_vat',
    KrBrn: 'kr_brn',
    EsCif: 'es_cif',
    SrFin: 'sr_fin',
    ChUid: 'ch_uid',
    ChVat: 'ch_vat',
    TwVat: 'tw_vat',
    TjTin: 'tj_tin',
    TzVat: 'tz_vat',
    ThVat: 'th_vat',
    TrTin: 'tr_tin',
    UgTin: 'ug_tin',
    UaVat: 'ua_vat',
    AeTrn: 'ae_trn',
    GbVat: 'gb_vat',
    UsEin: 'us_ein',
    UyRuc: 'uy_ruc',
    UzTin: 'uz_tin',
    UzVat: 'uz_vat',
    VeRif: 've_rif',
    VnTin: 'vn_tin',
    ZmTin: 'zm_tin',
    ZwTin: 'zw_tin'
};
function instanceOfTaxIdTypeEnum(value) {
    for (const key in exports.TaxIdTypeEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.TaxIdTypeEnum, key)) {
            if (exports.TaxIdTypeEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function TaxIdTypeEnumFromJSON(json) {
    return TaxIdTypeEnumFromJSONTyped(json, false);
}
function TaxIdTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function TaxIdTypeEnumToJSON(value) {
    return value;
}
function TaxIdTypeEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
