// @getopenpay/client/models/UpdatePromoCodeRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdatePromoCodeRequest = instanceOfUpdatePromoCodeRequest;
exports.UpdatePromoCodeRequestFromJSON = UpdatePromoCodeRequestFromJSON;
exports.UpdatePromoCodeRequestFromJSONTyped = UpdatePromoCodeRequestFromJSONTyped;
exports.UpdatePromoCodeRequestToJSON = UpdatePromoCodeRequestToJSON;
exports.UpdatePromoCodeRequestToJSONTyped = UpdatePromoCodeRequestToJSONTyped;
/**
 * Check if a given object implements the UpdatePromoCodeRequest interface.
 */
function instanceOfUpdatePromoCodeRequest(value) {
    if (!('isActive' in value) || value['isActive'] === undefined)
        return false;
    return true;
}
function UpdatePromoCodeRequestFromJSON(json) {
    return UpdatePromoCodeRequestFromJSONTyped(json, false);
}
function UpdatePromoCodeRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'isActive': json['is_active'],
    };
}
function UpdatePromoCodeRequestToJSON(json) {
    return UpdatePromoCodeRequestToJSONTyped(json, false);
}
function UpdatePromoCodeRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'is_active': value['isActive'],
    };
}
