// @getopenpay/client/models/UpdateSubscriptionItemRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateSubscriptionItemRequest = instanceOfUpdateSubscriptionItemRequest;
exports.UpdateSubscriptionItemRequestFromJSON = UpdateSubscriptionItemRequestFromJSON;
exports.UpdateSubscriptionItemRequestFromJSONTyped = UpdateSubscriptionItemRequestFromJSONTyped;
exports.UpdateSubscriptionItemRequestToJSON = UpdateSubscriptionItemRequestToJSON;
exports.UpdateSubscriptionItemRequestToJSONTyped = UpdateSubscriptionItemRequestToJSONTyped;
const ProrationEnum_1 = require("./ProrationEnum");
/**
 * Check if a given object implements the UpdateSubscriptionItemRequest interface.
 */
function instanceOfUpdateSubscriptionItemRequest(value) {
    return true;
}
function UpdateSubscriptionItemRequestFromJSON(json) {
    return UpdateSubscriptionItemRequestFromJSONTyped(json, false);
}
function UpdateSubscriptionItemRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'customFields': json['custom_fields'] == null ? undefined : json['custom_fields'],
        'description': json['description'] == null ? undefined : json['description'],
        'dropAtEnd': json['drop_at_end'] == null ? undefined : json['drop_at_end'],
        'priceId': json['price_id'] == null ? undefined : json['price_id'],
        'prorationBehavior': json['proration_behavior'] == null ? undefined : (0, ProrationEnum_1.ProrationEnumFromJSON)(json['proration_behavior']),
        'quantity': json['quantity'] == null ? undefined : json['quantity'],
    };
}
function UpdateSubscriptionItemRequestToJSON(json) {
    return UpdateSubscriptionItemRequestToJSONTyped(json, false);
}
function UpdateSubscriptionItemRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'custom_fields': value['customFields'],
        'description': value['description'],
        'drop_at_end': value['dropAtEnd'],
        'price_id': value['priceId'],
        'proration_behavior': (0, ProrationEnum_1.ProrationEnumToJSON)(value['prorationBehavior']),
        'quantity': value['quantity'],
    };
}
