// @getopenpay/client/models/UpdateProductFamilyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateProductFamilyRequest = instanceOfUpdateProductFamilyRequest;
exports.UpdateProductFamilyRequestFromJSON = UpdateProductFamilyRequestFromJSON;
exports.UpdateProductFamilyRequestFromJSONTyped = UpdateProductFamilyRequestFromJSONTyped;
exports.UpdateProductFamilyRequestToJSON = UpdateProductFamilyRequestToJSON;
exports.UpdateProductFamilyRequestToJSONTyped = UpdateProductFamilyRequestToJSONTyped;
/**
 * Check if a given object implements the UpdateProductFamilyRequest interface.
 */
function instanceOfUpdateProductFamilyRequest(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('products' in value) || value['products'] === undefined)
        return false;
    return true;
}
function UpdateProductFamilyRequestFromJSON(json) {
    return UpdateProductFamilyRequestFromJSONTyped(json, false);
}
function UpdateProductFamilyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'description': json['description'] == null ? undefined : json['description'],
        'hierarchy': json['hierarchy'] == null ? undefined : json['hierarchy'],
        'name': json['name'],
        'products': json['products'],
    };
}
function UpdateProductFamilyRequestToJSON(json) {
    return UpdateProductFamilyRequestToJSONTyped(json, false);
}
function UpdateProductFamilyRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'description': value['description'],
        'hierarchy': value['hierarchy'],
        'name': value['name'],
        'products': value['products'],
    };
}
