// @getopenpay/client/models/UsageAggMethodEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageAggMethodEnum = void 0;
exports.instanceOfUsageAggMethodEnum = instanceOfUsageAggMethodEnum;
exports.UsageAggMethodEnumFromJSON = UsageAggMethodEnumFromJSON;
exports.UsageAggMethodEnumFromJSONTyped = UsageAggMethodEnumFromJSONTyped;
exports.UsageAggMethodEnumToJSON = UsageAggMethodEnumToJSON;
exports.UsageAggMethodEnumToJSONTyped = UsageAggMethodEnumToJSONTyped;
/**
 *
 * @export
 */
exports.UsageAggMethodEnum = {
    Sum: 'sum',
    LastDuringPeriod: 'last_during_period',
    LastEver: 'last_ever',
    Max: 'max'
};
function instanceOfUsageAggMethodEnum(value) {
    for (const key in exports.UsageAggMethodEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.UsageAggMethodEnum, key)) {
            if (exports.UsageAggMethodEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function UsageAggMethodEnumFromJSON(json) {
    return UsageAggMethodEnumFromJSONTyped(json, false);
}
function UsageAggMethodEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function UsageAggMethodEnumToJSON(value) {
    return value;
}
function UsageAggMethodEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
