// @getopenpay/client/models/UsageTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * OpenPay API
 * super charge your subscription management.
 *
 * The version of the OpenAPI document: 1.2.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageTypeEnum = void 0;
exports.instanceOfUsageTypeEnum = instanceOfUsageTypeEnum;
exports.UsageTypeEnumFromJSON = UsageTypeEnumFromJSON;
exports.UsageTypeEnumFromJSONTyped = UsageTypeEnumFromJSONTyped;
exports.UsageTypeEnumToJSON = UsageTypeEnumToJSON;
exports.UsageTypeEnumToJSONTyped = UsageTypeEnumToJSONTyped;
/**
 *
 * @export
 */
exports.UsageTypeEnum = {
    Metered: 'metered',
    Licensed: 'licensed'
};
function instanceOfUsageTypeEnum(value) {
    for (const key in exports.UsageTypeEnum) {
        if (Object.prototype.hasOwnProperty.call(exports.UsageTypeEnum, key)) {
            if (exports.UsageTypeEnum[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function UsageTypeEnumFromJSON(json) {
    return UsageTypeEnumFromJSONTyped(json, false);
}
function UsageTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function UsageTypeEnumToJSON(value) {
    return value;
}
function UsageTypeEnumToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
